---
title: "R Notebook"
output: html_notebook
---

```{r}
cleaned_CaH_Cohort2 = qread("Cohort2/sobjs/cleaned_CaH_Cohort2.qs")

cleaned_Put_Cohort2 = qread("Cohort2/sobjs/cleaned_Put_Cohort2.qs")
```

```{r}
cleaned_CaH_Cohort2@meta.data
cleaned_Put_Cohort2@meta.data
```


```{r}
sobj_list = list(cleaned_CaH_Cohort2,
cleaned_Put_Cohort2)

counts_list <- lapply(sobj_list, function(sobj) GetAssayData(sobj, slot = "counts"))

merged_counts <- do.call(cbind, counts_list)

metadata_list <- lapply(sobj_list, function(sobj) sobj@meta.data)

merged_metadata <- do.call(rbind, metadata_list)
rownames(merged_metadata) = paste0(merged_metadata$library, "__", merged_metadata$cell)
# Ensure that the row names of metadata match the colnames of merged counts
merged_metadata <- merged_metadata[match(colnames(merged_counts), rownames(merged_metadata)), ]
# Reorder merged_counts to match the cell names in the merged metadata
merged_counts <- merged_counts[, match(rownames(merged_metadata), colnames(merged_counts))]
# Check if the order of rownames and colnames match
all(rownames(merged_metadata) == colnames(merged_counts))  # Should return TRUE

stopifnot(all(rownames(merged_metadata) == colnames(merged_counts)))

# Create Seurat object
cleaned_CaH_Put_Cohort2 <- CreateSeuratObject(counts = merged_counts, meta.data = merged_metadata)
cleaned_CaH_Put_Cohort2@meta.data

# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(cleaned_CaH_Put_Cohort2, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
cleaned_CaH_Put_Cohort2 = (cleaned_CaH_Put_Cohort2
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(cleaned_CaH_Put_Cohort2, group.by = "donor_id") 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "library") 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "Condition") 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "Region") 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "cell_class", label=T)
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "neuron_cell_class", label=T)
```


```{r}
library(harmony)
cleaned_CaH_Put_Cohort2 = (cleaned_CaH_Put_Cohort2
    %>% RunHarmony(
        group.by = "donor_id")
    %>% FindNeighbors(reduction='harmony', dims=1:20)
    %>% FindClusters(res=0.05)
    %>% FindClusters(res=0.1)
    %>% FindClusters(res=0.2)
    %>% FindClusters(res=0.3)
    %>% FindClusters(res=0.4)
    %>% FindClusters(res=0.5)
    %>% FindClusters(res=0.6)
    %>% FindClusters(res=0.7)
    %>% FindClusters(res=0.8)    
    %>% RunUMAP(reduction="harmony", dims=1:20)
)
cleaned_CaH_Put_Cohort2
cleaned_CaH_Put_Cohort2@meta.data
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "donor_id") 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "library") 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "Condition") 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "Region") 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.05", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.1", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "cell_class", label=T)
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "neuron_cell_class", label=T)

FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("OLIG1", "MOG", "MOBP", "MBP"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 

```


```{r}
Idents(cleaned_CaH_Put_Cohort2) = cleaned_CaH_Put_Cohort2$RNA_snn_res.0.2

classes = c("oligo", "oligo", "astrocyte", "neuron", "neuron", "microglia", "opc", "neuron", "neuron", "neuron", "neuron", "endothelial", "ependymal", "neuron", "immune")


cleaned_CaH_Put_Cohort2= assignCellClasses(cleaned_CaH_Put_Cohort2, classes=classes, cluster_col="RNA_snn_res.0.2", class_col = "final_cell_class_merged")

DimPlot(cleaned_CaH_Put_Cohort2, group.by = "final_cell_class_merged" ,label = T, raster = FALSE)


```

```{r}
qsave(cleaned_CaH_Put_Cohort2, "Cohort2/sobjs/final_CaH_Put_Cohort2.qs")
```

```{r}
cleaned_CaH_Put_Cohort2 = qread("Cohort2/sobjs/final_CaH_Put_Cohort2.qs")
```

```{r}
cleaned_CaH_Put_Cohort2@meta.data
```



```{r}
cohort_metadata = read.csv("/broad/macosko/kimkathl/XDP_Cohorts_1_2_3_metadata.csv", header = T)
cohort_metadata

cleaned_CaH_Put_Cohort2@meta.data = merge(cleaned_CaH_Put_Cohort2@meta.data, cohort_metadata, by.x = "donor_id", by.y = "Donor")
cleaned_CaH_Put_Cohort2@meta.data

cleaned_CaH_Put_Cohort2@meta.data$Sex = cleaned_CaH_Put_Cohort2@meta.data$Sex.x
cleaned_CaH_Put_Cohort2@meta.data$Sex.x = NULL
cleaned_CaH_Put_Cohort2@meta.data$Sex.y = NULL

cleaned_CaH_Put_Cohort2@meta.data$Condition = cleaned_CaH_Put_Cohort2@meta.data$Condition.x
cleaned_CaH_Put_Cohort2@meta.data$Condition.x = NULL
cleaned_CaH_Put_Cohort2@meta.data$Condition.y = NULL
cleaned_CaH_Put_Cohort2@meta.data


```




















```{r}

  library(ggplot2)
  library(Seurat)
  library(Matrix)
  library(rhdf5)
  library(dplyr)
  library(qs)
  # add any other libraries you need

library_list = list("ca_rxn1", "ca_rxn2", "put_rxn1", "put_rxn2")
RAW_COUNTS_BASENAME = "raw_feature_bc_matrix.h5"
FILTERED_COUNTS_BASENAME = "out_filtered.h5"
MOLECULE_INFO_BASENAME = "molecule_info.h5"
DONOR_ID_BASENAME = "vireo_final_out/donor_ids.tsv"
this_base_path = "/broad/macosko/kimkathl/Cohort2"

raw_dgc_list = list()
filtered_dgc_list = list()
mol_df_list = list()
donor_ids_list = list()
metadata_df_list = list()
seurat_object_list = list()


#helper function for cellbender loading
sum_duplicate_rownames_of_dgc_matrix=function(dgc){   
    # some umi_data has repeated rownames which causes problems down the road
    # we will sum the umi counts in these rows
    # collect rows with duplicate rownames as a dataframe
    n_occur = data.frame(table(rownames(dgc)))
    repeated_rownames = n_occur[n_occur$Freq > 1,]$Var1
    duplicated_dgc = as.data.frame(dgc[rownames(dgc) %in% repeated_rownames,])
    duplicated_dgc['symbol'] = sub("\\.\\d$", "", rownames(duplicated_dgc))

    # sum across numeric columns
    numeric_cols = setdiff(colnames(duplicated_dgc), "symbol")
    summed_duped_umi_rows = duplicated_dgc %>%
        group_by(symbol) %>%
        summarise_at(vars(numeric_cols), sum, na.rm = TRUE) %>%
        ungroup() %>%
        as.data.frame()
    # set rownames to be symbol column, then remove it
    rownames(summed_duped_umi_rows) = summed_duped_umi_rows$symbol
    summed_duped_umi_rows = summed_duped_umi_rows[, !(names(summed_duped_umi_rows) %in% c('symbol'))]

    # Hack
    # each round of the loop will remove ONE of each duplicate row 
    for(i in 1:max(n_occur$Freq)){
        dgc = dgc[!rownames(dgc) %in% repeated_rownames,]
    }

    # finally, replace the deduped rows with the summed rows via an rbind
    dgc = rbind(dgc, as(summed_duped_umi_rows, "sparseMatrix"))

    return(dgc)
}

  # Read your data
for (name in library_list) {
  
cat(name)
cat("\n")

raw_counts_path = file.path(this_base_path, name, RAW_COUNTS_BASENAME)
filtered_counts_path = file.path(this_base_path, name, paste0("pXDPsHS_v2_" , name, "_", FILTERED_COUNTS_BASENAME))
molecule_info_path = file.path(this_base_path, name, MOLECULE_INFO_BASENAME)
donor_id_path = file.path(this_base_path, name, DONOR_ID_BASENAME)

h5ls(filtered_counts_path)

file_path = filtered_counts_path # the path to the *_out_filtered.h5 file
    h5_data = h5read(file_path, "/matrix")
    counts = h5_data$data
    indices = h5_data$indices
    indptr = h5_data$indptr
    barcodes = h5_data$barcodes
    num_genes = length(h5_data$features$name)  # Number of genes
    num_cells = length(barcodes)  # Number of cells

    counts_matrix = sparseMatrix(i=indices + 1,  # +1 for 1-based indexing in R
                                    p=indptr,
                                    x=counts,
                                    dims=c(num_genes, num_cells))
    
 
    rownames(counts_matrix) = h5_data$features$name # set rownames to be genes
    colnames(counts_matrix) = barcodes # set colnames to be cell barcodes
    counts_matrix
    filtered_dgc = counts_matrix
    
       filtered_dgc_list[[name]] = filtered_dgc
       
       

print("READING RAW DGC .H5")
   raw_counts = Read10X_h5(raw_counts_path)
   raw_dgc_list[[name]] = raw_counts

   cd = data.frame(
       row.names=colnames(raw_counts),
       nUMI=colSums(raw_counts),
       log10_nUMI = log10(colSums(raw_counts) + 1),
       nGene=colSums(raw_counts > 0),
       is_in_filtered=colnames(raw_counts) %in% colnames(filtered_dgc_list[[name]])
   )

# load the molecule data such that you can map the barcodes to what came out of CellRanger
   # load umi_type to get whether or not the umi was intronic
   h5fetch = function(x){return(h5read(molecule_info_path, x))}
   mol_df = data.frame(
               barcode=h5fetch("barcodes")[h5fetch("barcode_idx")+1] %>% paste0("-1"),
               feature=h5fetch("features/name")[h5fetch("feature_idx")+1],
               umi=h5fetch("umi"),
               umi_type=h5fetch("umi_type"),
               count=h5fetch("count")
       )
      
   # only consider cells that are in the filtered umi data
   mol_df = mol_df[mol_df$barcode %in% colnames(filtered_dgc),]

mol_df_grouped = mol_df %>% group_by(barcode) %>%
       summarize(
           nUmi=n(),
           nRead=sum(count),
           pct_intronic=sum(umi_type==0)/nUmi)
   rownames(mol_df_grouped) = mol_df_grouped$barcode
   mol_df_list[[name]] = mol_df_grouped

donor_ids = read.table(donor_id_path, header=T)
  donor_ids = donor_ids[, c("cell", "donor_id", "prob_max")]
  donor_ids_list[[name]] = donor_ids

mol_df_rxn = mol_df_list[[name]] 
  metadata = merge(donor_ids, mol_df_rxn, by.x = "cell", by.y = "barcode", all = TRUE)
  rownames(metadata) <- metadata$cell
  
  metadata <- metadata[rownames(metadata) %in% colnames(filtered_dgc), ]
metadata

all(colnames(filtered_dgc) == rownames(metadata))
  metadata <- metadata[match(colnames(filtered_dgc), rownames(metadata)), ]
  all(colnames(filtered_dgc) == rownames(metadata))
  
metadata

metadata <- metadata[!is.na(metadata$donor_id), ]
metadata

file_path = file.path(this_base_path, name, paste0("pXDPsHS_v2_" , name, "_", FILTERED_COUNTS_BASENAME))
    h5_data = h5read(file_path, "/matrix")
    counts = h5_data$data
    indices = h5_data$indices
    indptr = h5_data$indptr
    barcodes = h5_data$barcodes
    num_genes = length(h5_data$features$name)  # Number of genes
    num_cells = length(barcodes)  # Number of cells

    counts_matrix = sparseMatrix(i=indices + 1,  # add 1 for 1-based indexing in R
                                    p=indptr,
                                    x=counts,
                                    dims=c(num_genes, num_cells))
    

    rownames(counts_matrix) = h5_data$features$name # set rownames to be genes
    colnames(counts_matrix) = barcodes # set colnames to be cell barcodes

# now we want to look only at cells that are in BOTh the CellRanger and CellBender objects
    # use filtered_dgc from earlier in the loop
    cellranger_filtered = filtered_dgc[, colnames(filtered_dgc) %in% all_of(colnames(counts_matrix))]
    cellranger_filtered

counts_matrix_filtered = counts_matrix[, colnames(counts_matrix) %in% all_of(colnames(cellranger_filtered))]
    counts_matrix_filtered
    print(dim(counts_matrix_filtered))
   print("1")
    # for whatever reason, there can be duplicated rownames (i.e. genes) in the cellbender output, causing errors later on. Sum these together.
    counts_matrix_filtered = sum_duplicate_rownames_of_dgc_matrix(counts_matrix_filtered) 
    print(dim(counts_matrix_filtered))
   print("2")
    counts_matrix_filtered = counts_matrix_filtered[, match(colnames(cellranger_filtered), colnames(counts_matrix_filtered))]
    print(dim(counts_matrix_filtered))
   print("3")

# create your metadata here....
metadata = metadata[rownames(metadata) %in% colnames(counts_matrix_filtered), ]
metadata

metadata_df_list[[name]] = metadata 
    # add metadata argument to CreateSeuratObject

# Compare the first few entries directly
head(colnames(counts_matrix_filtered))
head(metadata$cell)

counts_matrix_filtered
metadata

extra_barcodes <- setdiff(colnames(counts_matrix_filtered), metadata$cell)
length(extra_barcodes)  # Number of extra barcodes
extra_barcodes  # List of extra barcodes
counts_matrix_filtered <- counts_matrix_filtered[, !(colnames(counts_matrix_filtered) %in% extra_barcodes)]

# Sort both and compare
all(sort(colnames(counts_matrix_filtered)) == sort(metadata$cell))
# Check types
str(colnames(counts_matrix_filtered))
str(metadata$cell)
# Convert both to character if needed
metadata$cell <- as.character(metadata$cell)
colnames(counts_matrix_filtered) <- as.character(colnames(counts_matrix_filtered))
# Check unique values
unique(colnames(counts_matrix_filtered))[1:10]  # First 10 in counts_matrix
unique(metadata$cell)[1:10]  # First 10 in metadata
metadata_ordered <- metadata[match(colnames(counts_matrix_filtered), metadata$cell), ]

cellbender_sobj = CreateSeuratObject(counts_matrix_filtered, meta.data = metadata)

seurat_object_list[[name]] = cellbender_sobj
}
seurat_object_list
qsave(seurat_object_list, "Cohort2.qs")
```




```{r}
CaH_rxn1_cellbender = seurat_object_list[["ca_rxn1"]]
CaH_rxn2_cellbender = seurat_object_list[["ca_rxn2"]]
Put_rxn1_cellbender = seurat_object_list[["put_rxn1"]]
Put_rxn2_cellbender = seurat_object_list[["put_rxn2"]]
```

```{r}
CaH_rxn1_cellbender@meta.data
CaH_rxn2_cellbender@meta.data
Put_rxn1_cellbender@meta.data
Put_rxn2_cellbender@meta.data
```

```{r}
#qsave(CaH_rxn1_cellbender, "Cohort2/sobjs/CaH_rxn1_cellbender.qs")
#qsave(CaH_rxn2_cellbender, "Cohort2/sobjs/CaH_rxn2_cellbender.qs")
#qsave(Put_rxn1_cellbender, "Cohort2/sobjs/Put_rxn1_cellbender.qs")
#qsave(Put_rxn2_cellbender, "Cohort2/sobjs/Put_rxn2_cellbender.qs")
```

```{r}
table(CaH_rxn1_cellbender$donor_id)
table(CaH_rxn2_cellbender$donor_id)
table(Put_rxn1_cellbender$donor_id)
table(Put_rxn2_cellbender$donor_id)
```

```{r}
CaH_rxn1 = subset(CaH_rxn1_cellbender, subset = donor_id != "doublet" & donor_id != "unassigned")
CaH_rxn1

CaH_rxn2 = subset(CaH_rxn2_cellbender, subset = donor_id != "doublet" & donor_id != "unassigned")
CaH_rxn2

Put_rxn1 = subset(Put_rxn1_cellbender, subset = donor_id != "doublet" & donor_id != "unassigned")
Put_rxn1

Put_rxn2 = subset(Put_rxn2_cellbender, subset = donor_id != "doublet" & donor_id != "unassigned")
Put_rxn2
```
#calcualte pct mito
```{r}
# Step 1: Identify mitochondrial genes (assuming mitochondrial genes start with "MT-" in gene names)
mito_genes <- grep(pattern = "^MT-", x = rownames(Put_rxn2), value = TRUE)
length(mito_genes)
# Step 2: Access counts from RNA assay using GetAssayData()
mito_counts <- Matrix::colSums(GetAssayData(Put_rxn2, assay = "RNA", slot = "counts")[mito_genes, ])
total_counts <- Matrix::colSums(GetAssayData(Put_rxn2, assay = "RNA", slot = "counts"))

# Step 3: Calculate percentage of mitochondrial counts
pct_mito <- (mito_counts / total_counts) * 100

# Step 4: Add this as a metadata column in Seurat object
Put_rxn2$pct_mito <- pct_mito

# Step 5: View the result
head(Put_rxn2$pct_mito)

Put_rxn2@meta.data
```

```{r}
sobj_list = list(CaH_rxn1,CaH_rxn2)

sobj_list
```

#merge caudate libraries 
```{r}
sobj_list = list("CaH_rxn1" = CaH_rxn1,"CaH_rxn2" =CaH_rxn2 )

for(name in names(sobj_list)){
     sobj =sobj_list[[name]]
     sobj$library = name
     sobj = RenameCells(sobj, new.name = paste(sobj$library, colnames(sobj), sep="__"))
     sobj_list[[name]] = sobj
     
}

sobj_list
```
```{r}
counts_list <- lapply(sobj_list, function(sobj) GetAssayData(sobj, slot = "counts"))


merged_counts <- do.call(cbind, counts_list)


metadata_list <- lapply(sobj_list, function(sobj) sobj@meta.data)
metadata_list
```

```{r}
merged_metadata <- do.call(rbind, metadata_list)
merged_metadata
rownames(merged_metadata) <- unlist(lapply(metadata_list, rownames))
merged_metadata
```

```{r}
stopifnot(all(rownames(merged_metadata) == colnames(merged_counts)))

# Create Seurat object
CaH_Cohort2 <- CreateSeuratObject(counts = merged_counts, meta.data = merged_metadata)
CaH_Cohort2
```

```{r}
CaH_Cohort2@meta.data
```

#merge putamen libraries 
```{r}
sobj_list = list("Put_rxn1" = Put_rxn1,"Put_rxn2" =Put_rxn2 )

for(name in names(sobj_list)){
     sobj =sobj_list[[name]]
     sobj$library = name
     sobj = RenameCells(sobj, new.name = paste(sobj$library, colnames(sobj), sep="__"))
     sobj_list[[name]] = sobj
     
}

sobj_list

counts_list <- lapply(sobj_list, function(sobj) GetAssayData(sobj, slot = "counts"))


merged_counts <- do.call(cbind, counts_list)


metadata_list <- lapply(sobj_list, function(sobj) sobj@meta.data)
metadata_list
```

```{r}
merged_metadata <- do.call(rbind, metadata_list)
merged_metadata
rownames(merged_metadata) <- unlist(lapply(metadata_list, rownames))
merged_metadata
```

```{r}
stopifnot(all(rownames(merged_metadata) == colnames(merged_counts)))

# Create Seurat object
Put_Cohort2 <- CreateSeuratObject(counts = merged_counts, meta.data = merged_metadata)
Put_Cohort2

Put_Cohort2@meta.data
```

```{r}
unique(CaH_Cohort2@meta.data$donor_id)
```
```{r}
CaH_Cohort2@meta.data$sex <- ifelse(grepl("CF$", CaH_Cohort2@meta.data$donor_id), "F", "M")
table(CaH_Cohort2@meta.data$sex, CaH_Cohort2$donor_id)

CaH_Cohort2@meta.data$Condition <- ifelse(grepl("CF$|CM$", CaH_Cohort2@meta.data$donor_id), "Control", "XDP")

table(CaH_Cohort2@meta.data$Condition, CaH_Cohort2$donor_id)
```

```{r}
Put_Cohort2@meta.data$sex <- ifelse(grepl("CF$", Put_Cohort2@meta.data$donor_id), "F", "M")
table(Put_Cohort2@meta.data$sex, Put_Cohort2$donor_id)

Put_Cohort2@meta.data$Condition <- ifelse(grepl("CF$|CM$", Put_Cohort2@meta.data$donor_id), "Control", "XDP")

table(Put_Cohort2@meta.data$Condition, Put_Cohort2$donor_id)
```

#redo filtering!
#nvm its the same..
```{r}
CaH_Cohort2
Put_Cohort2

filtered_CaH_Cohort2
filtered_Put_Cohort2
```

```{r}
CaH_Cohort2
filtered_CaH_Cohort2 <- subset(CaH_Cohort2, subset = pct_mito < 10)
filtered_CaH_Cohort2
filtered_CaH_Cohort2 = subset(filtered_CaH_Cohort2, subset = pct_intronic >= 0.25)
filtered_CaH_Cohort2 = subset(filtered_CaH_Cohort2, subset = nUmi > 500)
filtered_CaH_Cohort2
```

```{r}
Put_Cohort2
filtered_Put_Cohort2 <- subset(Put_Cohort2, subset = pct_mito < 10)
filtered_Put_Cohort2
filtered_Put_Cohort2 = subset(filtered_Put_Cohort2, subset = pct_intronic >= 0.25)
filtered_Put_Cohort2 = subset(filtered_Put_Cohort2, subset = nUmi > 500)
filtered_Put_Cohort2
```

```{r}
# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(filtered_CaH_Cohort2, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
filtered_CaH_Cohort2 = (filtered_CaH_Cohort2
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_CaH_Cohort2$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_CaH_Cohort2$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(filtered_CaH_Cohort2, group.by = "donor_id") 
DimPlot(filtered_CaH_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(filtered_CaH_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(filtered_CaH_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(filtered_CaH_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(filtered_CaH_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(filtered_CaH_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(filtered_CaH_Cohort2, group.by = "RNA_snn_res.0.8", label=T)

```

```{r}
 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN", "GAPDH","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = filtered_CaH_Cohort2, features = marker_genes, group.by = "RNA_snn_res.0.3")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```

```{r}
Idents(filtered_CaH_Cohort2) = filtered_CaH_Cohort2$RNA_snn_res.0.3

classes = c("oligo", "astrocyte", "oligo", "neuron", "oligo", "microglia", "astrocyte", "neuron", "opc", "oligo", "neuron", "oligo", "oligo", "neuron", "neuron", "oligo", "oligo", "oligo", "neuron", "neuron", "oligo", "ependymal", "endothelial", "neuron", "neuron", "astrocyte", "neuron", "neuron", "immune", "neuron")


filtered_CaH_Cohort2= assignCellClasses(filtered_CaH_Cohort2, classes=classes, cluster_col="RNA_snn_res.0.3", class_col = "cell_class")

DimPlot(filtered_CaH_Cohort2, group.by = "cell_class" ,label = T, raster = FALSE)

```

```{r}
FeaturePlot(filtered_CaH_Cohort2, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(filtered_CaH_Cohort2, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(filtered_CaH_Cohort2, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(filtered_CaH_Cohort2, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(filtered_CaH_Cohort2, features = c("OLIG1", "MOG", "MOBP"), raster = F)
FeaturePlot(filtered_CaH_Cohort2, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(filtered_CaH_Cohort2, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(filtered_CaH_Cohort2, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(filtered_CaH_Cohort2, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
```



```{r}
# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(filtered_Put_Cohort2, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
filtered_Put_Cohort2 = (filtered_Put_Cohort2
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_CaH_Cohort2$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_CaH_Cohort2$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(filtered_Put_Cohort2, group.by = "donor_id") 
DimPlot(filtered_Put_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(filtered_Put_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(filtered_Put_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(filtered_Put_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(filtered_Put_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(filtered_Put_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(filtered_Put_Cohort2, group.by = "RNA_snn_res.0.8", label=T)

```

```{r}
 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN", "GAPDH","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = filtered_Put_Cohort2, features = marker_genes, group.by = "RNA_snn_res.0.3")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```

```{r}
Idents(filtered_Put_Cohort2) = filtered_Put_Cohort2$RNA_snn_res.0.3

classes = c("oligo", "oligo", "oligo", "neuron", "neuron", "astrocyte", "oligo", "oligo", "oligo", "microglia", "oligo", "oligo", "opc", "neuron", "neuron", "astrocyte", "oligo", "oligo", "neuron", "endothelial", "neuron", "neuron", "neuron", "neuron", "immune")


filtered_Put_Cohort2= assignCellClasses(filtered_Put_Cohort2, classes=classes, cluster_col="RNA_snn_res.0.3", class_col = "cell_class")

DimPlot(filtered_Put_Cohort2, group.by = "cell_class" ,label = T, raster = FALSE)

```

```{r}
FeaturePlot(filtered_Put_Cohort2, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(filtered_Put_Cohort2, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(filtered_Put_Cohort2, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(filtered_Put_Cohort2, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(filtered_Put_Cohort2, features = c("OLIG1", "MOG", "MOBP"), raster = F)
FeaturePlot(filtered_Put_Cohort2, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(filtered_Put_Cohort2, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(filtered_Put_Cohort2, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(filtered_Put_Cohort2, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
```


#QC

```{r}
#histogram for pct_intronic
median_pct_intronic = median(Put_Cohort2$pct_intronic)
title = paste0("Put_Cohort2: pct intronic, " , "Median: ", sprintf("%0.3f", median_pct_intronic))
pct_intronic_hist = hist(sobj$pct_intronic, main = title, xlab = "pct intronic", col="grey", breaks = 100)

median_pct_intronic = median(filtered_Put_Cohort2$pct_intronic)
title = paste0("filtered_Put_Cohort2: pct intronic, " , "Median: ", sprintf("%0.3f", median_pct_intronic))
pct_intronic_hist = hist(sobj$pct_intronic, main = title, xlab = "pct intronic", col="grey", breaks = 100)
```

```{r}
#histogram for nUMI --> take log10 of nUMI tonormalize
log_nUMI = log10(Put_Cohort2$nUmi)
median_log_nUMI = median(log_nUMI)
title = paste0("Put_Cohort2: log10(nUMI)", "Median: ", sprintf("%0.3f", median_log_nUMI))
log_nUMI_hist = hist(log_nUMI, main= title, xlab = "log10(nUMI)", col="grey", breaks = 100)
```

```{r}
saturation <- Put_Cohort2$nRead/Put_Cohort2$nUmi
median_umi_per_read = median(saturation)
title = paste0("Put_Cohort2: nReads/nUMI, " , "Median: ", sprintf("%0.3f", median_umi_per_read))  
saturation_hist = hist(saturation, main= title, xlab = "nRead/nUMI", col="grey", breaks = 100)

saturation <- filtered_Put_Cohort2$nRead/filtered_Put_Cohort2$nUmi
median_umi_per_read = median(saturation)
title = paste0("filtered_Put_Cohort2: nReads/nUMI, " , "Median: ", sprintf("%0.3f", median_umi_per_read))  
saturation_hist = hist(saturation, main= title, xlab = "nRead/nUMI", col="grey", breaks = 100)

```

```{r}
median_pct_mito = median(Put_Cohort2$pct_mito)
title = paste0("Put_Cohort2: pct mito, " , "Median: ", sprintf("%0.3f", median_pct_mito))
pct_intronic_hist = hist(sobj$pct_mito, main = title, xlab = "pct mito", col="grey", breaks = 100)

median_pct_mito = median(filtered_Put_Cohort2$pct_mito)
title = paste0("filtered_Put_Cohort2: pct mito, " , "Median: ", sprintf("%0.3f", median_pct_mito))
pct_intronic_hist = hist(sobj$pct_mito, main = title, xlab = "pct mito", col="grey", breaks = 100)
```
#QC by library and donor
```{r}
#histogram for pct_intronic
summary_data <- filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(
    median_pct_intronic = median(pct_intronic, na.rm = TRUE), 
    log_nUMI = median(log10(nUmi), na.rm = TRUE),  # Compute log inside summarize
    saturation = median(nRead / nUmi, na.rm = TRUE),  # Ensure division inside summarize
    median_pct_mito = median(pct_mito, na.rm = TRUE)
  )

summary_data

```

```{r}
ggplot(filtered_Put_Cohort2@meta.data, aes(x = filtered_Put_Cohort2$pct_intronic)) + facet_wrap(~ donor_id) + geom_histogram(bins = 100) 
```

```{r}
cah_meta = filtered_CaH_Cohort2@meta.data
put_meta = filtered_Put_Cohort2@meta.data 
```


```{r}
medians <- cah_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
cah_meta$donor_id <- factor(cah_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(cah_meta, aes(x = pct_intronic, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_pct_intronic, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Caudate pct intronic by donor") +ylab("Donor")



# Compute median for each donor
medians <- put_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
put_meta$donor_id <- factor(put_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(put_meta, aes(x = pct_intronic, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_pct_intronic, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Putamen pct intronic by donor") +ylab("Donor")


medians <- cah_meta %>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_log_nUMI = median(log10(nUmi), na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
cah_meta$donor_id <- factor(cah_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(cah_meta, aes(x = log10(nUmi), fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_log_nUMI, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Caudate log(UMI) by donor") +ylab("Donor")



# Compute median for each donor
medians <- put_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_log_nUMI = median(log10(nUmi), na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
put_meta$donor_id <- factor(put_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(put_meta, aes(x = log10(nUmi), fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_log_nUMI, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Putamen log(UMI) by donor") +ylab("Donor")


cah_meta$saturation <- cah_meta$nRead/cah_meta$nUmi
put_meta$saturation <- put_meta$nRead/put_meta$nUmi

medians <- cah_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_saturation = median(saturation, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
cah_meta$donor_id <- factor(cah_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)


# Plot histogram with facet per donor and median as text
ggplot(cah_meta, aes(x = saturation, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_saturation, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Caudate saturation by donor") +ylab("Donor")



# Compute median for each donor
medians <- put_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_saturation = median(saturation, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
put_meta$donor_id <- factor(put_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text

ggplot(put_meta, aes(x = saturation, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_saturation, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Putamen saturation by donor") +ylab("Donor")


medians <- cah_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
cah_meta$donor_id <- factor(cah_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(cah_meta, aes(x = pct_mito, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_pct_mito, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Caudate pct mito by donor") +ylab("Donor")



# Compute median for each donor
medians <- put_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
put_meta$donor_id <- factor(put_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(put_meta, aes(x = pct_mito, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_pct_mito, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Putamen pct mito by donor") +ylab("Donor")

```



```{r}
median_pct_intronic = median(filtered_Put_Cohort2$pct_intronic)
title = paste0("filtered_Put_Cohort2: pct intronic, " , "Median: ", sprintf("%0.3f", median_pct_intronic))
pct_intronic_hist = hist(filtered_Put_Cohort2$pct_intronic, main = title, xlab = "pct intronic", col="grey", breaks = 100)
```

```{r}
#histogram for nUMI --> take log10 of nUMI tonormalize
log_nUMI = log10(Put_Cohort2$nUmi)
title = paste0("Put_Cohort2: log10(nUMI)")
log_nUMI_hist = hist(log_nUMI, main= title, xlab = "log10(nUMI)", col="grey", breaks = 100)
```

```{r}
saturation <- filtered_Put_Cohort2$nRead/filtered_Put_Cohort2$nUmi
median_umi_per_read = median(saturation)
title = paste0("filtered_Put_Cohort2: nReads/nUMI, " , "Median: ", sprintf("%0.3f", median_umi_per_read))  
saturation_hist = hist(saturation, main= title, xlab = "nRead/nUMI", col="grey", breaks = 100)

```

```{r}
median_pct_mito = median(filtered_Put_Cohort2$pct_mito)
title = paste0("filtered_Put_Cohort2: pct mito, " , "Median: ", sprintf("%0.3f", median_pct_mito))
pct_intronic_hist = hist(sobj$pct_mito, main = title, xlab = "pct mito", col="grey", breaks = 100)
```













```{r}
CaH_Cohort2
Put_Cohort2
```


```{r}
table(filtered_CaH_Cohort2$donor_id)
table(filtered_Put_Cohort2$donor_id)
```
```{r}
filtered_CaH_Cohort2@meta.data
```


```{r}
total_donors_cells_caudate <- as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id))
colnames(total_donors_cells_caudate) <- c("donor_id", "cell_count")

# Add condition information
total_donors_cells_caudate$Condition <- filtered_CaH_Cohort2@meta.data$Condition[match(total_donors_cells_caudate$donor_id, filtered_CaH_Cohort2@meta.data$donor_id)]

total_donors_cells_caudate$donor_id <- factor(total_donors_cells_caudate$donor_id, levels = total_donors_cells_caudate$donor_id[order(total_donors_cells_caudate$cell_count, decreasing = TRUE)])


ggplot(total_donors_cells_caudate, aes(x = donor_id, y = cell_count, fill = Condition)) + 
  geom_bar(stat = "identity") + 
  geom_text(aes(label = cell_count), vjust = -0.3, color = "black", size = 3) + 
  xlab("Donors from Caudate Village") + 
  ylab("Number of Cells") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
scale_fill_manual(values = c("XDP" = "#F8766D", "Control" = "#00BFC4"))

```

```{r}
total_donors_cells_caudate <- as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id,filtered_CaH_Cohort2@meta.data$library ))
colnames(total_donors_cells_caudate) <- c("donor_id", "library", "cell_count")
total_donors_cells_caudate
```


```{r}
# Add condition information
total_donors_cells_caudate$Condition <- filtered_CaH_Cohort2@meta.data$Condition[match(total_donors_cells_caudate$donor_id, filtered_CaH_Cohort2@meta.data$donor_id)]
total_donors_cells_caudate
```



```{r}
caudate_cell = as.data.frame(table(filtered_CaH_Cohort2@meta.data$Condition))
caudate_cell$Condition = caudate_cell$Var1
caudate_cell$Var1 = NULL
caudate_cell$Village = "Caudate"
putamen_cell = as.data.frame(table(filtered_Put_Cohort2@meta.data$Condition))
putamen_cell$Condition = putamen_cell$Var1
putamen_cell$Var1 = NULL
putamen_cell$Village = "Putamen"
cells_villages = rbind(caudate_cell, putamen_cell)

total_freq_per_village <- aggregate(Freq ~ Village, data = cells_villages, FUN = sum)
total_freq_per_village

ggplot(cells_villages, aes(x = Village, y = Freq, fill = Condition)) +
  geom_bar(stat = "identity") +  
  geom_text(aes(label = Freq), position = position_stack(vjust = 0.5), size = 3, color = "white") +  
  geom_text(data = total_freq_per_village, aes(x = Village, y = Freq, label = Freq), vjust = -0.5, size = 4, inherit.aes = FALSE) +  
  labs(title = "Cell Counts by Region and Condition", x = "Region", y = "Cell Counts", fill = "Condition") +
  theme_minimal()


```

```{r}
caudate_cell = as.data.frame(table(filtered_CaH_Cohort2@meta.data$library))
caudate_cell$library = caudate_cell$Var1
caudate_cell$Var1 = NULL
caudate_cell$Village = "Caudate"
putamen_cell = as.data.frame(table(filtered_Put_Cohort2@meta.data$library))
putamen_cell$library = putamen_cell$Var1
putamen_cell$Var1 = NULL
putamen_cell$Village = "Putamen"
cells_villages = rbind(caudate_cell, putamen_cell)

total_freq_per_village <- aggregate(Freq ~ Village, data = cells_villages, FUN = sum)
total_freq_per_village

ggplot(cells_villages, aes(x = Village, y = Freq, fill = library)) +
  geom_bar(stat = "identity") +  
  geom_text(aes(label = Freq), position = position_stack(vjust = 0.5), size = 3, color = "white") +  
  geom_text(data = total_freq_per_village, aes(x = Village, y = Freq, label = Freq), vjust = -0.5, size = 4, inherit.aes = FALSE) +  
  labs(title = "Cell Counts by Region and Library", x = "Region", y = "Cell Counts", fill = "Condition") +
  theme_minimal()
```

```{r}
unique(filtered_CaH_Cohort2@meta.data$donor_id)
```


```{r}
per_donor = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id))
per_donor

cells_per_donor_per_rxn = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id, filtered_CaH_Cohort2@meta.data$library))
cells_per_donor_per_rxn = merge(cells_per_donor_per_rxn, per_donor, by = "Var1" )
cells_per_donor_per_rxn

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  


cells_per_donor_per_rxn$Var1 = factor(cells_per_donor_per_rxn$Var1, levels = x_order)

ggplot(cells_per_donor_per_rxn, aes(x = Var1, y = Freq.x, fill = Var2)) +
  geom_bar(stat = "identity") + xlab("Donors from Caudate Village") +  geom_text(data = cells_per_donor_per_rxn, aes(x = Var1, y = Freq.y, label = Freq.y), vjust = -0.2, size = 3, nudge_y = 0.5) + ylab("Number of Cells") +
 labs(fill = "Library")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r}
per_donor = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id))
per_donor

cells_per_donor_per_rxn = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id, filtered_Put_Cohort2@meta.data$library))
cells_per_donor_per_rxn = merge(cells_per_donor_per_rxn, per_donor, by = "Var1" )
cells_per_donor_per_rxn

x_order = c("SCF_23_072CF", "SCF_22_049CCF","SCF_23_079CF", "SCF_22_063CM","SCF_22_067CM",  "SCF_23_075CM",  "SCF_23_074CF", "SCF_22_065CM",   
"SCF_21_032","SCF_21_038",  "SCF_21_041","SCF_23_070_MMC", "SCF_21_034", "SCF_21_029","SCF_21_027")  


cells_per_donor_per_rxn$Var1 = factor(cells_per_donor_per_rxn$Var1, levels = x_order)

ggplot(cells_per_donor_per_rxn, aes(x = Var1, y = Freq.x, fill = Var2)) +
  geom_bar(stat = "identity") + xlab("Donors from Putamen Village") +  geom_text(data = cells_per_donor_per_rxn, aes(x = Var1, y = Freq.y, label = Freq.y), vjust = -0.2, size = 3, nudge_y = 0.5) + ylab("Number of Cells") +
 labs(fill = "Library")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r}
DimPlot(filtered_CaH_Cohort2, group.by = "Condition")
DimPlot(filtered_CaH_Cohort2, group.by = "library")
DimPlot(filtered_CaH_Cohort2, group.by = "donor_id")

DimPlot(filtered_Put_Cohort2, group.by = "Condition")
DimPlot(filtered_Put_Cohort2, group.by = "library")
DimPlot(filtered_Put_Cohort2, group.by = "donor_id")

FeaturePlot(filtered_CaH_Cohort2, features = c("nUmi", "pct_mito"))
```












```{r}
cells_per_donor_per_rxn = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id, filtered_CaH_Cohort2@meta.data$library))
cells_per_donor_per_rxn

ggplot(cells_per_donor_per_rxn, aes(x=Var1, y = Freq, fill=Var2)) + geom_point(shape = 21, size = 3)+ xlab("Donors") + ylab("Number of Cells") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylim(0, 2500)
```


```{r}
median_genes_per_donor <-filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_genes = median(nFeature_RNA, na.rm = TRUE))

median_genes_per_donor

median_genes_per_donor <-filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id, library) %>%
  summarize(median_genes = median(nFeature_RNA, na.rm = TRUE))

median_genes_per_donor

ggplot(median_genes_per_donor, aes(x=donor_id, y = median_genes, fill=library)) + geom_point(shape = 21, size = 3)+ xlab("Donors") + ylab("Median number of genes") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylim(0,4000) +ggtitle("Putamen: Median genes per library")
```

```{r}
median_umis_per_donor <-filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id, library) %>%
  summarize(mUMI = median(nUmi, na.rm = TRUE))

median_umis_per_donor

ggplot(median_umis_per_donor, aes(x=donor_id, y = mUMI, fill=library)) + geom_point(shape = 21, size = 3)+ xlab("Donors") + ylab("Median UMIs") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) +ggtitle("Putamen: Median UMIs")
```



```{r}
total_donors_cells_caudate_per_rxn = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id, filtered_CaH_Cohort2$library))

#install.packages("forcats")
library(forcats)

# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_caudate_per_rxn %>%
  group_by(Var1) %>%
  summarize(total_Freq = sum(Freq))

# Reorder Var1 based on total_Freq
total_donors_cells_caudate_per_rxn <- total_donors_cells_caudate_per_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_Freq[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>%
  mutate(Var1 = fct_reorder(Var1, -total_Freq))

# Create the stacked bar plot
ggplot(total_donors_cells_caudate_per_rxn, aes(x = Var1, y = Freq, fill = Var2)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_Freq, label = total_Freq), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Caudate Village") +
  ylab("Number of Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```


```{r}
total_donors_cells_putamen_per_rxn = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id, filtered_Put_Cohort2$library))
total_donors_cells_putamen_per_rxn

#install.packages("forcats")
#library(forcats)

# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_putamen_per_rxn %>%
  group_by(Var1) %>%
  summarize(total_Freq = sum(Freq))

# Reorder Var1 based on total_Freq
total_donors_cells_putamen_per_rxn <- total_donors_cells_putamen_per_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_Freq[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>%
  mutate(Var1 = fct_reorder(Var1, -total_Freq))

# Create the stacked bar plot
ggplot(total_donors_cells_putamen_per_rxn, aes(x = Var1, y = Freq, fill = Var2)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_Freq, label = total_Freq), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Putamen Village") +
  ylab("Number of Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```



#calculate how many cells are being filtered out from EACH donor
```{r}
total_donors_cells_caudate_by_rxn = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id, filtered_CaH_Cohort2@meta.data$library))
total_donors_cells_caudate_by_rxn

total_donors_cells_putamen_by_rxn = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id, filtered_Put_Cohort2@meta.data$library))
total_donors_cells_putamen_by_rxn

#prefilter
prefilter_caudate_by_rxn = as.data.frame(table(CaH_Cohort2@meta.data$donor_id, CaH_Cohort2@meta.data$library))
prefilter_caudate_by_rxn = subset(prefilter_caudate_by_rxn, subset = Var1 != "SCF_21_036B")
prefilter_caudate_by_rxn

prefilter_putamen_by_rxn = as.data.frame(table(Put_Cohort2@meta.data$donor_id, Put_Cohort2@meta.data$library))
prefilter_putamen_by_rxn
```

```{r}
total_donors_cells_caudate_by_rxn$filtered_cells = prefilter_caudate_by_rxn$Freq - total_donors_cells_caudate_by_rxn$Freq
total_donors_cells_caudate_by_rxn

total_donors_cells_putamen_by_rxn$filtered_cells =  prefilter_putamen_by_rxn$Freq - total_donors_cells_putamen_by_rxn$Freq
total_donors_cells_putamen_by_rxn

```
```{r}
total_donors_cells_caudate_by_rxn$proportion = total_donors_cells_caudate_by_rxn$filtered_cells/total_donors_cells_caudate_by_rxn$Freq
total_donors_cells_caudate_by_rxn$proportion <- round(total_donors_cells_caudate_by_rxn$proportion, 2)
total_donors_cells_caudate_by_rxn
```

```{r}
total_donors_cells_caudate_by_rxn
summary_data
```


```{r}
# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_caudate_by_rxn %>%
  group_by(Var1) %>%
  summarize(total_filtered_cells = sum(filtered_cells), total_cells = sum(Freq))

summary_data$total_proportion = summary_data$total_filtered_cells/summary_data$total_cells
summary_data$total_proportion <- round(summary_data$total_proportion, 2)

summary_data

# Reorder Var1 based on total_filtered_cells
total_donors_cells_caudate_by_rxn <- total_donors_cells_caudate_by_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_proportion[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>% mutate(Var1 = fct_reorder(Var1, -total_proportion))
summary_data

# Create the stacked bar plot
ggplot(summary_data, aes(x = Var1, y = total_proportion)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_proportion, label = total_proportion), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Caudate Village") +
  ylab("Proportion of Filtered Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```
```{r}
# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_caudate_by_rxn %>%
  group_by(Var1) %>%
  summarize(total_filtered_cells = sum(filtered_cells))

summary_data

# Reorder Var1 based on total_filtered_cells
total_donors_cells_caudate_by_rxn <- total_donors_cells_caudate_by_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_filtered_cells[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>% mutate(Var1 = fct_reorder(Var1, -total_filtered_cells))
summary_data

# Create the stacked bar plot
ggplot(total_donors_cells_caudate_by_rxn, aes(x = Var1, y = filtered_cells, fill = Var2)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_filtered_cells, label = total_filtered_cells), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Caudate Village") +
  ylab("Number of Filtered Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```


```{r}
# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_putamen_by_rxn %>%
  group_by(Var1) %>%
  summarize(total_filtered_cells = sum(filtered_cells))


# Reorder Var1 based on total_filtered_cells
total_donors_cells_putamen_by_rxn <- total_donors_cells_putamen_by_rxn %>%
mutate(Var1 = fct_reorder(Var1, -summary_data$total_filtered_cells[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>% mutate(Var1 = fct_reorder(Var1, -total_filtered_cells))
summary_data

# Create the stacked bar plot
ggplot(total_donors_cells_putamen_by_rxn, aes(x = Var1, y = filtered_cells, fill = Var2)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_filtered_cells, label = total_filtered_cells), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Putamen Village") +
  ylab("Number of Filtered Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```


```{r}
total_donors_cells_putamen_by_rxn$proportion = total_donors_cells_putamen_by_rxn$filtered_cells/total_donors_cells_putamen_by_rxn$Freq
total_donors_cells_putamen_by_rxn$proportion <- round(total_donors_cells_putamen_by_rxn$proportion, 2)
total_donors_cells_putamen_by_rxn

# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_putamen_by_rxn %>%
  group_by(Var1) %>%
  summarize(total_filtered_cells = sum(filtered_cells), total_cells = sum(Freq), total_proportion = sum(proportion))

summary_data


# Reorder Var1 based on total_filtered_cells
total_donors_cells_putamen_by_rxn <- total_donors_cells_putamen_by_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_proportion[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>% mutate(Var1 = fct_reorder(Var1, -total_proportion))
summary_data

# Create the stacked bar plot
ggplot(total_donors_cells_putamen_by_rxn, aes(x = Var1, y = proportion, fill = Var2)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_proportion, label = total_proportion), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Putamen Village") +
  ylab("Proportion of Filtered Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r}
# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_putamen_by_rxn %>%
  group_by(Var1) %>%
  summarize(total_filtered_cells = sum(filtered_cells), total_cells = sum(Freq))

summary_data$total_proportion = summary_data$total_filtered_cells/summary_data$total_cells
summary_data$total_proportion <- round(summary_data$total_proportion, 2)

summary_data

# Reorder Var1 based on total_filtered_cells
total_donors_cells_putamen_by_rxn <- total_donors_cells_putamen_by_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_proportion[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>% mutate(Var1 = fct_reorder(Var1, -total_proportion))
summary_data

# Create the stacked bar plot
ggplot(summary_data, aes(x = Var1, y = total_proportion)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_proportion, label = total_proportion), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Putamen Village") +
  ylab("Proportion of Filtered Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```


```{r}
cells_per_donor = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(filtered_CaH_Cohort2@meta.data$cell_class, filtered_CaH_Cohort2@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df


cell_class_df

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  
            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = x_order)

ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors from Caudate Village") + ylab("Number of cells by cell type") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors from Caudate Village") + ylab("Cell Type Proportion") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  
#+ facet_wrap(~ Var1)
```
```{r}
cells_per_donor = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(filtered_Put_Cohort2@meta.data$cell_class, filtered_Put_Cohort2@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df


cell_class_df

x_order = c("SCF_23_072CF", "SCF_22_049CCF","SCF_23_079CF", "SCF_22_063CM","SCF_22_065CM", "SCF_22_067CM",  "SCF_23_075CM",  "SCF_23_074CF",   
"SCF_21_032","SCF_21_038",  "SCF_21_041","SCF_23_070_MMC", "SCF_21_034", "SCF_21_029","SCF_21_027") 
            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = x_order)

ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors from Putamen Village") + ylab("Number of cells by cell type") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors from Putamen Village") + ylab("Cell Type Proportion") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  
#+ facet_wrap(~ Var1)
```
```{r}
qsave(CaH_Cohort2, "Cohort2/sobjs/CaH_Cohort2_unfiltered.qs")
qsave(Put_Cohort2, "Cohort2/sobjs/Put_Cohort2_unfiltered.qs")

qsave(filtered_CaH_Cohort2, "Cohort2/sobjs/filtered_CaH_Cohort2.qs")
qsave(filtered_Put_Cohort2, "Cohort2/sobjs/filtered_Put_Cohort2.qs")
```



```{r}
median(filtered_CaH_Cohort2$pct_mito)
```


```{r}
#totalcells
cells_per_library = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id))
names(cells_per_library)[names(cells_per_library) == "Var1"] ="library"
names(cells_per_library)[names(cells_per_library) == "Freq"] ="total_cells"
cells_per_library  

#median/mean nUMI
median_nUmi = filtered_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi

# mean_nUmi = filtered_CaH_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_UMI = mean(nUmi, na.rm = TRUE))
# mean_nUmi

#median/mean nGenes
median_nGenes = filtered_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_nGenes = median(nFeature_RNA, na.rm = TRUE))
median_nGenes

# mean_nGenes = filtered_CaH_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_nGenes = mean(nFeature_RNA, na.rm = TRUE))
# mean_nGenes

#median/mean reads/numi
median_reads_per_umi = filtered_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_reads_per_umi = median((nRead/nUmi), na.rm = TRUE))
median_reads_per_umi

# mean_reads_per_umi = filtered_CaH_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_reads_per_umi = mean((nRead/nUmi), na.rm = TRUE))
# mean_reads_per_umi

#median/mean pctintronic
median_pct_intronic = filtered_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))
median_pct_intronic

# mean_pct_intronic = filtered_CaH_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_pct_intronic = mean(pct_intronic, na.rm = TRUE))
# mean_pct_intronic

#median/mean pctmito
median_pct_mito = filtered_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))
median_pct_mito

# mean_pct_mito = filtered_CaH_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_pct_mito = mean(pct_mito, na.rm = TRUE))
# mean_pct_mito
```
```{r}
library(gridExtra)
caudate_library_qc_df_list = list(median_nUmi, median_nGenes, median_reads_per_umi, median_pct_intronic,  median_pct_mito)
caudate_library_qc_df = reduce(caudate_library_qc_df_list, left_join, by = "donor_id")
caudate_library_qc_df

caudate_library_qc_df[, -1] <- round(caudate_library_qc_df[, -1], digits = 2)
caudate_library_qc_df
table_grob = tableGrob(caudate_library_qc_df)
ggsave("caudate_donor_qc_df.png", table_grob, width = 15, height = 6, dpi = 300)
```


```{r}
library(purrr)
# List of data frames to merge
df_list <- list(median_nUmi, median_nGenes,median_reads_per_umi, median_pct_intronic, median_pct_mito )

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "donor_id"))

# Print the merged data frame
print(merged_df)

```


```{r}
qc_graphs = function(data, y, xlab, ylab, title){

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_22_065CM", "SCF_22_067CM",  
  "SCF_23_072CF", "SCF_23_074CF", "SCF_23_075CM", "SCF_23_079CF",  
  "SCF_21_027", "SCF_21_029", "SCF_21_032", "SCF_21_034",  
  "SCF_21_038", "SCF_21_041", "SCF_23_070_MMC")  

data$donor_id = factor(data$donor_id, levels = x_order)
median = median(y)

ggplot(data, aes(x = donor_id, y = y, fill = donor_id)) +
  geom_bar(stat = "identity") + xlab(xlab) + ylab(ylab)+ labs(fill = "donor_id")+ ggtitle(label = title, subtitle = paste("Median: ", sprintf("%0.2f", median)))+geom_hline(yintercept = median, linetype = "dashed", color = "black") +
  geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
}
```

```{r}
qc_graphs(merged_df, merged_df$median_UMI, "Donors", "median UMI", "CaH Village: median UMI")
qc_graphs(merged_df, merged_df$median_nGenes, "Donors", "median nGenes", "CaH Village: median nGenes")
qc_graphs(merged_df, merged_df$median_reads_per_umi, "Donors", "median reads/UMI", "CaH Village: median reads/UMI")
qc_graphs(merged_df, merged_df$median_pct_intronic, "Donors", "median pct intronic", "CaH Village: median pct intronic")
qc_graphs(merged_df, merged_df$median_pct_mito, "Donors", "median pct mito", "CaH Village: median pct mito")
```


#putamen
```{r}
#totalcells
cells_per_library = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id))
names(cells_per_library)[names(cells_per_library) == "Var1"] ="library"
names(cells_per_library)[names(cells_per_library) == "Freq"] ="total_cells"
cells_per_library  

#median/mean nUMI
median_nUmi = filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi

# mean_nUmi = filtered_Put_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_UMI = mean(nUmi, na.rm = TRUE))
# mean_nUmi

#median/mean nGenes
median_nGenes = filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_nGenes = median(nFeature_RNA, na.rm = TRUE))
median_nGenes

# mean_nGenes = filtered_Put_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_nGenes = mean(nFeature_RNA, na.rm = TRUE))
# mean_nGenes

#median/mean reads/numi
median_reads_per_umi = filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_reads_per_umi = median((nRead/nUmi), na.rm = TRUE))
median_reads_per_umi

# mean_reads_per_umi = filtered_Put_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_reads_per_umi = mean((nRead/nUmi), na.rm = TRUE))
# mean_reads_per_umi

#median/mean pctintronic
median_pct_intronic = filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))
median_pct_intronic

# mean_pct_intronic = filtered_Put_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_pct_intronic = mean(pct_intronic, na.rm = TRUE))
# mean_pct_intronic

#median/mean pctmito
median_pct_mito = filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))
median_pct_mito

# mean_pct_mito = filtered_Put_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_pct_mito = mean(pct_mito, na.rm = TRUE))
# mean_pct_mito

# List of data frames to merge
df_list <- list(median_nUmi, median_nGenes,median_reads_per_umi, median_pct_intronic, median_pct_mito )

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "donor_id"))

# Print the merged data frame
print(merged_df)
```

```{r}
putamen_library_qc_df_list = list(median_nUmi, median_nGenes, median_reads_per_umi, median_pct_intronic,  median_pct_mito)
putamen_library_qc_df = reduce(putamen_library_qc_df_list, left_join, by = "donor_id")
putamen_library_qc_df

putamen_library_qc_df[, -1] <- round(putamen_library_qc_df[, -1], digits = 2)
putamen_library_qc_df
table_grob = tableGrob(putamen_library_qc_df)
ggsave("putamen_donor_qc_df.png", table_grob, width = 15, height = 6, dpi = 300)
```

```{r}
qc_graphs(merged_df, merged_df$median_UMI, "Donors", "median UMI", "Put Village: median UMI")
qc_graphs(merged_df, merged_df$median_nGenes, "Donors", "median nGenes", "Put Village: median nGenes")
qc_graphs(merged_df, merged_df$median_reads_per_umi, "Donors", "median reads/UMI", "Put Village: median reads/UMI")
qc_graphs(merged_df, merged_df$median_pct_intronic, "Donors", "median pct intronic", "Put Village: median pct intronic")
qc_graphs(merged_df, merged_df$median_pct_mito, "Donors", "median pct mito", "Put Village: median pct mito")
```



```{r}
#totalcells
cells_per_library = as.data.frame(table(filtered_CaH_Cohort2@meta.data$library))
names(cells_per_library)[names(cells_per_library) == "Var1"] ="library"
names(cells_per_library)[names(cells_per_library) == "Freq"] ="total_cells"
cells_per_library  

#median/mean nUMI
median_nUmi = filtered_CaH_Cohort2@meta.data %>%
  group_by(library) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi

#median/mean nGenes
median_nGenes = filtered_CaH_Cohort2@meta.data %>%
  group_by(library) %>%
  summarize(median_nGenes = median(nFeature_RNA, na.rm = TRUE))
median_nGenes

#median/mean reads/numi
median_reads_per_umi = filtered_CaH_Cohort2@meta.data %>%
  group_by(library) %>%
  summarize(median_reads_per_umi = median((nRead/nUmi), na.rm = TRUE))
median_reads_per_umi

#median/mean pctintronic
median_pct_intronic = filtered_CaH_Cohort2@meta.data %>%
  group_by(library) %>%
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))
median_pct_intronic


#median/mean pctmito
median_pct_mito = filtered_CaH_Cohort2@meta.data %>%
  group_by(library) %>%
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))
median_pct_mito

```
```{r}
library(gridExtra)
caudate_library_qc_df_list = list(median_nUmi, median_nGenes, median_reads_per_umi, median_pct_intronic,  median_pct_mito)
caudate_library_qc_df = reduce(caudate_library_qc_df_list, left_join, by = "library")
caudate_library_qc_df

caudate_library_qc_df[, -1] <- round(caudate_library_qc_df[, -1], digits = 2)
caudate_library_qc_df
table_grob = tableGrob(caudate_library_qc_df)
ggsave("caudate_library_qc_df.png", table_grob, width = 15, height = 6, dpi = 300)
```


```{r}
library(purrr)
# List of data frames to merge
df_list <- list(median_nUmi, median_nGenes,median_reads_per_umi, median_pct_intronic, median_pct_mito )

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "library"))

# Print the merged data frame
print(merged_df)

```

```{r}
merged_df_final_combined = rbind(merged_df,merged_df_final)
merged_df_final_combined
```



```{r}
qc_graphs_lib = function(data, y, xlab, ylab, title){

x_order = c("CaH_rxn1", "CaH_rxn2", "Put_rxn1", "Put_rxn2")  

data$library = factor(data$library, levels = x_order)
median = median(y)

ggplot(data, aes(x = library, y = y, fill = library)) +
  geom_bar(stat = "identity") + xlab(xlab) + ylab(ylab)+ labs(fill = "library")+ ggtitle(label = title, subtitle = paste("Median: ", sprintf("%0.2f", median)))+geom_hline(yintercept = median, linetype = "dashed", color = "black") +
  geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
}
```

```{r}
qc_graphs_lib(merged_df_final_combined, merged_df_final_combined$median_UMI, "Library", "median UMI", "CaH + Put: median UMI")
qc_graphs_lib(merged_df_final_combined, merged_df_final_combined$median_nGenes, "Library", "median nGenes", "CaH + Put: median nGenes")
qc_graphs_lib(merged_df_final_combined, merged_df_final_combined$median_reads_per_umi, "Library", "median reads/UMI", "CaH + Put: median reads/UMI")
qc_graphs_lib(merged_df_final_combined, merged_df_final_combined$median_pct_intronic, "Library", "median pct intronic", "CaH + Put: median pct intronic")
qc_graphs_lib(merged_df_final_combined, merged_df_final_combined$median_pct_mito, "Library", "median pct mito", "CaH + Put: median pct mito")
```



```{r}
filtered_CaH_Cohort2
filtered_Put_Cohort2
```

#Caudate cleaning + integration

```{r}
table(filtered_CaH_Cohort2$cell_class)
```


```{r}
temp_sobj = subset(filtered_CaH_Cohort2, subset = cell_class == "neuron")

# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(temp_sobj, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
temp_sobj = (temp_sobj
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(temp_sobj, group.by = "donor_id") 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.8", label=T) 

FeaturePlot(temp_sobj, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(temp_sobj, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(temp_sobj, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(temp_sobj, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(temp_sobj, features = c("OLIG1", "MOG", "MOBP", "MBP"), raster = F)
FeaturePlot(temp_sobj, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(temp_sobj, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(temp_sobj, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(temp_sobj, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 

```


```{r}
marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7", "VCAN")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "RNA_snn_res.0.3")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)


 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "RNA_snn_res.0.3")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```
```{r}
table(temp_sobj$RNA_snn_res.0.5)
```


```{r}
temp_sobj
temp_sobj = subset(temp_sobj, subset = RNA_snn_res.0.5 != "4" & RNA_snn_res.0.5 != "6"& RNA_snn_res.0.5 != "7")
temp_sobj
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.5")
```


```{r}
FeaturePlot(temp_sobj, features = c("SYT1", "AQP4", "C1QA","FLT1"),raster=FALSE)
FeaturePlot(temp_sobj, features = c("OLIG1", "OLIG2", "MOG" ,"CD96"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("MBP", "MOBP", "TF", "ST18"),raster=FALSE)
FeaturePlot(temp_sobj, features = c("RBFOX3", "CX3CR1", "GFAP" ,"AQP4"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("GAD1", "GAD2", "SLC17A7" ,"SLC17A6"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("DRD1", "DRD2", "CASZ1" ,"PPP1R1B"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("EPHA4", "SEMA3E", "SLC17A7" ,"SLC17A6"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("P2RY12", "CX3CR1", "C1QB", "BCAS1"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SEMA5B", "KREMEN1", "PDE1C"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("TAC1", "NNAT", "OPRM1", "ID4"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SGK1", "EPHA4", "SV2B"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SST", "CALB1", "PVALB", "LAMP5"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("pct_mito"),raster=FALSE) 

```



```{r}
Idents(temp_sobj) = temp_sobj$RNA_snn_res.0.2

classes = c("D1_matrix", "D2_matrix", "doublet", "D1_patch", "interneuron", "D2_patch", "interneuron", "eSPN", "interneuron", "doublet", "doublet", "notsure", "interneuron", "doublet", "interneuron")

temp_sobj= assignCellClasses(temp_sobj, classes=classes, cluster_col="RNA_snn_res.0.3", class_col = "neuron_cell_class")

DimPlot(temp_sobj, group.by = "neuron_cell_class" ,label = T, raster = FALSE)
```

```{r}
marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)

marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7", "VCAN")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)


 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```
```{r}
temp_sobj
temp_sobj = subset(temp_sobj, subset = neuron_cell_class != "doublet")
temp_sobj
DimPlot(temp_sobj, group.by = "neuron_cell_class")
```
```{r}
CaH_astro
CaH_endo
CaH_ependymal
CaH_immune
CaH_micro
CaH_oligo
CaH_opc
CaH_neuron
#= temp_sobj
```

```{r}
CaH_astro@meta.data$neuron_cell_class = CaH_astro$cell_class
CaH_endo@meta.data$neuron_cell_class = CaH_endo$cell_class
CaH_ependymal@meta.data$neuron_cell_class = CaH_ependymal$cell_class
CaH_immune@meta.data$neuron_cell_class = CaH_immune$cell_class
CaH_micro@meta.data$neuron_cell_class = CaH_micro$cell_class

CaH_neuron

CaH_oligo@meta.data$neuron_cell_class = CaH_oligo$cell_class
CaH_opc@meta.data$neuron_cell_class = CaH_opc$cell_class
```





```{r}
sobj_list = list(CaH_astro,
CaH_endo,
CaH_ependymal,
CaH_immune,
CaH_micro,
CaH_oligo,
CaH_opc,
CaH_neuron)

counts_list <- lapply(sobj_list, function(sobj) GetAssayData(sobj, slot = "counts"))

merged_counts <- do.call(cbind, counts_list)

metadata_list <- lapply(sobj_list, function(sobj) sobj@meta.data)

merged_metadata <- do.call(rbind, metadata_list)
merged_metadata

stopifnot(all(rownames(merged_metadata) == colnames(merged_counts)))

# Create Seurat object
cleaned_CaH_Cohort2 <- CreateSeuratObject(counts = merged_counts, meta.data = merged_metadata)
cleaned_CaH_Cohort2
```


```{r}
unique(cleaned_CaH_Cohort2@meta.data$donor_id)
```


```{r}
# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(cleaned_CaH_Cohort2, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
cleaned_CaH_Cohort2 = (cleaned_CaH_Cohort2
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(cleaned_CaH_Cohort2, group.by = "donor_id") 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(cleaned_CaH_Cohort2, group.by = "cell_class", label=T)
DimPlot(cleaned_CaH_Cohort2, group.by = "neuron_cell_class", label=T)
```


```{r}
library(harmony)
cleaned_CaH_Cohort2 = (cleaned_CaH_Cohort2
    %>% RunHarmony(
        group.by = "donor_id")
    %>% FindNeighbors(reduction='harmony', dims=1:20)
    %>% FindClusters(res=0.05)
    %>% FindClusters(res=0.1)
    %>% FindClusters(res=0.2)
    %>% FindClusters(res=0.3)
    %>% FindClusters(res=0.4)
    %>% FindClusters(res=0.5)
    %>% FindClusters(res=0.6)
    %>% FindClusters(res=0.7)
    %>% FindClusters(res=0.8)    
    %>% RunUMAP(reduction="harmony", dims=1:20)
)
cleaned_CaH_Cohort2
cleaned_CaH_Cohort2@meta.data
DimPlot(cleaned_CaH_Cohort2, group.by = "donor_id") 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.05", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.1", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(cleaned_CaH_Cohort2, group.by = "cell_class", label=T)
DimPlot(cleaned_CaH_Cohort2, group.by = "neuron_cell_class", label=T)
```

```{r}
FeaturePlot(cleaned_CaH_Cohort2, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(cleaned_CaH_Cohort2, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(cleaned_CaH_Cohort2, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(cleaned_CaH_Cohort2, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(cleaned_CaH_Cohort2, features = c("OLIG1", "MOG", "MOBP", "MBP"), raster = F)
FeaturePlot(cleaned_CaH_Cohort2, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(cleaned_CaH_Cohort2, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(cleaned_CaH_Cohort2, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(cleaned_CaH_Cohort2, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
FeaturePlot(cleaned_CaH_Cohort2, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 

```


```{r}
Idents(cleaned_CaH_Cohort2) = cleaned_CaH_Cohort2$RNA_snn_res.0.2

classes = c("oligo", "oligo", "neuron", "astrocyte", "neuron", "microglia", "astrocyte", "opc", "neuron", "neuron", "ependymal", "neuron", "endothelial", "neuron", "neuron", "immune","neuron")


cleaned_CaH_Cohort2= assignCellClasses(cleaned_CaH_Cohort2, classes=classes, cluster_col="RNA_snn_res.0.2", class_col = "final_cell_class")

DimPlot(cleaned_CaH_Cohort2, group.by = "final_cell_class" ,label = T, raster = FALSE)

```

```{r}
DimPlot(cleaned_CaH_Cohort2, group.by = "donor_id") 
DimPlot(cleaned_CaH_Cohort2, group.by = "Condition") 
DimPlot(cleaned_CaH_Cohort2, group.by = "cell_class", label=T)
DimPlot(cleaned_CaH_Cohort2, group.by = "final_cell_class", label=T)
```

```{r}
cells_per_donor = as.data.frame(table(cleaned_CaH_Cohort2@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(cleaned_CaH_Cohort2@meta.data$cell_class, cleaned_CaH_Cohort2@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df


cell_class_df

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  
            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = x_order)

ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors from Caudate Village") + ylab("Number of cells by cell type") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors from Caudate Village") + ylab("Cell Type Proportion") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  
#+ facet_wrap(~ Var1)
```
```{r}
cleaned_CaH_Cohort2@meta.data
co2_metadata = read.csv("/broad/macosko/kimkathl/Cohort2/co2_metadata.csv", header = T)
co2_metadata
```

```{r}
cleaned_CaH_Cohort2@meta.data = merge(cleaned_CaH_Cohort2@meta.data, co2_metadata, by.x = "donor_id", by.y = "Donor")
cleaned_CaH_Cohort2@meta.data
```


```{r}
library(lme4)
sobj = cleaned_CaH_Cohort2
model_cols = c("donor_id", "final_cell_class", "Condition", "Gender", "Age.at.Death") #include any other covariates of interest, replace cell_class with whatever column defines the cluster annotations
pd = sobj@meta.data[,model_cols]
pd = pd[complete.cases(pd),] # don't want any NAs
pd$case_control_factor = as.factor(pd$Condition)

masc_df = .sconline.MASCfn(
    dataset=pd,
    cluster=pd$final_cell_class, # cluster annotations
    contrast="case_control_factor", # name of contrast annotations (what you want to run the test for)
    random_effects="donor_id", # name of random effects annotations (not interested in these coefficients, but account for variability in probable sources ob batch effects, in this case donor
    fixed_effects = c("Age.at.Death", "Gender") # your covariates
)

masc_df

my_order <- unique(masc_df$cluster)
masc_df$cluster_name <- sub("cluster", "", masc_df$cluster) # for legibility
masc_df$log2_or = log2(masc_df$case_control_factorXDP.OR) # the names of the case_control_factor<whatever> columns will change from project to project depending on the unique values of your Condition column
masc_df$log2_or_ci_low = log2(masc_df$case_control_factorXDP.OR.95pct.ci.lower)
masc_df$log2_or_ci_high = log2(masc_df$case_control_factorXDP.OR.95pct.ci.upper)

# order the graph to put disease-enriched populations on top
masc_df = masc_df[order(-masc_df$log2_or), ] 
masc_df$cluster_name = factor(masc_df$cluster_name, levels = masc_df$cluster_name[order(masc_df$log2_or)])
masc_df


library(RColorBrewer)
library(ggplot2)

# Create the forest plot with ticks on error bars, axis lines with ticks, RdBu color map, and opaque white circles on top
ggplot(masc_df, aes(y = cluster_name, x = log2_or)) +
  ggtitle("XDP vs. Control: Caudate Cohort 2") +
  geom_vline(xintercept = 0, linetype = "dotted", color = "gray") +  # Add dotted vertical line at x=0
  geom_segment(aes(x = log2_or_ci_low, xend = log2_or_ci_high, y = cluster_name, yend = cluster_name, color = log2_or), size = 1) +  # Add horizontal error bars
  geom_point(size = 3, aes(color = log2_or), shape = 1) +  # Add points for effect sizes
  geom_point(size = 3, shape = 21, fill = "white") +  # Add opaque white circle on top of the error bar line
  scale_color_gradientn(colors = RColorBrewer::brewer.pal(10, "RdBu")) +  # Use RdBu color map
  theme_minimal() +  # Minimal theme
  labs(x = "log2(OR)", y = "Cell Classes") +  # Axis labels
  theme(
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), 
    legend.position = "none",
    plot.title = element_text(size=16),
    axis.line = element_line(color = "black"),  # Add axis lines
    axis.ticks = element_line(color = "black"),  # Add axis ticks
    axis.text = element_text(size = 14),  # Increase tick label font size
    axis.title = element_text(size = 15)  # Increase axis label font size
  )
```




```{r}
qsave(cleaned_CaH_Cohort2, "Cohort2/sobjs/cleaned_CaH_Cohort2.qs")
```



```{r}
#totalcells
cells_per_library = as.data.frame(table(cleaned_CaH_Cohort2@meta.data$donor_id))
names(cells_per_library)[names(cells_per_library) == "Var1"] ="library"
names(cells_per_library)[names(cells_per_library) == "Freq"] ="total_cells"
cells_per_library  

#median/mean nUMI
median_nUmi = cleaned_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi


#median/mean nGenes
median_nGenes = cleaned_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_nGenes = median(nFeature_RNA, na.rm = TRUE))
median_nGenes


#median/mean reads/numi
median_reads_per_umi = cleaned_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_reads_per_umi = median((nRead/nUmi), na.rm = TRUE))
median_reads_per_umi


#median/mean pctintronic
median_pct_intronic = cleaned_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))
median_pct_intronic

#median/mean pctmito
median_pct_mito = cleaned_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))
median_pct_mito


```

```{r}
library(purrr)
# List of data frames to merge
df_list <- list(median_nUmi, median_nGenes,median_reads_per_umi, median_pct_intronic, median_pct_mito )

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "donor_id"))

# Print the merged data frame
print(merged_df)

```



```{r}
qc_graphs(merged_df, merged_df$median_UMI, "Donors", "median UMI", "CaH Village: median UMI")
qc_graphs(merged_df, merged_df$median_nGenes, "Donors", "median nGenes", "CaH Village: median nGenes")
qc_graphs(merged_df, merged_df$median_reads_per_umi, "Donors", "median reads/UMI", "CaH Village: median reads/UMI")
qc_graphs(merged_df, merged_df$median_pct_intronic, "Donors", "median pct intronic", "CaH Village: median pct intronic")
qc_graphs(merged_df, merged_df$median_pct_mito, "Donors", "median pct mito", "CaH Village: median pct mito")
```




#Putamen cleaning + integration

```{r}
table(filtered_Put_Cohort2$cell_class)
```


```{r}
temp_sobj = subset(filtered_Put_Cohort2, subset = cell_class == "neuron")

# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(temp_sobj, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
temp_sobj = (temp_sobj
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(temp_sobj, group.by = "donor_id") 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.8", label=T) 

FeaturePlot(temp_sobj, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(temp_sobj, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(temp_sobj, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(temp_sobj, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(temp_sobj, features = c("OLIG1", "MOG", "MOBP", "MBP"), raster = F)
FeaturePlot(temp_sobj, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(temp_sobj, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(temp_sobj, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(temp_sobj, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 

```


```{r}
marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7", "VCAN")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "RNA_snn_res.0.3")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)


 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "RNA_snn_res.0.3")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```
```{r}
table(temp_sobj$RNA_snn_res.0.3)
```


```{r}
temp_sobj
temp_sobj = subset(temp_sobj, subset = RNA_snn_res.0.3 != "2" & RNA_snn_res.0.3 != "3")
temp_sobj
DimPlot(temp_sobj, group.by = "RNA_snn_res.0.3")
```




```{r}
FeaturePlot(temp_sobj, features = c("SYT1", "AQP4", "C1QA","FLT1"),raster=FALSE)
FeaturePlot(temp_sobj, features = c("OLIG1", "OLIG2", "MOG" ,"CD96"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("MBP", "MOBP", "TF", "ST18"),raster=FALSE)
FeaturePlot(temp_sobj, features = c("RBFOX3", "CX3CR1", "GFAP" ,"AQP4"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("GAD1", "GAD2", "SLC17A7" ,"SLC17A6"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("DRD1", "DRD2", "CASZ1" ,"PPP1R1B"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("EPHA4", "SEMA3E", "SLC17A7" ,"SLC17A6"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("P2RY12", "CX3CR1", "C1QB", "BCAS1"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SEMA5B", "KREMEN1", "PDE1C"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("TAC1", "NNAT", "OPRM1", "ID4"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SGK1", "EPHA4", "SV2B"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SST", "CALB1", "PVALB", "LAMP5"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("pct_mito"),raster=FALSE) 

```



```{r}
Idents(temp_sobj) = temp_sobj$RNA_snn_res.0.5

classes = c("D1_matrix", "D2_matrix", "D2_matrix", "unsure", "doublet", "eSPN", "D2_patch", "D1_patch", "interneuron", "interneuron", "interneuron", "D1_matrix", "interneuron", "doublet", "interneuron", "interneuron", "interneuron", "doublet", "interneuron")

temp_sobj= assignCellClasses(temp_sobj, classes=classes, cluster_col="RNA_snn_res.0.5", class_col = "neuron_cell_class")

DimPlot(temp_sobj, group.by = "neuron_cell_class" ,label = T, raster = FALSE)
```

```{r}
marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)

marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7", "VCAN")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)


 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```
```{r}
temp_sobj
temp_sobj = subset(temp_sobj, subset = neuron_cell_class != "doublet")
temp_sobj
DimPlot(temp_sobj, group.by = "neuron_cell_class")
```
```{r}
Put_astro
Put_endo
Put_immune
Put_micro 
Put_opc
Put_oligo
Put_neuron
#= temp_sobj

```

```{r}
Put_astro@meta.data$neuron_cell_class = Put_astro$cell_class
Put_endo@meta.data$neuron_cell_class = Put_endo$cell_class
Put_immune@meta.data$neuron_cell_class = Put_immune$cell_class
Put_micro@meta.data$neuron_cell_class = Put_micro$cell_class

Put_neuron

Put_oligo@meta.data$neuron_cell_class = Put_oligo$cell_class
Put_opc@meta.data$neuron_cell_class = Put_opc$cell_class
```

```{r}
sobj_list = list(Put_astro,
Put_endo,
Put_immune,
Put_micro,
Put_oligo,
Put_opc,
Put_neuron)

counts_list <- lapply(sobj_list, function(sobj) GetAssayData(sobj, slot = "counts"))

merged_counts <- do.call(cbind, counts_list)

metadata_list <- lapply(sobj_list, function(sobj) sobj@meta.data)

merged_metadata <- do.call(rbind, metadata_list)
merged_metadata

stopifnot(all(rownames(merged_metadata) == colnames(merged_counts)))

# Create Seurat object
cleaned_Put_Cohort2 <- CreateSeuratObject(counts = merged_counts, meta.data = merged_metadata)
cleaned_Put_Cohort2
```


```{r}
# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(cleaned_Put_Cohort2, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
cleaned_Put_Cohort2 = (cleaned_Put_Cohort2
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(cleaned_Put_Cohort2, group.by = "donor_id") 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(cleaned_Put_Cohort2, group.by = "cell_class", label=T)
DimPlot(cleaned_Put_Cohort2, group.by = "neuron_cell_class", label=T)
```

```{r}
library(harmony)
cleaned_Put_Cohort2 = (cleaned_Put_Cohort2
    %>% RunHarmony(
        group.by = "donor_id")
    %>% FindNeighbors(reduction='harmony', dims=1:20)
    %>% FindClusters(res=0.05)
    %>% FindClusters(res=0.1)
    %>% FindClusters(res=0.2)
    %>% FindClusters(res=0.3)
    %>% FindClusters(res=0.4)
    %>% FindClusters(res=0.5)
    %>% FindClusters(res=0.6)
    %>% FindClusters(res=0.7)
    %>% FindClusters(res=0.8)    
    %>% RunUMAP(reduction="harmony", dims=1:20)
)
cleaned_Put_Cohort2
cleaned_Put_Cohort2@meta.data
DimPlot(cleaned_Put_Cohort2, group.by = "donor_id") 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.05", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.1", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(cleaned_Put_Cohort2, group.by = "cell_class", label=T)
DimPlot(cleaned_Put_Cohort2, group.by = "neuron_cell_class", label=T)
```
```{r}
FeaturePlot(cleaned_Put_Cohort2, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(cleaned_Put_Cohort2, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(cleaned_Put_Cohort2, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(cleaned_Put_Cohort2, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(cleaned_Put_Cohort2, features = c("OLIG1", "MOG", "MOBP", "MBP"), raster = F)
FeaturePlot(cleaned_Put_Cohort2, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(cleaned_Put_Cohort2, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(cleaned_Put_Cohort2, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(cleaned_Put_Cohort2, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
FeaturePlot(cleaned_Put_Cohort2, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 

```


```{r}
Idents(cleaned_Put_Cohort2) = cleaned_Put_Cohort2$RNA_snn_res.0.2

classes = c("oligo", "oligo", "oligo", "neuron", "astrocyte", "neuron", "oligo", "microglia", "opc", "neuron", "neuron", "endothelial", "neuron", "immune")


cleaned_Put_Cohort2= assignCellClasses(cleaned_Put_Cohort2, classes=classes, cluster_col="RNA_snn_res.0.2", class_col = "final_cell_class")

DimPlot(cleaned_Put_Cohort2, group.by = "final_cell_class" ,label = T, raster = FALSE)

```

```{r}
DimPlot(cleaned_Put_Cohort2, group.by = "donor_id") 
DimPlot(cleaned_Put_Cohort2, group.by = "Condition") 
DimPlot(cleaned_Put_Cohort2, group.by = "cell_class", label=T)
DimPlot(cleaned_Put_Cohort2, group.by = "final_cell_class", label=T)
```

```{r}
cells_per_donor = as.data.frame(table(cleaned_Put_Cohort2@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(cleaned_Put_Cohort2@meta.data$cell_class, cleaned_Put_Cohort2@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df


cell_class_df

x_order = c("SCF_23_072CF", "SCF_22_049CCF","SCF_23_079CF", "SCF_22_063CM","SCF_22_065CM",  "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF",   
"SCF_21_032","SCF_21_038",  "SCF_21_041","SCF_23_070_MMC", "SCF_21_034", "SCF_21_029","SCF_21_027") 
            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = x_order)

ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors from Putamen Village") + ylab("Number of cells by cell type") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors from Putamen Village") + ylab("Cell Type Proportion") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  
#+ facet_wrap(~ Var1)
```
```{r}
cleaned_Put_Cohort2@meta.data
co2_metadata = read.csv("/broad/macosko/kimkathl/Cohort2/co2_metadata.csv", header = T)
co2_metadata


cleaned_Put_Cohort2@meta.data = merge(cleaned_Put_Cohort2@meta.data, co2_metadata, by.x = "donor_id", by.y = "Donor")
cleaned_Put_Cohort2@meta.data
```


```{r}
library(lme4)
sobj = cleaned_Put_Cohort2
model_cols = c("donor_id", "final_cell_class", "Condition", "Gender", "Age.at.Death") #include any other covariates of interest, replace cell_class with whatever column defines the cluster annotations
pd = sobj@meta.data[,model_cols]
pd = pd[complete.cases(pd),] # don't want any NAs
pd$case_control_factor = as.factor(pd$Condition)

masc_df = .sconline.MASCfn(
    dataset=pd,
    cluster=pd$final_cell_class, # cluster annotations
    contrast="case_control_factor", # name of contrast annotations (what you want to run the test for)
    random_effects="donor_id", # name of random effects annotations (not interested in these coefficients, but account for variability in probable sources ob batch effects, in this case donor
    fixed_effects = c("Age.at.Death", "Gender") # your covariates
)

masc_df

my_order <- unique(masc_df$cluster)
masc_df$cluster_name <- sub("cluster", "", masc_df$cluster) # for legibility
masc_df$log2_or = log2(masc_df$case_control_factorXDP.OR) # the names of the case_control_factor<whatever> columns will change from project to project depending on the unique values of your Condition column
masc_df$log2_or_ci_low = log2(masc_df$case_control_factorXDP.OR.95pct.ci.lower)
masc_df$log2_or_ci_high = log2(masc_df$case_control_factorXDP.OR.95pct.ci.upper)

# order the graph to put disease-enriched populations on top
masc_df = masc_df[order(-masc_df$log2_or), ] 
masc_df$cluster_name = factor(masc_df$cluster_name, levels = masc_df$cluster_name[order(masc_df$log2_or)])
masc_df


library(RColorBrewer)
library(ggplot2)

# Create the forest plot with ticks on error bars, axis lines with ticks, RdBu color map, and opaque white circles on top
ggplot(masc_df, aes(y = cluster_name, x = log2_or)) +
  ggtitle("XDP vs. Control: Putamen Cohort 2") +
  geom_vline(xintercept = 0, linetype = "dotted", color = "gray") +  # Add dotted vertical line at x=0
  geom_segment(aes(x = log2_or_ci_low, xend = log2_or_ci_high, y = cluster_name, yend = cluster_name, color = log2_or), size = 1) +  # Add horizontal error bars
  geom_point(size = 3, aes(color = log2_or), shape = 1) +  # Add points for effect sizes
  geom_point(size = 3, shape = 21, fill = "white") +  # Add opaque white circle on top of the error bar line
  scale_color_gradientn(colors = RColorBrewer::brewer.pal(10, "RdBu")) +  # Use RdBu color map
  theme_minimal() +  # Minimal theme
  labs(x = "log2(OR)", y = "Cell Classes") +  # Axis labels
  theme(
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), 
    legend.position = "none",
    plot.title = element_text(size=16),
    axis.line = element_line(color = "black"),  # Add axis lines
    axis.ticks = element_line(color = "black"),  # Add axis ticks
    axis.text = element_text(size = 14),  # Increase tick label font size
    axis.title = element_text(size = 15)  # Increase axis label font size
  )
```




```{r}
qsave(cleaned_Put_Cohort2, "Cohort2/sobjs/cleaned_Put_Cohort2.qs")
```



```{r}
#totalcells
cells_per_library = as.data.frame(table(cleaned_Put_Cohort2@meta.data$donor_id))
names(cells_per_library)[names(cells_per_library) == "Var1"] ="library"
names(cells_per_library)[names(cells_per_library) == "Freq"] ="total_cells"
cells_per_library  

#median/mean nUMI
median_nUmi = cleaned_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi


#median/mean nGenes
median_nGenes = cleaned_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_nGenes = median(nFeature_RNA, na.rm = TRUE))
median_nGenes


#median/mean reads/numi
median_reads_per_umi = cleaned_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_reads_per_umi = median((nRead/nUmi), na.rm = TRUE))
median_reads_per_umi


#median/mean pctintronic
median_pct_intronic = cleaned_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))
median_pct_intronic

#median/mean pctmito
median_pct_mito = cleaned_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))
median_pct_mito


```

```{r}
library(purrr)
# List of data frames to merge
df_list <- list(median_nUmi, median_nGenes,median_reads_per_umi, median_pct_intronic, median_pct_mito )

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "donor_id"))

# Print the merged data frame
print(merged_df)

```

```{r}
table(cleaned_Put_Cohort2$donor_id)
```


```{r}
qc_graphs(merged_df, merged_df$median_UMI, "Donors", "median UMI", "Put Village: median UMI")
qc_graphs(merged_df, merged_df$median_nGenes, "Donors", "median nGenes", "Put Village: median nGenes")
qc_graphs(merged_df, merged_df$median_reads_per_umi, "Donors", "median reads/UMI", "Put Village: median reads/UMI")
qc_graphs(merged_df, merged_df$median_pct_intronic, "Donors", "median pct intronic", "Put Village: median pct intronic")
qc_graphs(merged_df, merged_df$median_pct_mito, "Donors", "median pct mito", "Put Village: median pct mito")
```


#TAF1

```{r}
FeaturePlot(cleaned_CaH_Cohort2, features = c("TAF1"))
FeaturePlot(cleaned_Put_Cohort2, features = c("TAF1"))
```

```{r}
xdp_co1 = qread("/broad/macosko/kimkathl/temp_disco/more_temp_1205/xdp_cah_put_sct_new.qs")
xdp_co1
xdp_co1@meta.data
table(xdp_co1$neuron_classes, xdp_co1$Region)
```
```{r}
DefaultAssay(xdp_co1) = "RNA"
xdp_co1
```

```{r}
cleaned_CaH_Cohort2
cleaned_Put_Cohort2
xdp_co1
```
```{r}
# Define genes of interest
genes_of_interest <- c("TAF1", "GAPDH")  # Replace GAPDH with a suitable control gene

# Fetch data from each experiment
expr_nextgem <- FetchData(xdp_co1, vars = genes_of_interest)
expr_gemx <- GetAssayData(cleaned_CaH_Cohort2, slot = "counts")[gene_of_interest, ]
expr_nextgem
expr_gemx

# Add metadata
expr_nextgem$Experiment <- "Cohort1"
expr_gemx$Experiment <- "Cohort2"

# Add donor information
expr_nextgem$Donor <- xdp_co1@meta.data$donor_id
expr_gemx$Donor <- cleaned_CaH_Cohort2@meta.data$donor_id

# Add cell type information (optional)
expr_nextgem$Cell_Type <- xdp_co1@meta.data$cell_type_column
expr_gemx$Cell_Type <- cleaned_CaH_Cohort2@meta.data$cell_type_column

# Merge both experiments into one dataframe
expr_data <- bind_rows(expr_nextgem, expr_gemx)
expr_data
# Convert to long format for ggplot
expr_long <- expr_data %>%
  pivot_longer(cols = c("TAF1", "GAPDH"), names_to = "Gene", values_to = "Expression")
expr_long

```


```{r}
library(ggplot2)

ggplot(expr_long, aes(x = Experiment, y = Expression, fill = Experiment)) +
  geom_violin(alpha = 0.5) +
  geom_jitter(aes(color = Donor), width = 0.2, size = 2) +
  facet_wrap(~Gene, scales = "free_y") +  # Separate plots for TAF1 and the control gene
  stat_summary(fun = mean, geom = "errorbar", width = 0.2, aes(ymax = ..y.., ymin = ..y..), color = "black", size = 1.2) +
  theme_minimal() +
  labs(title = "TAF1 vs Control Gene Expression: NextGem vs GemX", 
       x = "Experiment", 
       y = "Expression")
# Wilcoxon test for TAF1
taf1_test <- wilcox.test(Expression ~ Experiment, data = filter(expr_long, Gene == "TAF1"))

# Wilcoxon test for control gene
gapdh_test <- wilcox.test(Expression ~ Experiment, data = filter(expr_long, Gene == "GAPDH"))

# Print p-values
print(paste("TAF1 p-value:", taf1_test$p.value))
print(paste("Control Gene p-value:", gapdh_test$p.value))
```


```{r}
# Normalize TAF1 expression to total counts per cell
expr_nextgem$TAF1_norm <- expr_nextgem$TAF1 / rowSums(FetchData(nextgem_obj, vars = rownames(nextgem_obj)))
expr_gemx$TAF1_norm <- expr_gemx$TAF1 / rowSums(FetchData(gemx_obj, vars = rownames(gemx_obj)))

# Merge back
expr_norm_data <- bind_rows(expr_nextgem, expr_gemx)

ggplot(expr_norm_data, aes(x = Experiment, y = TAF1_norm, fill = Experiment)) +
  geom_violin(alpha = 0.5) +
  geom_jitter(aes(color = Donor), width = 0.2, size = 2) +
  stat_summary(fun = mean, geom = "errorbar", width = 0.2, aes(ymax = ..y.., ymin = ..y..), color = "black", size = 1.2) +
  theme_minimal() +
  labs(title = "Normalized TAF1 Expression: NextGem vs GemX", 
       x = "Experiment", 
       y = "TAF1 Expression (Normalized)")

```







```{r}
# Define the gene of interest
gene_of_interest <- "TAF1"

# Extract expression data
expr_data <- FetchData(xdp_co1, vars = gene_of_interest)

# Get cell class metadata
expr_data$cell_class <- xdp_co1@meta.data$neuron_classes  # Change to your actual cell class column name
expr_data$donor =  xdp_co1@meta.data$donor_id
expr_data$Condition =  xdp_co1@meta.data$Condition
expr_data$Region = xdp_co1@meta.data$Region

# Compute average expression per cell class
avg_expr <- expr_data %>%
  group_by(cell_class, donor, Condition, Region) %>%
  summarise(avg_expression = mean(.data[[gene_of_interest]], na.rm = TRUE))
avg_expr

library(ggplot2)

avg_expr_cah = subset(avg_expr, subset = Region == "Caudate")

p_values <- avg_expr_cah %>%
  group_by(cell_class) %>%
  summarise(p_value = wilcox.test(avg_expression ~ Condition)$p.value) %>%
  mutate(p_label = paste0(cell_class, "\n(p = ", signif(p_value, 3), ")"))  # Format p-value

# Merge p-values with original data
avg_expr_cah <- avg_expr_cah %>%
  left_join(p_values, by = "cell_class")

# Plot with p-values in facet labels
ggplot(avg_expr_cah, aes(x = Condition, y = avg_expression)) +
  geom_point(aes(color = Condition), size = 2, position = position_jitter(width = 0.2, height = 0)) +  # One dot per donor
  stat_summary(fun = mean, geom = "errorbar", width = 0.2, aes(ymax = ..y.., ymin = ..y..), color = "black", size = 1.2) +  # Black bar for average
  facet_wrap(~p_label, nrow = 3, ncol = 4, scales = "free_y") +  # Use new labels with p-values
  theme_minimal() +
  labs(title = "Caudate: TAF1 Expression per Cell Class", 
       x = "Condition (Control vs Case)", 
       y = "Average Expression") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

avg_expr_put = subset(avg_expr, subset = Region == "Putamen")

p_values <- avg_expr_put %>%
  group_by(cell_class) %>%
  summarise(p_value = wilcox.test(avg_expression ~ Condition)$p.value) %>%
  mutate(p_label = paste0(cell_class, "\n(p = ", signif(p_value, 3), ")"))  # Format p-value

# Merge p-values with original data
avg_expr_put <- avg_expr_put %>%
  left_join(p_values, by = "cell_class")

# Plot with p-values in facet labels
ggplot(avg_expr_put, aes(x = Condition, y = avg_expression)) +
  geom_point(aes(color = Condition), size = 2, position = position_jitter(width = 0.2, height = 0)) +  # One dot per donor
  stat_summary(fun = mean, geom = "errorbar", width = 0.2, aes(ymax = ..y.., ymin = ..y..), color = "black", size = 1.2) +  # Black bar for average
  facet_wrap(~p_label, nrow = 3, ncol = 4, scales = "free_y") +  # Use new labels with p-values
  theme_minimal() +
  labs(title = "Putamen: TAF1 Expression per Cell Class", 
       x = "Condition (Control vs Case)", 
       y = "Average Expression") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))



```



```{r}
cleaned_CaH_Cohort2@meta.data
cleaned_Put_Cohort2@meta.data
```
```{r}
# Assuming caudate_obj and putamen_obj are your Seurat objects
# Add a region label to each Seurat object before merging
cleaned_CaH_Cohort2@meta.data$region <- "caudate"
cleaned_Put_Cohort2@meta.data$region <- "putamen"
```

```{r}
sobj_list = list(cleaned_CaH_Cohort2,
cleaned_Put_Cohort2)

counts_list <- lapply(sobj_list, function(sobj) GetAssayData(sobj, slot = "counts"))

merged_counts <- do.call(cbind, counts_list)

metadata_list <- lapply(sobj_list, function(sobj) sobj@meta.data)

merged_metadata <- do.call(rbind, metadata_list)
rownames(merged_metadata) = paste0(merged_metadata$library, "__", merged_metadata$cell)
# Ensure that the row names of metadata match the colnames of merged counts
merged_metadata <- merged_metadata[match(colnames(merged_counts), rownames(merged_metadata)), ]
# Reorder merged_counts to match the cell names in the merged metadata
merged_counts <- merged_counts[, match(rownames(merged_metadata), colnames(merged_counts))]
# Check if the order of rownames and colnames match
all(rownames(merged_metadata) == colnames(merged_counts))  # Should return TRUE

stopifnot(all(rownames(merged_metadata) == colnames(merged_counts)))

# Create Seurat object
cleaned_CaH_Put_Cohort2 <- CreateSeuratObject(counts = merged_counts, meta.data = merged_metadata)
cleaned_CaH_Put_Cohort2@meta.data
```


```{r}
# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(cleaned_CaH_Put_Cohort2, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
cleaned_CaH_Put_Cohort2 = (cleaned_CaH_Put_Cohort2
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(cleaned_CaH_Put_Cohort2, group.by = "donor_id") 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "cell_class", label=T)
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "neuron_cell_class", label=T)
```


```{r}
library(harmony)
cleaned_CaH_Put_Cohort2 = (cleaned_CaH_Put_Cohort2
    %>% RunHarmony(
        group.by = "donor_id")
    %>% FindNeighbors(reduction='harmony', dims=1:20)
    %>% FindClusters(res=0.05)
    %>% FindClusters(res=0.1)
    %>% FindClusters(res=0.2)
    %>% FindClusters(res=0.3)
    %>% FindClusters(res=0.4)
    %>% FindClusters(res=0.5)
    %>% FindClusters(res=0.6)
    %>% FindClusters(res=0.7)
    %>% FindClusters(res=0.8)    
    %>% RunUMAP(reduction="harmony", dims=1:20)
)
cleaned_CaH_Put_Cohort2
cleaned_CaH_Put_Cohort2@meta.data
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "donor_id") 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.05", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.1", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "cell_class", label=T)
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "neuron_cell_class", label=T)
```

```{r}
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "region", label=T)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("OLIG1", "MOG", "MOBP", "MBP"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
FeaturePlot(cleaned_CaH_Put_Cohort2, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 

```


```{r}
Idents(cleaned_CaH_Put_Cohort2) = cleaned_CaH_Put_Cohort2$RNA_snn_res.0.2

classes = c("oligo", "oligo", "astrocyte", "neuron", "neuron", "microglia", "opc", "neuron", "neuron", "neuron", "neuron", "endothelial", "ependymal", "neuron", "immune")


cleaned_CaH_Put_Cohort2= assignCellClasses(cleaned_CaH_Put_Cohort2, classes=classes, cluster_col="RNA_snn_res.0.2", class_col = "final_cell_class_merged")

DimPlot(cleaned_CaH_Put_Cohort2, group.by = "final_cell_class_merged" ,label = T, raster = FALSE)

```

```{r}
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "region")
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "donor_id")
DimPlot(cleaned_CaH_Put_Cohort2, group.by = "final_cell_class_merged", label=T)
```

```{r}
qsave(cleaned_CaH_Put_Cohort2, "Cohort2/sobjs/cleaned_CaH_Put_Cohort2.qs")
```

```{r}
table(sobj$donor_id)
```
```{r}
rownames(cleaned_Put_Cohort2@meta.data) = paste0(cleaned_Put_Cohort2@meta.data$library, "__", cleaned_Put_Cohort2@meta.data$cell)
cleaned_Put_Cohort2@meta.data
```


```{r}
cleaned_CaH_Put_Cohort2@meta.data
```

#MASC Combo
```{r}
library(lme4)
sobj = cleaned_CaH_Put_Cohort2
sobj = subset(sobj, subset = donor_id != "SCF_22_049CCF")
sobj
model_cols = c("donor_id", "final_cell_class_merged", "Condition", "Gender", "Age.at.Death", "region") #include any other covariates of interest, replace cell_class with whatever column defines the cluster annotations
pd = sobj@meta.data[,model_cols]
pd = pd[complete.cases(pd),] # don't want any NAs
pd$case_control_factor = as.factor(pd$Condition)

masc_df = .sconline.MASCfn(
    dataset=pd,
    cluster=pd$final_cell_class, # cluster annotations
    contrast="case_control_factor", # name of contrast annotations (what you want to run the test for)
    random_effects="donor_id", # name of random effects annotations (not interested in these coefficients, but account for variability in probable sources ob batch effects, in this case donor
    fixed_effects = c("Age.at.Death", "Gender", "region") # your covariates
)

masc_df

my_order <- unique(masc_df$cluster)
masc_df$cluster_name <- sub("cluster", "", masc_df$cluster) # for legibility
masc_df$log2_or = log2(masc_df$case_control_factorXDP.OR) # the names of the case_control_factor<whatever> columns will change from project to project depending on the unique values of your Condition column
masc_df$log2_or_ci_low = log2(masc_df$case_control_factorXDP.OR.95pct.ci.lower)
masc_df$log2_or_ci_high = log2(masc_df$case_control_factorXDP.OR.95pct.ci.upper)

# order the graph to put disease-enriched populations on top
masc_df = masc_df[order(-masc_df$log2_or), ] 
masc_df$cluster_name = factor(masc_df$cluster_name, levels = masc_df$cluster_name[order(masc_df$log2_or)])
masc_df


library(RColorBrewer)
library(ggplot2)

# Create the forest plot with ticks on error bars, axis lines with ticks, RdBu color map, and opaque white circles on top
ggplot(masc_df, aes(y = cluster_name, x = log2_or)) +
  ggtitle("Combined Cohort 2 CaH + Put (no CCF)") +
  geom_vline(xintercept = 0, linetype = "dotted", color = "gray") +  # Add dotted vertical line at x=0
  geom_segment(aes(x = log2_or_ci_low, xend = log2_or_ci_high, y = cluster_name, yend = cluster_name, color = log2_or), size = 1) +  # Add horizontal error bars
  geom_point(size = 3, aes(color = log2_or), shape = 1) +  # Add points for effect sizes
  geom_point(size = 3, shape = 21, fill = "white") +  # Add opaque white circle on top of the error bar line
  scale_color_gradientn(colors = RColorBrewer::brewer.pal(10, "RdBu")) +  # Use RdBu color map
  theme_minimal() +  # Minimal theme
  labs(x = "log2(OR)", y = "Cell Classes") +  # Axis labels
  theme(
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), 
    legend.position = "none",
    plot.title = element_text(size=16),
    axis.line = element_line(color = "black"),  # Add axis lines
    axis.ticks = element_line(color = "black"),  # Add axis ticks
    axis.text = element_text(size = 14),  # Increase tick label font size
    axis.title = element_text(size = 15)  # Increase axis label font size
  )
```







```{r}
1+1
```



```{r}
log_nUMI = log10(CaH_Cohort2$nUmi)
median_log_nUMI = median(log_nUMI)
title = paste0("CaH_Cohort2: log10(nUMI)", ", Median: ", sprintf("%0.3f", median_log_nUMI))
log_nUMI_hist = hist(log_nUMI, main= title, xlab = "log10(nUMI)", col="grey", breaks = 100)
```








```{r}
options(future.globals.maxSize = 400 * 1024^3)  
library(sctransform)
cleaned_CaH_Put_Cohort2 = SCTransform(cleaned_CaH_Put_Cohort2, vars.to.regress = "pct_mito", verbose = FALSE)
DefaultAssay(cleaned_CaH_Put_Cohort2) = "SCT"
cleaned_CaH_Put_Cohort2

```

```{r}
library(UCell)
```

```{r}
NFKB_genes =  c("MARCKS", "IL23A", "NINJ1", "TNFSF9", "SIK1", "ATF3", "SERPINE1", "MYC", "HES1", "CCNL1", "CCN1", "EGR1", "EGR2", "JAG1", "EGR3", "ABCA1", "GADD45B", "GADD45A", "PLK2", "KLF10", 
           "EIF1", "EHD1", "FOSL2", "FOSL1", "GPR183", "PLPP3", "IFIT2", "ICAM1", "ZC3H12A", "IER2", 
           "IL12B", "JUNB", "IER5", "IER3", "STAT5A", "DUSP5", "EDN1", "JUN", "DUSP4", "DUSP1", 
           "DUSP2", "TSC22D1", "CCL20", "SPHK1", "LIF", "IL18", "TUBB2A", "RHOB", "VEGFA", "PTPRE", 
           "IL1A", "TLR2", "IL1B", "BHLHE40", "ID2", "CLCF1", "REL", "FJX1", "SGK1", "BTG3", "BTG2", 
           "BTG1", "SDC4", "LITAF", "AREG", "SOCS3", "PANX1", "RIPK2", "NFIL3", "SERPINB2", "GCH1", 
           "IFNGR2", "G0S2", "FOS", "SERPINB8", "F3", "SPSB1", "FOSB", "PER1", "F2RL1", "HBEGF", 
           "CD44", "TRIP10", "CDKN1A", "PTGER4", "PTGS2", "IFIH1", "NAMPT", "OLR1", "ICOSLG", 
           "PHLDA1", "ZBTB10", "TAP1", "PNRC1", "CXCL10", "CXCL11", "IL6ST", "CD69", "SQSTM1", 
           "RELA", "CSF2", "CD83", "CSF1", "PPP1R15A", "CD80", "TNC", "TNF", "TANK", "RELB", "ZFP36", 
           "CCND1", "RNF19B", "CCRL2", "DENND5A", "PHLDA2", "MAP3K8", "LDLR", "SLC16A6", "SMAD3", 
           "TGIF1", "MAP2K3", "DDX58", "TRAF1", "INHBA", "NFKB1", "NFKB2", "GEM", "NR4A3", "MAFF", 
           "RCAN1", "NR4A2", "EFNA1", "MXD1", "BIRC2", "YRDC", "BIRC3", "IL7R", "PFKFB3", "IRS2", 
           "SLC2A3", "PLAU", "SLC2A6", "SAT1", "ETS2", "NR4A1", "SNN", "PMEPA1", "TNFRSF9", "MSC", 
           "TIPARP", "LAMB3", "GFPT2", "CFLAR", "TNIP1", "IRF1", "NFKBIA", "BMP2", "IL6", "TNIP2", 
           "BCL6", "BCL3", "NFKBIE", "NFE2L2", "B4GALT1", "NFAT5", "TNFAIP8", "BCL2A1", "TNFAIP6", 
           "TNFAIP3", "CXCL2", "CXCL1", "TNFAIP2", "CXCL3", "CXCL6", "FUT4", "DRAM1", "DNAJB4", 
           "PDE4B", "PDLIM5", "MCL1", "KDM6B", "IL15RA", "PLAUR", "ATP2B1", "KLF4", "KLF2", "SOD2", 
           "KLF9", "KLF6", "ACKR3", "PTX3", "B4GALT5", "TRIB1", "CEBPB", "CEBPD", "PLEK", "KYNU", 
           "CCL5", "CCL4", "CCL2")

interferon_alpha = c("IL4R", "CD74", "SP110", "MX1", "RSAD2", "TAP1", "PSMB9", "CXCL10", "CXCL11", "IFI44L", "LAP3", "LPAR6", "IFI27", "TRAFD1", "TRIM14", "LY6E", "IFITM2", "IFITM3", "SAMD9", "CSF1", 
           "IFITM1", "UBE2L6", "SAMD9L", "RNF31", "HERC6", "LAMP3", "CCRL2", "MVB12A", "B2M", "CMTR1", 
           "GBP2", "GBP4", "EPSTI1", "BATF2", "HLA-C", "TENT5A", "TDRD7", "ISG15", "IFI44", "PARP14", 
           "MOV10", "ISG20", "PARP12", "PLSCR1", "PROCR", "SELL", "PSME2", "PSME1", "CNP", "LGALS3BP", 
           "IFI35", "IFIT2", "IFI30", "DDX60", "IFIT3", "NUB1", "HELZ2", "PNPT1", "IL15", "NCOA7", 
           "EIF2AK2", "NMI", "OAS1", "BST2", "IRF1", "ELF1", "IL7", "IRF2", "IRF9", "IRF7", "UBA7", 
           "GMPR", "ADAR", "CASP8", "TRIM5", "RIPK2", "CASP1", "STAT2", "WARS1", "PARP9", "PSMA3", 
           "TXNIP", "CMPK2", "CD47", "RTP4", "C1S", "OGFR", "TMEM140", "USP18", "OASL", "SLC25A28", 
           "IFIH1", "PSMB8", "DHX58", "TRIM25", "TRIM26", "TRIM21")



cleaned_CaH_Put_Cohort2 <- AddModuleScore_UCell(cleaned_CaH_Put_Cohort2,
  features = list(Score = NFKB_genes ),
  name = 'NFKB'
)

cleaned_CaH_Put_Cohort2 <- AddModuleScore_UCell(cleaned_CaH_Put_Cohort2,
  features = list(Score = interferon_alpha),
  name = 'interferon_alpha'
)

cleaned_CaH_Put_Cohort2 <- AddModuleScore_UCell(cleaned_CaH_Put_Cohort2,features = list(C_minus = C_minus), name = '_scores')
cleaned_CaH_Put_Cohort2 <- AddModuleScore_UCell(cleaned_CaH_Put_Cohort2,features = list(C_plus = C_plus), name = '_scores')

DefaultAssay(xdp_co1) = "SCT"
xdp_co1 <- AddModuleScore_UCell(xdp_co1,features = list(C_minus = C_minus), name = '_scores')
xdp_co1 <- AddModuleScore_UCell(xdp_co1,features = list(C_plus = C_plus), name = '_scores')
```

```{r}
cleaned_CaH_Put_Cohort2@meta.data
xdp_co1@meta.data
```

```{r}
qsave(cleaned_CaH_Put_Cohort2, "cleaned_CaH_Put_Cohort2_sct_ucell.qs")
```



```{r}
xdp_co1@meta.data$cause_of_death = "Other"
xdp_co1$cause_of_death[xdp_co1$donor_id == "SCF-18-003" | 
                             xdp_co1$donor_id == "SCF-18-004" | 
                             xdp_co1$donor_id == "SCF-19-009"| 
                             xdp_co1$donor_id == "SCF-19-014" | 
                             xdp_co1$donor_id == "SCF-20-025" | 
                             xdp_co1$donor_id == "SCF_21-037CM2"] = "Infection-related Death"


cleaned_CaH_Put_Cohort2@meta.data$cause_of_death = "Other"
cleaned_CaH_Put_Cohort2$cause_of_death[cleaned_CaH_Put_Cohort2$donor_id == "SCF_23_072CF"] = "Infection-related Death"

xdp_co1$cause_of_death_case_control = paste(xdp_co1$Condition, xdp_co1$cause_of_death) 
cleaned_CaH_Put_Cohort2$cause_of_death_case_control = paste(cleaned_CaH_Put_Cohort2$Condition, cleaned_CaH_Put_Cohort2$cause_of_death) 


cleaned_CaH_Put_Cohort2_meta = cleaned_CaH_Put_Cohort2@meta.data
xdp_co1_meta = xdp_co1@meta.data
cleaned_CaH_Put_Cohort2_meta
xdp_co1_meta
```
```{r}
cleaned_CaH_Put_Cohort2_meta$neuron_classes = cleaned_CaH_Put_Cohort2_meta$neuron_cell_class
```


```{r}
histograms_by_all_celltype(cleaned_CaH_Put_Cohort2_meta, "ScoreNFKB", "ScoreNFKB", 20, 10)
```

```{r}
histograms_by_donor(xdp_co1_meta, "ScoreNFKB", "astrocyte" ,"ScoreNFKB", 25, 15)
```


```{r}
library(dplyr)
library(ggplot2)

data <- xdp_co1_meta
df = subset(data, subset = final_cell_class == "neuron")
df
df_avg <- df %>%
  group_by(donor_id, cause_of_death_case_control, Condition, cause_of_death) %>%
  summarise(avg_score = mean(Scoreinterferon_alpha), .groups = 'drop')
df_avg

# Plot the average score per donor for each condition
a = ggplot(df_avg, aes(x = Condition, y = avg_score, color = Condition, shape = cause_of_death)) +
  geom_point(aes(group = donor_id), size = 3, alpha = 0.8) + scale_color_manual(values = rev(scales::hue_pal()(2)))+ # One point per donor
  labs(x = "Condition", y = "Mean interferonalpha Score", color = "Condition", shape = "Cause of Death", title = "Astrocytes")+ scale_shape_manual(values = rev(c(16, 17))) +
  theme_minimal()+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  + theme(
            plot.title = element_text(size = 20), # title font size
            axis.line = element_line(color = "black"),  # Add axis lines
            axis.ticks = element_line(color = "black"),  # Add axis ticks
            axis.text = element_text(size = 14),  # Increase tick label font size
            axis.title = element_text(size = 15),  # Increase axis label font size
            axis.text.x = element_text(angle = 45, hjust = 1)  # tilt axis labels
        )+
  theme(
    legend.text = element_text(size = 16),  # Increase legend text size
    legend.title = element_text(size = 18),  # Increase legend title size
   # legend.key.size = unit(1.5, "cm")  # Increase legend key size
  ) 

print(a)

ggsave(a, filename = "dotplot.png", width = 7, height = 6)

```


#wilcox test
```{r}
xdp_meta = xdp_co1_meta
xdp_meta = subset(xdp_meta, final_cell_class == "astrocyte")
xdp_meta

xdp_meta_cases = subset(xdp_meta, Condition == "XDP")
xdp_meta_controls = subset(xdp_meta, Condition == "Control")
xdp_meta_cases
xdp_meta_controls

XDP_scores <- xdp_meta_cases$ScoreNFKB  
XDP_Control_scores <- xdp_meta_controls$ScoreNFKB 

result1 <- wilcox.test(XDP_scores, XDP_Control_scores, alternative = "two.sided")
result2 <- wilcox.test(XDP_scores, XDP_Control_scores, alternative = "less")
result3 <- wilcox.test(XDP_scores, XDP_Control_scores, alternative = "greater")
print(result1)
print(result2)
print(result3)
```




---
title: "R Notebook"
output: html_notebook
---

```{r}
library(Seurat)
library(qs)
library(tidyverse)
library(ggplot2)
```


#redo filtering 
```{r}
CaH_Cohort2
Put_Cohort2

filtered_CaH_Cohort2
filtered_Put_Cohort2
```

```{r}
hist(filtered_CaH_Cohort2$prob_max, breaks =100)
hist(filtered_Put_Cohort2$prob_max, breaks =100)

hist(filtered_CaH_Cohort2$nUmi, breaks =100)
hist(filtered_Put_Cohort2$nUmi, breaks =100)
```
```{r}
min(filtered_CaH_Cohort2$nUmi)
min(filtered_Put_Cohort2$nUmi)
```

```{r}
CaH_Cohort2@meta.data 
```


```{r}
CaH_Cohort2
filtered_CaH_Cohort2 <- subset(CaH_Cohort2, subset = pct_mito < 10)
filtered_CaH_Cohort2
filtered_CaH_Cohort2 = subset(filtered_CaH_Cohort2, subset = pct_intronic >= 0.25)
filtered_CaH_Cohort2
filtered_CaH_Cohort2 = subset(filtered_CaH_Cohort2, subset = nUmi > 500)
filtered_CaH_Cohort2
filtered_CaH_Cohort2 = subset(filtered_CaH_Cohort2, subset = prob_max >= 0.9)
filtered_CaH_Cohort2
```

```{r}
Put_Cohort2
filtered_Put_Cohort2 <- subset(Put_Cohort2, subset = pct_mito < 10)
filtered_Put_Cohort2
filtered_Put_Cohort2 = subset(filtered_Put_Cohort2, subset = pct_intronic >= 0.25)
filtered_Put_Cohort2
filtered_Put_Cohort2 = subset(filtered_Put_Cohort2, subset = nUmi > 500)
filtered_Put_Cohort2
filtered_Put_Cohort2 = subset(filtered_Put_Cohort2, subset = prob_max >= 0.9)
filtered_Put_Cohort2
```

```{r}
options(future.globals.maxSize = 400 * 1024^3)  
library(sctransform)
filtered_CaH_Cohort2_sct = SCTransform(filtered_CaH_Cohort2, vars.to.regress = "pct_mito", verbose = FALSE)
DefaultAssay(filtered_CaH_Cohort2_sct) = "SCT"
filtered_CaH_Cohort2_sct
```

```{r}
filtered_Put_Cohort2_sct = SCTransform(filtered_Put_Cohort2, vars.to.regress = "pct_mito", verbose = FALSE)
DefaultAssay(filtered_Put_Cohort2_sct) = "SCT"
filtered_Put_Cohort2_sct
```
```{r}
1+1
```

#start here kk
```{r}
# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(filtered_CaH_Cohort2_sct, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
filtered_CaH_Cohort2_sct = (filtered_CaH_Cohort2_sct
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_CaH_Cohort2$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_CaH_Cohort2$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(filtered_CaH_Cohort2_sct, group.by = "donor_id") 
DimPlot(filtered_CaH_Cohort2_sct, group.by = "SCT_snn_res.0.2", label=T) 
DimPlot(filtered_CaH_Cohort2_sct, group.by = "SCT_snn_res.0.3", label=T) 
DimPlot(filtered_CaH_Cohort2_sct, group.by = "SCT_snn_res.0.4", label=T) 
DimPlot(filtered_CaH_Cohort2_sct, group.by = "SCT_snn_res.0.5", label=T) 
DimPlot(filtered_CaH_Cohort2_sct, group.by = "SCT_snn_res.0.6", label=T) 
DimPlot(filtered_CaH_Cohort2_sct, group.by = "SCT_snn_res.0.7", label=T) 
DimPlot(filtered_CaH_Cohort2_sct, group.by = "SCT_snn_res.0.8", label=T)

```

```{r}
 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN", "GAPDH","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = filtered_CaH_Cohort2_sct, features = marker_genes, group.by = "SCT_snn_res.0.2")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```

```{r}
Idents(filtered_CaH_Cohort2_sct) = filtered_CaH_Cohort2_sct$SCT_snn_res.0.2

classes = c("oligo", "oligo", "neuron", "astrocyte", "astrocyte", "microglia", "neuron", "opc", "neuron", "neuron", "neuron", "neuron", "neuron", "oligo", "ependymal", "endothelial", "neuron", "neuron", "astrocyte", "neuron")



filtered_CaH_Cohort2_sct= assignCellClasses(filtered_CaH_Cohort2_sct, classes=classes, cluster_col="SCT_snn_res.0.2", class_col = "sct_cell_class")

DimPlot(filtered_CaH_Cohort2_sct, group.by = "sct_cell_class" ,label = T, raster = FALSE)

```

```{r}
FeaturePlot(filtered_CaH_Cohort2_sct, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(filtered_CaH_Cohort2_sct, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(filtered_CaH_Cohort2_sct, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(filtered_CaH_Cohort2_sct, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(filtered_CaH_Cohort2_sct, features = c("OLIG1", "MOG", "MOBP"), raster = F)
FeaturePlot(filtered_CaH_Cohort2_sct, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(filtered_CaH_Cohort2_sct, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(filtered_CaH_Cohort2_sct, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(filtered_CaH_Cohort2_sct, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
```

```{r}
DimPlot(filtered_CaH_Cohort2_sct, group.by = "sct_cell_class",label = T, raster = FALSE)
DimPlot(filtered_CaH_Cohort2_sct, group.by = "donor_id", raster = FALSE)
DimPlot(filtered_CaH_Cohort2_sct, group.by = "Condition" ,label = T, raster = FALSE)
FeaturePlot(filtered_CaH_Cohort2_sct, features = c("nUmi", "pct_intronic", "pct_mito"),raster=FALSE) 
```

```{r}
filtered_Put_Cohort2_sct
```


```{r}
# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(filtered_Put_Cohort2_sct, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
filtered_Put_Cohort2_sct = (filtered_Put_Cohort2_sct
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_CaH_Cohort2$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_CaH_Cohort2$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(filtered_Put_Cohort2_sct, group.by = "donor_id") 
DimPlot(filtered_Put_Cohort2_sct, group.by = "SCT_snn_res.0.2", label=T) 
DimPlot(filtered_Put_Cohort2_sct, group.by = "SCT_snn_res.0.3", label=T) 
DimPlot(filtered_Put_Cohort2_sct, group.by = "SCT_snn_res.0.4", label=T) 
DimPlot(filtered_Put_Cohort2_sct, group.by = "SCT_snn_res.0.5", label=T) 
DimPlot(filtered_Put_Cohort2_sct, group.by = "SCT_snn_res.0.6", label=T) 
DimPlot(filtered_Put_Cohort2_sct, group.by = "SCT_snn_res.0.7", label=T) 
DimPlot(filtered_Put_Cohort2_sct, group.by = "SCT_snn_res.0.8", label=T)

```

```{r}
 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN", "GAPDH","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = filtered_Put_Cohort2_sct, features = marker_genes, group.by = "SCT_snn_res.0.2")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```

```{r}
Idents(filtered_Put_Cohort2_sct) = filtered_Put_Cohort2_sct$SCT_snn_res.0.2

classes = c("oligo", "oligo", "oligo", "oligo", "astrocyte", "oligo", "neuron", "neuron", "oligo", "microglia", "opc", "neuron", "neuron", "neuron", "neuron", "astrocyte", "neuron", "neuron", "endothelial", "neuron", "neuron")


filtered_Put_Cohort2_sct= assignCellClasses(filtered_Put_Cohort2_sct, classes=classes, cluster_col="SCT_snn_res.0.2", class_col = "sct_cell_class")

DimPlot(filtered_Put_Cohort2_sct, group.by = "sct_cell_class" ,label = T, raster = FALSE)

```


```{r}
FeaturePlot(filtered_Put_Cohort2_sct, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(filtered_Put_Cohort2_sct, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(filtered_Put_Cohort2_sct, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(filtered_Put_Cohort2_sct, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(filtered_Put_Cohort2_sct, features = c("OLIG1", "MOG", "MOBP"), raster = F)
FeaturePlot(filtered_Put_Cohort2_sct, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(filtered_Put_Cohort2_sct, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(filtered_Put_Cohort2_sct, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(filtered_Put_Cohort2_sct, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
```

```{r}
DimPlot(filtered_Put_Cohort2_sct, group.by = "sct_cell_class",label = T, raster = FALSE)
DimPlot(filtered_Put_Cohort2_sct, group.by = "donor_id", raster = FALSE)
DimPlot(filtered_Put_Cohort2_sct, group.by = "Condition" ,label = T, raster = FALSE)
FeaturePlot(filtered_Put_Cohort2_sct, features = c("nUmi", "pct_intronic", "pct_mito"),raster=FALSE) 
```

#QC from filtered cells
```{r}
#histogram for pct_intronic
median_pct_intronic = median(filtered_CaH_Cohort2_sct$pct_intronic *100)
title = paste0("filtered_CaH_Cohort2_sct: pct intronic, " , "Median: ", sprintf("%0.3f", median_pct_intronic), "%")
pct_intronic_hist = hist(sobj$pct_intronic*100, main = title, xlab = "pct intronic", col="grey", breaks = 100)

#histogram for nUMI --> take log10 of nUMI tonormalize
log_nUMI = log10(filtered_CaH_Cohort2_sct$nUmi)
median_log_nUMI = median(log_nUMI)
title = paste0("filtered_CaH_Cohort2_sct: log10(nUMI), ", "Median: ", sprintf("%0.3f", median_log_nUMI))
log_nUMI_hist = hist(log_nUMI, main= title, xlab = "log10(nUMI)", col="grey", breaks = 100)

saturation <- filtered_CaH_Cohort2_sct$nRead/filtered_CaH_Cohort2_sct$nUmi
median_umi_per_read = median(saturation)
title = paste0("filtered_CaH_Cohort2_sct: nReads/nUMI, ", "Median: ", sprintf("%0.3f", median_umi_per_read))
saturation_hist = hist(saturation, main= title, xlab = "nRead/nUMI", col="grey", breaks = 100)

median_pct_mito = median(filtered_CaH_Cohort2_sct$pct_mito)
title = paste0("filtered_CaH_Cohort2_sct: pct mito, " , "Median: ", sprintf("%0.3f", median_pct_mito), "%")
pct_intronic_hist = hist(sobj$pct_mito, main = title, xlab = "pct mito", col="grey", breaks = 100)
```

```{r}
#histogram for pct_intronic
median_pct_intronic = median(filtered_Put_Cohort2_sct$pct_intronic*100)
title = paste0("filtered_Put_Cohort2_sct: pct intronic, " , "Median: ", sprintf("%0.3f", median_pct_intronic),"%")
pct_intronic_hist = hist(sobj$pct_intronic*100, main = title, xlab = "pct intronic", col="grey", breaks = 100)

#histogram for nUMI --> take log10 of nUMI tonormalize
log_nUMI = log10(filtered_Put_Cohort2_sct$nUmi)
median_log_nUMI = median(log_nUMI)
title = paste0("filtered_Put_Cohort2_sct: log10(nUMI)", "Median: ", sprintf("%0.3f", median_log_nUMI))
log_nUMI_hist = hist(log_nUMI, main= title, xlab = "log10(nUMI)", col="grey", breaks = 100)

saturation <- filtered_Put_Cohort2_sct$nRead/filtered_Put_Cohort2_sct$nUmi
median_umi_per_read = median(saturation)
title = paste0("filtered_Put_Cohort2_sct: nReads/nUMI, ", "Median: ", sprintf("%0.3f", median_umi_per_read))
saturation_hist = hist(saturation, main= title, xlab = "nRead/nUMI", col="grey", breaks = 100)

median_pct_mito = median(filtered_Put_Cohort2_sct$pct_mito)
title = paste0("filtered_Put_Cohort2_sct: pct mito, " , "Median: ", sprintf("%0.3f", median_pct_mito),"%")
pct_intronic_hist = hist(sobj$pct_mito, main = title, xlab = "pct mito", col="grey", breaks = 100)
```

#Compare sct clustering vs normal
```{r}
filtered_CaH_Cohort2@meta.data
filtered_CaH_Cohort2_sct@meta.data


filtered_Put_Cohort2@meta.data
filtered_Put_Cohort2_sct@meta.data
```


```{r}
a = filtered_CaH_Cohort2@meta.data %>% select(library, cell_class) 
a$name = rownames(a)
a
b =filtered_CaH_Cohort2_sct@meta.data %>% select(library, sct_cell_class) 
b$name = rownames(b)
b

c = merge(a,b, by = "name")
c

table(c$cell_class, c$sct_cell_class)


library(ggplot2)

# Create a contingency table (confusion matrix)
cluster_table <- as.data.frame(table(c$cell_class, c$sct_cell_class))

# Plot heatmap with labels
ggplot(cluster_table, aes(x = Var1, y = Var2, fill = Freq)) +
  geom_tile(color = "grey") +  # Add grid lines for clarity
  geom_text(aes(label = Freq), color = "black", size = 5) +  # Add count labels
  scale_fill_gradient(low = "white", high = "blue") +
  labs(x = "Original Clustering", y = "SCTransform Clustering", fill = "Count") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +ggtitle("Caudate") # Rotate x-axis labels if needed



a = filtered_Put_Cohort2@meta.data %>% select(library, cell_class) 
a$name = rownames(a)
a
b =filtered_Put_Cohort2_sct@meta.data %>% select(library, sct_cell_class) 
b$name = rownames(b)
b

c = merge(a,b, by = "name")
c

table(c$cell_class, c$sct_cell_class)


library(ggplot2)

# Create a contingency table (confusion matrix)
cluster_table <- as.data.frame(table(c$cell_class, c$sct_cell_class))

# Plot heatmap with labels
ggplot(cluster_table, aes(x = Var1, y = Var2, fill = Freq)) +
  geom_tile(color = "grey") +  # Add grid lines for clarity
  geom_text(aes(label = Freq), color = "black", size = 5) +  # Add count labels
  scale_fill_gradient(low = "white", high = "blue") +
  labs(x = "Original Clustering", y = "SCTransform Clustering", fill = "Count") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +ggtitle("Putamen")  # Rotate x-axis labels if needed


```






#SCTransform on individual cell classes to identify doublets
#Caudate cleaning + integration


```{r}
table(filtered_CaH_Cohort2_sct$sct_cell_class)
```


```{r}
temp_sobj = subset(filtered_CaH_Cohort2_sct, subset = sct_cell_class == "neuron")
temp_sobj = SCTransform(temp_sobj, vars.to.regress = "pct_mito", verbose = FALSE)
DefaultAssay(temp_sobj) = "SCT"
temp_sobj

# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(temp_sobj, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
temp_sobj = (temp_sobj
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(temp_sobj, group.by = "donor_id") 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.2", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.3", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.4", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.5", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.6", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.7", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.8", label=T) 

FeaturePlot(temp_sobj, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(temp_sobj, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(temp_sobj, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(temp_sobj, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(temp_sobj, features = c("OLIG1", "MOG", "MOBP", "MBP"), raster = F)
FeaturePlot(temp_sobj, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(temp_sobj, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(temp_sobj, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(temp_sobj, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 

```


```{r}
marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7", "VCAN")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "SCT_snn_res.0.4")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)


 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "SCT_snn_res.0.4")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```
```{r}
table(temp_sobj$SCT_snn_res.0.4) 
```


```{r}
temp_sobj
temp_sobj = subset(temp_sobj, subset = SCT_snn_res.0.4 != "4" &  SCT_snn_res.0.4 != "6" & SCT_snn_res.0.4 != "7")
temp_sobj
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.4")
```
```{r}
 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "SCT_snn_res.0.4")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```
```{r}
CaH_astro_sct
CaH_endo_sct
CaH_ependymal_sct
CaH_micro_sct
#CaH_micro_sct$sct_cell_class[CaH_micro_sct$SCT_snn_res.0.4 == "7"] = "immune" 
CaH_oligo_sct

CaH_opc_sct

CaH_neuron_sct
#= temp_sobj


```





```{r}
FeaturePlot(temp_sobj, features = c("SYT1", "AQP4", "C1QA","FLT1"),raster=FALSE)
FeaturePlot(temp_sobj, features = c("OLIG1", "OLIG2", "MOG" ,"CD96"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("MBP", "MOBP", "TF", "ST18"),raster=FALSE)
FeaturePlot(temp_sobj, features = c("RBFOX3", "CX3CR1", "GFAP" ,"AQP4"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("GAD1", "GAD2", "SLC17A7" ,"SLC17A6"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("DRD1", "DRD2", "CASZ1" ,"PPP1R1B"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("EPHA4", "SEMA3E", "SLC17A7" ,"SLC17A6"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("P2RY12", "CX3CR1", "C1QB", "BCAS1"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SEMA5B", "KREMEN1", "PDE1C"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("TAC1", "NNAT", "OPRM1", "ID4"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SGK1", "EPHA4", "SV2B"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SST", "CALB1", "PVALB", "LAMP5"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("pct_mito"),raster=FALSE) 

```



```{r}
Idents(temp_sobj) = temp_sobj$SCT_snn_res.0.4

classes = c("D1_matrix", "D2_matrix", "D1_patch", "D2_patch", "doublet", "non_SPN", "non_SPN", "eSPN", "doublet", "non_SPN", "unsure", "doublet", "doublet", "non_SPN", "doublet", "doublet", "non_SPN", "doublet", "unsure", "doublet")

temp_sobj= assignCellClasses(temp_sobj, classes=classes, cluster_col="SCT_snn_res.0.4", class_col = "neuron_cell_class")

DimPlot(temp_sobj, group.by = "neuron_cell_class" ,label = T, raster = FALSE)
```

```{r}
marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)

marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7", "VCAN")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)


 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```

```{r}
temp_sobj
temp_sobj = subset(temp_sobj, subset = neuron_cell_class != "doublet")
temp_sobj
DimPlot(temp_sobj, group.by = "neuron_cell_class")

```

```{r}
1+1
```

```{r}
CaH_astro_sct@meta.data 
CaH_endo_sct@meta.data 
CaH_ependymal_sct@meta.data 
CaH_micro_sct@meta.data 
CaH_oligo_sct@meta.data 
CaH_opc_sct@meta.data 
CaH_neuron_sct@meta.data 
```

```{r}
CaH_astro_sct@meta.data$neuron_cell_class = CaH_astro_sct$sct_cell_class
CaH_endo_sct@meta.data$neuron_cell_class = CaH_endo_sct$sct_cell_class
CaH_ependymal_sct@meta.data$neuron_cell_class = CaH_ependymal_sct$sct_cell_class
CaH_micro_sct@meta.data$neuron_cell_class = CaH_micro_sct$sct_cell_class

CaH_neuron_sct

CaH_oligo_sct@meta.data$neuron_cell_class = CaH_oligo_sct$sct_cell_class
CaH_opc_sct@meta.data$neuron_cell_class = CaH_opc_sct$sct_cell_class
```
```{r}
sobj_list = list(CaH_astro_sct, 
CaH_endo_sct,
CaH_ependymal_sct,
CaH_micro_sct,
CaH_oligo_sct,
CaH_opc_sct,
CaH_neuron_sct)

qsave(sobj_list, "cah_sct_sobj_list.qs")
```

```{r}
sobj_list = qread("cah_sct_sobj_list.qs")
sobj_list
```

```{r}
CaH_astro_sct = sobj_list[[1]]
CaH_astro_sct 
DefaultAssay(CaH_astro_sct) = "RNA"
```


```{r}
CaH_endo_sct = sobj_list[[2]]
DefaultAssay(CaH_endo_sct) = "RNA"

CaH_ependymal_sct = sobj_list[[3]]
DefaultAssay(CaH_ependymal_sct) = "RNA"

CaH_micro_sct = sobj_list[[4]]
DefaultAssay(CaH_micro_sct) = "RNA"

CaH_oligo_sct = sobj_list[[5]]
DefaultAssay(CaH_oligo_sct) = "RNA"

CaH_opc_sct = sobj_list[[6]]
DefaultAssay(CaH_opc_sct) = "RNA"

CaH_neuron_sct = sobj_list[[7]]
DefaultAssay(CaH_neuron_sct) = "RNA"
```


```{r}
sobj_list = list(CaH_astro_sct, 
CaH_endo_sct,
CaH_ependymal_sct,
CaH_micro_sct,
CaH_oligo_sct,
CaH_opc_sct,
CaH_neuron_sct)
```



```{r}
counts_list <- lapply(sobj_list, function(sobj) GetAssayData(sobj, slot = "counts"))

merged_counts <- do.call(cbind, counts_list)

metadata_list <- lapply(sobj_list, function(sobj) sobj@meta.data)

merged_metadata <- do.call(rbind, metadata_list)
merged_metadata

stopifnot(all(rownames(merged_metadata) == colnames(merged_counts)))

# Create Seurat object
sct_cleaned_CaH_Cohort2 <- CreateSeuratObject(counts = merged_counts, meta.data = merged_metadata)
sct_cleaned_CaH_Cohort2
```


```{r}
unique(sct_cleaned_CaH_Cohort2@meta.data$donor_id)
```


```{r}
#sct_cleaned_CaH_Cohort2 = SCTransform(sct_cleaned_CaH_Cohort2, vars.to.regress = "pct_mito", verbose = FALSE)
#DefaultAssay(sct_cleaned_CaH_Cohort2) = "SCT"

DefaultAssay(sct_cleaned_CaH_Cohort2) = "RNA"


# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(sct_cleaned_CaH_Cohort2, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
sct_cleaned_CaH_Cohort2 = (sct_cleaned_CaH_Cohort2
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(sct_cleaned_CaH_Cohort2, group.by = "donor_id") 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "sct_cell_class", label=T)
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "neuron_cell_class", label=T)
```


```{r}
library(harmony)
sct_cleaned_CaH_Cohort2 = (sct_cleaned_CaH_Cohort2
    %>% RunHarmony(
        group.by = "donor_id")
    %>% FindNeighbors(reduction='harmony', dims=1:20)
    %>% FindClusters(res=0.05)
    %>% FindClusters(res=0.1)
    %>% FindClusters(res=0.2)
    %>% FindClusters(res=0.3)
    %>% FindClusters(res=0.4)
    %>% FindClusters(res=0.5)
    %>% FindClusters(res=0.6)
    %>% FindClusters(res=0.7)
    %>% FindClusters(res=0.8)    
    %>% RunUMAP(reduction="harmony", dims=1:20)
)
sct_cleaned_CaH_Cohort2
sct_cleaned_CaH_Cohort2@meta.data
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "donor_id") 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.05", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.1", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "sct_cell_class", label=T)
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "neuron_cell_class", label=T)
```



```{r}
FeaturePlot(sct_cleaned_CaH_Cohort2, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(sct_cleaned_CaH_Cohort2, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(sct_cleaned_CaH_Cohort2, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(sct_cleaned_CaH_Cohort2, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(sct_cleaned_CaH_Cohort2, features = c("OLIG1", "MOG", "MOBP", "MBP"), raster = F)
FeaturePlot(sct_cleaned_CaH_Cohort2, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(sct_cleaned_CaH_Cohort2, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(sct_cleaned_CaH_Cohort2, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(sct_cleaned_CaH_Cohort2, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
FeaturePlot(sct_cleaned_CaH_Cohort2, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 

```

```{r}
marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7")

Dotplot = DotPlot(object = sct_cleaned_CaH_Cohort2, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)

marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7", "VCAN")

Dotplot = DotPlot(object = sct_cleaned_CaH_Cohort2, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)


 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = sct_cleaned_CaH_Cohort2, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```


```{r}
Idents(sct_cleaned_CaH_Cohort2) = sct_cleaned_CaH_Cohort2$RNA_snn_res.0.2

classes = c("oligo", "astrocyte", "neuron", "neuron", "microglia", "astrocyte", "oligo", "opc", "neuron", "neuron", "neuron", "ependymal", "neuron", "endothelial", "neuron", "immune", "neuron")


sct_cleaned_CaH_Cohort2= assignCellClasses(sct_cleaned_CaH_Cohort2, classes=classes, cluster_col="RNA_snn_res.0.2", class_col = "final_cell_class_RNA")

DimPlot(sct_cleaned_CaH_Cohort2, group.by = "final_cell_class_RNA" ,label = T, raster = FALSE)

```

```{r}
sct_cleaned_CaH_Cohort2@meta.data$final_cell_class = sct_cleaned_CaH_Cohort2$final_cell_class_RNA
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "donor_id") 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "Condition") 
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "sct_cell_class", label=T)
DimPlot(sct_cleaned_CaH_Cohort2, group.by = "final_cell_class", label=T)
```

```{r}
cells_per_donor = as.data.frame(table(sct_cleaned_CaH_Cohort2@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(sct_cleaned_CaH_Cohort2@meta.data$final_cell_class, sct_cleaned_CaH_Cohort2@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df


cell_class_df

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_23_072CF", "SCF_22_065CM",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  
            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = x_order)

ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors from Caudate Village") + ylab("Number of cells by cell type") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors from Caudate Village") + ylab("Cell Type Proportion") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  
#+ facet_wrap(~ Var1)
```


```{r}
library(dplyr)
co2_metadata = read.csv("/broad/macosko/kimkathl/Cohort2/co2_metadata.csv", header = T)
co2_metadata

# Ensure donor_id is a character
sct_cleaned_CaH_Cohort2$donor_id <- as.character(sct_cleaned_CaH_Cohort2$donor_id)
co2_metadata$Donor <- as.character(co2_metadata$Donor)

# Join without breaking rownames
new_meta <- left_join(sct_cleaned_CaH_Cohort2@meta.data, co2_metadata, 
                      by = c("donor_id" = "Donor"))

# Assign back without breaking rownames
rownames(new_meta) <- rownames(sct_cleaned_CaH_Cohort2@meta.data)
sct_cleaned_CaH_Cohort2@meta.data <- new_meta
sct_cleaned_CaH_Cohort2@meta.data
```


```{r}
library(lme4)
sobj = sct_cleaned_CaH_Cohort2
sobj = subset(sobj, donor_id != "SCF_22_049CCF")
model_cols = c("donor_id", "final_cell_class", "Condition", "Gender", "Age.at.Death") #include any other covariates of interest, replace cell_class with whatever column defines the cluster annotations
pd = sobj@meta.data[,model_cols]
pd = pd[complete.cases(pd),] # don't want any NAs
pd$case_control_factor = as.factor(pd$Condition)

masc_df = .sconline.MASCfn(
    dataset=pd,
    cluster=pd$final_cell_class, # cluster annotations
    contrast="case_control_factor", # name of contrast annotations (what you want to run the test for)
    random_effects="donor_id", # name of random effects annotations (not interested in these coefficients, but account for variability in probable sources ob batch effects, in this case donor
    fixed_effects = c("Age.at.Death", "Gender") # your covariates
)

masc_df

my_order <- unique(masc_df$cluster)
masc_df$cluster_name <- sub("cluster", "", masc_df$cluster) # for legibility
masc_df$log2_or = log2(masc_df$case_control_factorXDP.OR) # the names of the case_control_factor<whatever> columns will change from project to project depending on the unique values of your Condition column
masc_df$log2_or_ci_low = log2(masc_df$case_control_factorXDP.OR.95pct.ci.lower)
masc_df$log2_or_ci_high = log2(masc_df$case_control_factorXDP.OR.95pct.ci.upper)

# order the graph to put disease-enriched populations on top
masc_df = masc_df[order(-masc_df$log2_or), ] 
masc_df$cluster_name = factor(masc_df$cluster_name, levels = masc_df$cluster_name[order(masc_df$log2_or)])
masc_df


library(RColorBrewer)
library(ggplot2)

# Create the forest plot with ticks on error bars, axis lines with ticks, RdBu color map, and opaque white circles on top
a = ggplot(masc_df, aes(y = cluster_name, x = log2_or)) +
  ggtitle("XDP vs. Control: Caudate Cohort 2 (no CCF)") +
  geom_vline(xintercept = 0, linetype = "dotted", color = "gray") +  # Add dotted vertical line at x=0
  geom_segment(aes(x = log2_or_ci_low, xend = log2_or_ci_high, y = cluster_name, yend = cluster_name, color = log2_or), size = 1) +  # Add horizontal error bars
  geom_point(size = 3, aes(color = log2_or), shape = 1) +  # Add points for effect sizes
  geom_point(size = 3, shape = 21, fill = "white") +  # Add opaque white circle on top of the error bar line
  scale_color_gradientn(colors = RColorBrewer::brewer.pal(10, "RdBu")) +  # Use RdBu color map
  theme_minimal() +  # Minimal theme
  labs(x = "log2(OR)", y = "Cell Classes") +  # Axis labels
  theme(
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), 
    legend.position = "none",
    plot.title = element_text(size=16),
    axis.line = element_line(color = "black"),  # Add axis lines
    axis.ticks = element_line(color = "black"),  # Add axis ticks
    axis.text = element_text(size = 14),  # Increase tick label font size
    axis.title = element_text(size = 15)  # Increase axis label font size
  )

ggsave(a, filename = "pic.png", width = 7, height = 8)
```


```{r}
qsave(sct_cleaned_CaH_Cohort2, "Cohort2/sobjs/sct_cleaned_CaH_Cohort2_031725.qs")
```



```{r}
#totalcells
cells_per_library = as.data.frame(table(sct_cleaned_CaH_Cohort2@meta.data$donor_id))
names(cells_per_library)[names(cells_per_library) == "Var1"] ="library"
names(cells_per_library)[names(cells_per_library) == "Freq"] ="total_cells"
cells_per_library  

#median/mean nUMI
median_nUmi = sct_cleaned_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi


#median/mean nGenes
median_nGenes = sct_cleaned_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_nGenes = median(nFeature_RNA, na.rm = TRUE))
median_nGenes


#median/mean reads/numi
median_reads_per_umi = sct_cleaned_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_reads_per_umi = median((nRead/nUmi), na.rm = TRUE))
median_reads_per_umi


#median/mean pctintronic
median_pct_intronic = sct_cleaned_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_intronic = median(pct_intronic * 100, na.rm = TRUE))
median_pct_intronic

#median/mean pctmito
median_pct_mito = sct_cleaned_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))
median_pct_mito


```

```{r}
library(purrr)
# List of data frames to merge
df_list <- list(median_nUmi, median_nGenes,median_reads_per_umi, median_pct_intronic, median_pct_mito )

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "donor_id"))

# Print the merged data frame
print(merged_df)

```



```{r}
qc_graphs(merged_df, merged_df$median_UMI, "Donors", "median UMI", "CaH Village: median UMI")
qc_graphs(merged_df, merged_df$median_nGenes, "Donors", "median nGenes", "CaH Village: median nGenes")
qc_graphs(merged_df, merged_df$median_reads_per_umi, "Donors", "median reads/UMI", "CaH Village: median reads/UMI")
qc_graphs(merged_df, merged_df$median_pct_intronic, "Donors", "median pct intronic", "CaH Village: median pct intronic")
qc_graphs(merged_df, merged_df$median_pct_mito, "Donors", "median pct mito", "CaH Village: median pct mito")
```

#SCTransform on individual cell classes to identify doublets
#Putamen cleaning + integration

```{r}
table(filtered_Put_Cohort2_sct$sct_cell_class)
```


```{r}
temp_sobj = subset(filtered_Put_Cohort2_sct, subset = sct_cell_class == "neuron")
temp_sobj = SCTransform(temp_sobj, vars.to.regress = "pct_mito", verbose = FALSE)
DefaultAssay(temp_sobj) = "SCT"
temp_sobj

# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(temp_sobj, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
temp_sobj = (temp_sobj
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(temp_sobj, group.by = "donor_id") 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.2", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.3", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.4", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.5", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.6", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.7", label=T) 
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.8", label=T) 

FeaturePlot(temp_sobj, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(temp_sobj, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(temp_sobj, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(temp_sobj, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(temp_sobj, features = c("OLIG1", "MOG", "MOBP", "MBP"), raster = F)
FeaturePlot(temp_sobj, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(temp_sobj, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(temp_sobj, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(temp_sobj, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 

```


```{r}
marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7", "VCAN")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "SCT_snn_res.0.3")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)


 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "SCT_snn_res.0.3")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```

```{r}
table(temp_sobj$SCT_snn_res.0.3) 
```


```{r}
temp_sobj
temp_sobj = subset(temp_sobj, subset = SCT_snn_res.0.3 != "2" & SCT_snn_res.0.3 != "3")
temp_sobj
DimPlot(temp_sobj, group.by = "SCT_snn_res.0.3")
```

```{r}
 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "SCT_snn_res.0.3")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```

```{r}
Put_astro_sct
Put_endo_sct
Put_micro_sct
Put_micro_sct$sct_cell_class[Put_micro_sct$SCT_snn_res.0.3 == "7"] = "immune" 
Put_oligo_sct

Put_opc_sct

Put_neuron_sct
#= temp_sobj



```





```{r}
FeaturePlot(temp_sobj, features = c("SYT1", "AQP4", "C1QA","FLT1"),raster=FALSE)
FeaturePlot(temp_sobj, features = c("OLIG1", "OLIG2", "MOG" ,"CD96"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("MBP", "MOBP", "TF", "ST18"),raster=FALSE)
FeaturePlot(temp_sobj, features = c("RBFOX3", "CX3CR1", "GFAP" ,"AQP4"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("GAD1", "GAD2", "SLC17A7" ,"SLC17A6"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("DRD1", "DRD2", "CASZ1" ,"PPP1R1B"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("EPHA4", "SEMA3E", "SLC17A7" ,"SLC17A6"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("P2RY12", "CX3CR1", "C1QB", "BCAS1"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SEMA5B", "KREMEN1", "PDE1C"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("TAC1", "NNAT", "OPRM1", "ID4"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SGK1", "EPHA4", "SV2B"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("SST", "CALB1", "PVALB", "LAMP5"),raster=FALSE) 
FeaturePlot(temp_sobj, features = c("pct_mito"),raster=FALSE) 

```



```{r}
Idents(temp_sobj) = temp_sobj$SCT_snn_res.0.4

classes = c("D1_matrix", "D2_matrix", "D2_matrix", "doublet", "D1_patch", "eSPN", "D2_patch", "non_SPN", "D2_matrix", "non_SPN", "non_SPN", "doublet", "D1_exotic", "D2_exotic_eSPN", "doublet")

temp_sobj= assignCellClasses(temp_sobj, classes=classes, cluster_col="SCT_snn_res.0.4", class_col = "neuron_cell_class")

DimPlot(temp_sobj, group.by = "neuron_cell_class" ,label = T, raster = FALSE)
```

```{r}
marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)

marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7", "VCAN")

Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)


 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = temp_sobj, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```

```{r}
temp_sobj
temp_sobj = subset(temp_sobj, subset = neuron_cell_class != "doublet")
temp_sobj
DimPlot(temp_sobj, group.by = "neuron_cell_class")

```


```{r}
Put_astro_sct@meta.data 
Put_endo_sct@meta.data 
Put_micro_sct@meta.data 
Put_oligo_sct@meta.data 
Put_opc_sct@meta.data 
Put_neuron_sct@meta.data 
```

```{r}
Put_astro_sct@meta.data$neuron_cell_class = Put_astro_sct$sct_cell_class
Put_endo_sct@meta.data$neuron_cell_class = Put_endo_sct$sct_cell_class
Put_micro_sct@meta.data$neuron_cell_class = Put_micro_sct$sct_cell_class

Put_neuron_sct

Put_oligo_sct@meta.data$neuron_cell_class = Put_oligo_sct$sct_cell_class
Put_opc_sct@meta.data$neuron_cell_class = Put_opc_sct$sct_cell_class
```

```{r}
sobj_list = list(Put_astro_sct, 
Put_endo_sct,
Put_micro_sct,
Put_oligo_sct,
Put_opc_sct,
Put_neuron_sct)

qsave(sobj_list, "put_sct_sobj_list.qs")
```

```{r}
sobj_list = qread("put_sct_sobj_list.qs")
sobj_list
```

```{r}
Put_astro_sct = sobj_list[[1]]
Put_astro_sct 
DefaultAssay(Put_astro_sct) = "RNA"

Put_endo_sct = sobj_list[[2]]
DefaultAssay(Put_endo_sct) = "RNA"

Put_micro_sct = sobj_list[[3]]
DefaultAssay(Put_micro_sct) = "RNA"

Put_oligo_sct = sobj_list[[4]]
DefaultAssay(Put_oligo_sct) = "RNA"

Put_opc_sct = sobj_list[[5]]
DefaultAssay(Put_opc_sct) = "RNA"

Put_neuron_sct = sobj_list[[6]]
DefaultAssay(Put_neuron_sct) = "RNA"
```


```{r}
sobj_list = list(Put_astro_sct, 
Put_endo_sct,
Put_micro_sct,
Put_oligo_sct,
Put_opc_sct,
Put_neuron_sct)
sobj_list
```



```{r}
counts_list <- lapply(sobj_list, function(sobj) GetAssayData(sobj, slot = "counts"))

merged_counts <- do.call(cbind, counts_list)

metadata_list <- lapply(sobj_list, function(sobj) sobj@meta.data)

merged_metadata <- do.call(rbind, metadata_list)
merged_metadata

stopifnot(all(rownames(merged_metadata) == colnames(merged_counts)))

# Create Seurat object
sct_cleaned_Put_Cohort2 <- CreateSeuratObject(counts = merged_counts, meta.data = merged_metadata)
sct_cleaned_Put_Cohort2
```


```{r}
unique(sct_cleaned_Put_Cohort2@meta.data$donor_id)
```


```{r}
#sct_cleaned_Put_Cohort2 = SCTransform(sct_cleaned_Put_Cohort2, vars.to.regress = "pct_mito", verbose = FALSE)
#DefaultAssay(sct_cleaned_Put_Cohort2) = "SCT"

DefaultAssay(sct_cleaned_Put_Cohort2) = "RNA"


# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(sct_cleaned_Put_Cohort2, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
sct_cleaned_Put_Cohort2 = (sct_cleaned_Put_Cohort2
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(sct_cleaned_Put_Cohort2, group.by = "donor_id") 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(sct_cleaned_Put_Cohort2, group.by = "sct_cell_class", label=T)
DimPlot(sct_cleaned_Put_Cohort2, group.by = "neuron_cell_class", label=T)
```


```{r}
library(harmony)
sct_cleaned_Put_Cohort2 = (sct_cleaned_Put_Cohort2
    %>% RunHarmony(
        group.by = "donor_id")
    %>% FindNeighbors(reduction='harmony', dims=1:20)
    %>% FindClusters(res=0.05)
    %>% FindClusters(res=0.1)
    %>% FindClusters(res=0.2)
    %>% FindClusters(res=0.3)
    %>% FindClusters(res=0.4)
    %>% FindClusters(res=0.5)
    %>% FindClusters(res=0.6)
    %>% FindClusters(res=0.7)
    %>% FindClusters(res=0.8)    
    %>% RunUMAP(reduction="harmony", dims=1:20)
)
sct_cleaned_Put_Cohort2
sct_cleaned_Put_Cohort2@meta.data
DimPlot(sct_cleaned_Put_Cohort2, group.by = "donor_id") 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.05", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.1", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(sct_cleaned_Put_Cohort2, group.by = "sct_cell_class", label=T)
DimPlot(sct_cleaned_Put_Cohort2, group.by = "neuron_cell_class", label=T)
```



```{r}
FeaturePlot(sct_cleaned_Put_Cohort2, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(sct_cleaned_Put_Cohort2, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(sct_cleaned_Put_Cohort2, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(sct_cleaned_Put_Cohort2, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(sct_cleaned_Put_Cohort2, features = c("OLIG1", "MOG", "MOBP", "MBP"), raster = F)
FeaturePlot(sct_cleaned_Put_Cohort2, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(sct_cleaned_Put_Cohort2, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(sct_cleaned_Put_Cohort2, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(sct_cleaned_Put_Cohort2, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
FeaturePlot(sct_cleaned_Put_Cohort2, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 

```

```{r}
marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7")

Dotplot = DotPlot(object = sct_cleaned_Put_Cohort2, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)

marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "TF", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7", "VCAN")

Dotplot = DotPlot(object = sct_cleaned_Put_Cohort2, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)


 marker_genes <- c("SYT1", "RBFOX3", "GAD2", "SLC17A6","AQP4", "GINS3", "GFAP","C1QA", "C1QB", "CX3CR1", "P2RY12","FLT1", "DCN", "RGS5", "OLIG1", "MOG", "MOBP", "OLIG2", "VCAN","ZBBX", "CFAP157", "CFAP299", "BSG","CD96", "NKG7", "SKAP1")
 Dotplot = DotPlot(object = sct_cleaned_Put_Cohort2, features = marker_genes, group.by = "neuron_cell_class")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```


```{r}
Idents(sct_cleaned_Put_Cohort2) = sct_cleaned_Put_Cohort2$RNA_snn_res.0.3

classes = c("oligo", "oligo", "oligo", "neuron", "astrocyte", "neuron", "oligo", "microglia", "opc", "neuron", "neuron", "endothelial", "neuron", "immune")


sct_cleaned_Put_Cohort2= assignCellClasses(sct_cleaned_Put_Cohort2, classes=classes, cluster_col="RNA_snn_res.0.3", class_col = "final_cell_class_RNA")

DimPlot(sct_cleaned_Put_Cohort2, group.by = "final_cell_class_RNA" ,label = T, raster = FALSE)

sct_cleaned_Put_Cohort2@meta.data$final_cell_class = sct_cleaned_Put_Cohort2$final_cell_class_RNA
```

```{r}
sct_cleaned_Put_Cohort2@meta.data
```


```{r}
DimPlot(sct_cleaned_Put_Cohort2, group.by = "donor_id") 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "Condition") 
DimPlot(sct_cleaned_Put_Cohort2, group.by = "sct_cell_class", label=T)
DimPlot(sct_cleaned_Put_Cohort2, group.by = "final_cell_class", label=T)
```

```{r}
cells_per_donor = as.data.frame(table(sct_cleaned_Put_Cohort2@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(sct_cleaned_Put_Cohort2@meta.data$final_cell_class_RNA, sct_cleaned_Put_Cohort2@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df


cell_class_df

x_order = c("SCF_23_072CF", "SCF_22_049CCF","SCF_23_079CF", "SCF_22_063CM","SCF_22_065CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF",    
            "SCF_21_032", "SCF_21_038","SCF_21_041","SCF_23_070_MMC","SCF_21_034", "SCF_21_029", "SCF_21_027")
            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = x_order)

ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors from Putamen Village") + ylab("Number of cells by cell type") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors from Putamen Village") + ylab("Cell Type Proportion") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  
#+ facet_wrap(~ Var1)
```

```{r}
sct_cleaned_Put_Cohort2@meta.data
co2_metadata = read.csv("/broad/macosko/kimkathl/Cohort2/co2_metadata.csv", header = T)
co2_metadata

library(dplyr)

# Ensure donor_id is a character
sct_cleaned_Put_Cohort2$donor_id <- as.character(sct_cleaned_Put_Cohort2$donor_id)
co2_metadata$Donor <- as.character(co2_metadata$Donor)

# Join without breaking rownames
new_meta <- left_join(sct_cleaned_Put_Cohort2@meta.data, co2_metadata, 
                      by = c("donor_id" = "Donor"))

# Assign back without breaking rownames
rownames(new_meta) <- rownames(sct_cleaned_Put_Cohort2@meta.data)
sct_cleaned_Put_Cohort2@meta.data <- new_meta
sct_cleaned_Put_Cohort2@meta.data
```




```{r}
library(lme4)
sobj = sct_cleaned_Put_Cohort2
sobj = subset(sobj, subset = donor_id != "SCF_22_049CCF")
model_cols = c("donor_id", "final_cell_class", "Condition", "Gender", "Age.at.Death") #include any other covariates of interest, replace cell_class with whatever column defines the cluster annotations
pd = sobj@meta.data[,model_cols]
pd = pd[complete.cases(pd),] # don't want any NAs
pd$case_control_factor = as.factor(pd$Condition)

masc_df = .sconline.MASCfn(
    dataset=pd,
    cluster=pd$final_cell_class, # cluster annotations
    contrast="case_control_factor", # name of contrast annotations (what you want to run the test for)
    random_effects="donor_id", # name of random effects annotations (not interested in these coefficients, but account for variability in probable sources ob batch effects, in this case donor
    fixed_effects = c("Age.at.Death", "Gender") # your covariates
)

masc_df

my_order <- unique(masc_df$cluster)
masc_df$cluster_name <- sub("cluster", "", masc_df$cluster) # for legibility
masc_df$log2_or = log2(masc_df$case_control_factorXDP.OR) # the names of the case_control_factor<whatever> columns will change from project to project depending on the unique values of your Condition column
masc_df$log2_or_ci_low = log2(masc_df$case_control_factorXDP.OR.95pct.ci.lower)
masc_df$log2_or_ci_high = log2(masc_df$case_control_factorXDP.OR.95pct.ci.upper)

# order the graph to put disease-enriched populations on top
masc_df = masc_df[order(-masc_df$log2_or), ] 
masc_df$cluster_name = factor(masc_df$cluster_name, levels = masc_df$cluster_name[order(masc_df$log2_or)])
masc_df


library(RColorBrewer)
library(ggplot2)

# Create the forest plot with ticks on error bars, axis lines with ticks, RdBu color map, and opaque white circles on top
a = ggplot(masc_df, aes(y = cluster_name, x = log2_or)) +
  ggtitle("XDP vs. Control: Putamen Cohort 2 (no CCF)") +
  geom_vline(xintercept = 0, linetype = "dotted", color = "gray") +  # Add dotted vertical line at x=0
  geom_segment(aes(x = log2_or_ci_low, xend = log2_or_ci_high, y = cluster_name, yend = cluster_name, color = log2_or), size = 1) +  # Add horizontal error bars
  geom_point(size = 3, aes(color = log2_or), shape = 1) +  # Add points for effect sizes
  geom_point(size = 3, shape = 21, fill = "white") +  # Add opaque white circle on top of the error bar line
  scale_color_gradientn(colors = RColorBrewer::brewer.pal(10, "RdBu")) +  # Use RdBu color map
  theme_minimal() +  # Minimal theme
  labs(x = "log2(OR)", y = "Cell Classes") +  # Axis labels
  theme(
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), 
    legend.position = "none",
    plot.title = element_text(size=16),
    axis.line = element_line(color = "black"),  # Add axis lines
    axis.ticks = element_line(color = "black"),  # Add axis ticks
    axis.text = element_text(size = 14),  # Increase tick label font size
    axis.title = element_text(size = 15)  # Increase axis label font size
  )

ggsave(a, filename = "pic.png", width = 7, height = 8)
```


```{r}
qsave(sct_cleaned_Put_Cohort2, "Cohort2/sobjs/sct_cleaned_Put_Cohort2_031725.qs")
```



```{r}
#totalcells
cells_per_library = as.data.frame(table(sct_cleaned_Put_Cohort2@meta.data$donor_id))
names(cells_per_library)[names(cells_per_library) == "Var1"] ="library"
names(cells_per_library)[names(cells_per_library) == "Freq"] ="total_cells"
cells_per_library  

#median/mean nUMI
median_nUmi = sct_cleaned_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi


#median/mean nGenes
median_nGenes = sct_cleaned_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_nGenes = median(nFeature_RNA, na.rm = TRUE))
median_nGenes


#median/mean reads/numi
median_reads_per_umi = sct_cleaned_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_reads_per_umi = median((nRead/nUmi), na.rm = TRUE))
median_reads_per_umi


#median/mean pctintronic
median_pct_intronic = sct_cleaned_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_intronic = median(pct_intronic *100, na.rm = TRUE))
median_pct_intronic

#median/mean pctmito
median_pct_mito = sct_cleaned_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))
median_pct_mito


```

```{r}
library(purrr)
# List of data frames to merge
df_list <- list(median_nUmi, median_nGenes,median_reads_per_umi, median_pct_intronic, median_pct_mito )

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "donor_id"))

# Print the merged data frame
print(merged_df)

```

```{r}
table(sct_cleaned_Put_Cohort2$donor_id)
```


```{r}
qc_graphs(merged_df, merged_df$median_UMI, "Donors", "median UMI", "Put Village: median UMI")
qc_graphs(merged_df, merged_df$median_nGenes, "Donors", "median nGenes", "Put Village: median nGenes")
qc_graphs(merged_df, merged_df$median_reads_per_umi, "Donors", "median reads/UMI", "Put Village: median reads/UMI")
qc_graphs(merged_df, merged_df$median_pct_intronic, "Donors", "median pct intronic", "Put Village: median pct intronic")
qc_graphs(merged_df, merged_df$median_pct_mito, "Donors", "median pct mito", "Put Village: median pct mito")
```

#merge
```{r}
sct_cleaned_CaH_Cohort2@meta.data
sct_cleaned_Put_Cohort2@meta.data
```


```{r}
# Assuming caudate_obj and putamen_obj are your Seurat objects
# Add a region label to each Seurat object before merging
sct_cleaned_CaH_Cohort2@meta.data$region <- "caudate"
sct_cleaned_Put_Cohort2@meta.data$region <- "putamen"
```

```{r}
sobj_list = list(sct_cleaned_CaH_Cohort2,
sct_cleaned_Put_Cohort2)

counts_list <- lapply(sobj_list, function(sobj) GetAssayData(sobj, slot = "counts"))

merged_counts <- do.call(cbind, counts_list)

metadata_list <- lapply(sobj_list, function(sobj) sobj@meta.data)

merged_metadata <- do.call(rbind, metadata_list)
rownames(merged_metadata) = paste0(merged_metadata$library, "__", merged_metadata$cell)
# Ensure that the row names of metadata match the colnames of merged counts
merged_metadata <- merged_metadata[match(colnames(merged_counts), rownames(merged_metadata)), ]
# Reorder merged_counts to match the cell names in the merged metadata
merged_counts <- merged_counts[, match(rownames(merged_metadata), colnames(merged_counts))]
# Check if the order of rownames and colnames match
all(rownames(merged_metadata) == colnames(merged_counts))  # Should return TRUE

stopifnot(all(rownames(merged_metadata) == colnames(merged_counts)))

# Create Seurat object
sct_cleaned_CaH_Put_Cohort2 <- CreateSeuratObject(counts = merged_counts, meta.data = merged_metadata)
sct_cleaned_CaH_Put_Cohort2@meta.data
```


```{r}
# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(sct_cleaned_CaH_Put_Cohort2, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
sct_cleaned_CaH_Put_Cohort2 = (sct_cleaned_CaH_Put_Cohort2
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "donor_id") 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "final_cell_class", label=T)
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "neuron_cell_class", label=T)
```


```{r}
library(harmony)
sct_cleaned_CaH_Put_Cohort2 = (sct_cleaned_CaH_Put_Cohort2
    %>% RunHarmony(
        group.by = "donor_id")
    %>% FindNeighbors(reduction='harmony', dims=1:20)
    %>% FindClusters(res=0.05)
    %>% FindClusters(res=0.1)
    %>% FindClusters(res=0.2)
    %>% FindClusters(res=0.3)
    %>% FindClusters(res=0.4)
    %>% FindClusters(res=0.5)
    %>% FindClusters(res=0.6)
    %>% FindClusters(res=0.7)
    %>% FindClusters(res=0.8)    
    %>% RunUMAP(reduction="harmony", dims=1:20)
)
sct_cleaned_CaH_Put_Cohort2
sct_cleaned_CaH_Put_Cohort2@meta.data
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "donor_id") 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.05", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.1", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.2", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.3", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.4", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.5", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.6", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.7", label=T) 
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "RNA_snn_res.0.8", label=T)
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "final_cell_class", label=T)
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "neuron_cell_class", label=T)
```

```{r}
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "region", label=T)
FeaturePlot(sct_cleaned_CaH_Put_Cohort2, features = c("SYT1", "RBFOX3", "GAD2", "SLC17A6"), raster = F)
FeaturePlot(sct_cleaned_CaH_Put_Cohort2, features = c("AQP4", "GINS3", "GFAP"), raster = F)
FeaturePlot(sct_cleaned_CaH_Put_Cohort2, features = c("C1QA", "C1QB", "CX3CR1", "P2RY12"), raster = F)
FeaturePlot(sct_cleaned_CaH_Put_Cohort2, features = c("FLT1", "DCN", "RGS5"), raster = F)
FeaturePlot(sct_cleaned_CaH_Put_Cohort2, features = c("OLIG1", "MOG", "MOBP", "MBP"), raster = F)
FeaturePlot(sct_cleaned_CaH_Put_Cohort2, features = c("OLIG2", "VCAN", "GAPDH"), raster = F)
FeaturePlot(sct_cleaned_CaH_Put_Cohort2, features = c("ZBBX", "CFAP157", "CFAP299", "BSG"), raster = F)
FeaturePlot(sct_cleaned_CaH_Put_Cohort2, features = c("CD96", "NKG7", "SKAP1"), raster = F)
FeaturePlot(sct_cleaned_CaH_Put_Cohort2, features = c("UBB", "GAPDH", "TUBB2A"),raster=FALSE) 
FeaturePlot(sct_cleaned_CaH_Put_Cohort2, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 

```


```{r}
Idents(sct_cleaned_CaH_Put_Cohort2) = sct_cleaned_CaH_Put_Cohort2$RNA_snn_res.0.2

classes = c("oligo", "oligo", "astrocyte", "neuron", "neuron", "microglia", "opc", "neuron", "neuron", "neuron", "neuron", "endothelial", "ependymal", "neuron", "immune")


sct_cleaned_CaH_Put_Cohort2= assignCellClasses(sct_cleaned_CaH_Put_Cohort2, classes=classes, cluster_col="RNA_snn_res.0.2", class_col = "final_cell_class_merged")

DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "final_cell_class_merged" ,label = T, raster = FALSE)

```

```{r}
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "region")
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "donor_id")
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "Condition")
DimPlot(sct_cleaned_CaH_Put_Cohort2, group.by = "final_cell_class_merged", label=T)
```

```{r}
qsave(sct_cleaned_CaH_Put_Cohort2, "Cohort2/sobjs/sct_cleaned_CaH_Put_Cohort2_031725.qs")
```

```{r}
table(sobj$donor_id)
```

```{r}
#rownames(sct_cleaned_CaH_Put_Cohort2@meta.data) = paste0(sct_cleaned_CaH_Put_Cohort2@meta.data$library, "__", sct_cleaned_CaH_Put_Cohort2@meta.data$cell)
sct_cleaned_CaH_Put_Cohort2@meta.data
```


```{r}
sct_cleaned_CaH_Put_Cohort2@meta.data
```



#MASC Combo
```{r}
library(lme4)
sobj = sct_cleaned_CaH_Put_Cohort2
sobj = subset(sobj, subset = donor_id != "SCF_22_049CCF")
sobj
model_cols = c("donor_id", "final_cell_class_merged", "Condition", "Gender", "Age.at.Death", "region") #include any other covariates of interest, replace cell_class with whatever column defines the cluster annotations
pd = sobj@meta.data[,model_cols]
pd = pd[complete.cases(pd),] # don't want any NAs
pd$case_control_factor = as.factor(pd$Condition)

masc_df = .sconline.MASCfn(
    dataset=pd,
    cluster=pd$final_cell_class, # cluster annotations
    contrast="case_control_factor", # name of contrast annotations (what you want to run the test for)
    random_effects="donor_id", # name of random effects annotations (not interested in these coefficients, but account for variability in probable sources ob batch effects, in this case donor
    fixed_effects = c("Age.at.Death", "Gender", "region") # your covariates
)

masc_df

my_order <- unique(masc_df$cluster)
masc_df$cluster_name <- sub("cluster", "", masc_df$cluster) # for legibility
masc_df$log2_or = log2(masc_df$case_control_factorXDP.OR) # the names of the case_control_factor<whatever> columns will change from project to project depending on the unique values of your Condition column
masc_df$log2_or_ci_low = log2(masc_df$case_control_factorXDP.OR.95pct.ci.lower)
masc_df$log2_or_ci_high = log2(masc_df$case_control_factorXDP.OR.95pct.ci.upper)

# order the graph to put disease-enriched populations on top
masc_df = masc_df[order(-masc_df$log2_or), ] 
masc_df$cluster_name = factor(masc_df$cluster_name, levels = masc_df$cluster_name[order(masc_df$log2_or)])
masc_df


library(RColorBrewer)
library(ggplot2)

# Create the forest plot with ticks on error bars, axis lines with ticks, RdBu color map, and opaque white circles on top
a = ggplot(masc_df, aes(y = cluster_name, x = log2_or)) +
  ggtitle("Combined Cohort 2 CaH + Put (no CCF)") +
  geom_vline(xintercept = 0, linetype = "dotted", color = "gray") +  # Add dotted vertical line at x=0
  geom_segment(aes(x = log2_or_ci_low, xend = log2_or_ci_high, y = cluster_name, yend = cluster_name, color = log2_or), size = 1) +  # Add horizontal error bars
  geom_point(size = 3, aes(color = log2_or), shape = 1) +  # Add points for effect sizes
  geom_point(size = 3, shape = 21, fill = "white") +  # Add opaque white circle on top of the error bar line
  scale_color_gradientn(colors = RColorBrewer::brewer.pal(10, "RdBu")) +  # Use RdBu color map
  theme_minimal() +  # Minimal theme
  labs(x = "log2(OR)", y = "Cell Classes") +  # Axis labels
  theme(
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), 
    legend.position = "none",
    plot.title = element_text(size=16),
    axis.line = element_line(color = "black"),  # Add axis lines
    axis.ticks = element_line(color = "black"),  # Add axis ticks
    axis.text = element_text(size = 14),  # Increase tick label font size
    axis.title = element_text(size = 15)  # Increase axis label font size
  )

ggsave(a, filename = "pic.png", width = 6, height = 8)
```





```{r}
options(future.globals.maxSize = 400 * 1024^3)  
library(sctransform)
sct_cleaned_CaH_Put_Cohort2 = SCTransform(sct_cleaned_CaH_Put_Cohort2, vars.to.regress = "pct_mito", verbose = FALSE)
DefaultAssay(sct_cleaned_CaH_Put_Cohort2) = "SCT"
sct_cleaned_CaH_Put_Cohort2


library(UCell)

NFKB_genes =  c("MARCKS", "IL23A", "NINJ1", "TNFSF9", "SIK1", "ATF3", "SERPINE1", "MYC", "HES1", "CCNL1", "CCN1", "EGR1", "EGR2", "JAG1", "EGR3", "ABCA1", "GADD45B", "GADD45A", "PLK2", "KLF10", 
           "EIF1", "EHD1", "FOSL2", "FOSL1", "GPR183", "PLPP3", "IFIT2", "ICAM1", "ZC3H12A", "IER2", 
           "IL12B", "JUNB", "IER5", "IER3", "STAT5A", "DUSP5", "EDN1", "JUN", "DUSP4", "DUSP1", 
           "DUSP2", "TSC22D1", "CCL20", "SPHK1", "LIF", "IL18", "TUBB2A", "RHOB", "VEGFA", "PTPRE", 
           "IL1A", "TLR2", "IL1B", "BHLHE40", "ID2", "CLCF1", "REL", "FJX1", "SGK1", "BTG3", "BTG2", 
           "BTG1", "SDC4", "LITAF", "AREG", "SOCS3", "PANX1", "RIPK2", "NFIL3", "SERPINB2", "GCH1", 
           "IFNGR2", "G0S2", "FOS", "SERPINB8", "F3", "SPSB1", "FOSB", "PER1", "F2RL1", "HBEGF", 
           "CD44", "TRIP10", "CDKN1A", "PTGER4", "PTGS2", "IFIH1", "NAMPT", "OLR1", "ICOSLG", 
           "PHLDA1", "ZBTB10", "TAP1", "PNRC1", "CXCL10", "CXCL11", "IL6ST", "CD69", "SQSTM1", 
           "RELA", "CSF2", "CD83", "CSF1", "PPP1R15A", "CD80", "TNC", "TNF", "TANK", "RELB", "ZFP36", 
           "CCND1", "RNF19B", "CCRL2", "DENND5A", "PHLDA2", "MAP3K8", "LDLR", "SLC16A6", "SMAD3", 
           "TGIF1", "MAP2K3", "DDX58", "TRAF1", "INHBA", "NFKB1", "NFKB2", "GEM", "NR4A3", "MAFF", 
           "RCAN1", "NR4A2", "EFNA1", "MXD1", "BIRC2", "YRDC", "BIRC3", "IL7R", "PFKFB3", "IRS2", 
           "SLC2A3", "PLAU", "SLC2A6", "SAT1", "ETS2", "NR4A1", "SNN", "PMEPA1", "TNFRSF9", "MSC", 
           "TIPARP", "LAMB3", "GFPT2", "CFLAR", "TNIP1", "IRF1", "NFKBIA", "BMP2", "IL6", "TNIP2", 
           "BCL6", "BCL3", "NFKBIE", "NFE2L2", "B4GALT1", "NFAT5", "TNFAIP8", "BCL2A1", "TNFAIP6", 
           "TNFAIP3", "CXCL2", "CXCL1", "TNFAIP2", "CXCL3", "CXCL6", "FUT4", "DRAM1", "DNAJB4", 
           "PDE4B", "PDLIM5", "MCL1", "KDM6B", "IL15RA", "PLAUR", "ATP2B1", "KLF4", "KLF2", "SOD2", 
           "KLF9", "KLF6", "ACKR3", "PTX3", "B4GALT5", "TRIB1", "CEBPB", "CEBPD", "PLEK", "KYNU", 
           "CCL5", "CCL4", "CCL2")

interferon_alpha = c("IL4R", "CD74", "SP110", "MX1", "RSAD2", "TAP1", "PSMB9", "CXCL10", "CXCL11", "IFI44L", "LAP3", "LPAR6", "IFI27", "TRAFD1", "TRIM14", "LY6E", "IFITM2", "IFITM3", "SAMD9", "CSF1", 
           "IFITM1", "UBE2L6", "SAMD9L", "RNF31", "HERC6", "LAMP3", "CCRL2", "MVB12A", "B2M", "CMTR1", 
           "GBP2", "GBP4", "EPSTI1", "BATF2", "HLA-C", "TENT5A", "TDRD7", "ISG15", "IFI44", "PARP14", 
           "MOV10", "ISG20", "PARP12", "PLSCR1", "PROCR", "SELL", "PSME2", "PSME1", "CNP", "LGALS3BP", 
           "IFI35", "IFIT2", "IFI30", "DDX60", "IFIT3", "NUB1", "HELZ2", "PNPT1", "IL15", "NCOA7", 
           "EIF2AK2", "NMI", "OAS1", "BST2", "IRF1", "ELF1", "IL7", "IRF2", "IRF9", "IRF7", "UBA7", 
           "GMPR", "ADAR", "CASP8", "TRIM5", "RIPK2", "CASP1", "STAT2", "WARS1", "PARP9", "PSMA3", 
           "TXNIP", "CMPK2", "CD47", "RTP4", "C1S", "OGFR", "TMEM140", "USP18", "OASL", "SLC25A28", 
           "IFIH1", "PSMB8", "DHX58", "TRIM25", "TRIM26", "TRIM21")



sct_cleaned_CaH_Put_Cohort2 <- AddModuleScore_UCell(sct_cleaned_CaH_Put_Cohort2,
  features = list(Score = NFKB_genes ),
  name = 'NFKB'
)

sct_cleaned_CaH_Put_Cohort2 <- AddModuleScore_UCell(sct_cleaned_CaH_Put_Cohort2,
  features = list(Score = interferon_alpha),
  name = 'interferon_alpha'
)

sct_cleaned_CaH_Put_Cohort2 <- AddModuleScore_UCell(sct_cleaned_CaH_Put_Cohort2,features = list(C_minus = C_minus), name = '_scores')
sct_cleaned_CaH_Put_Cohort2 <- AddModuleScore_UCell(sct_cleaned_CaH_Put_Cohort2,features = list(C_plus = C_plus), name = '_scores')



sct_cleaned_CaH_Put_Cohort2 <- AddModuleScore_UCell(sct_cleaned_CaH_Put_Cohort2,features = list(antigen_processing = antigen_processing_genes), name = '_scores')


qsave(sct_cleaned_CaH_Put_Cohort2, "cleaned_CaH_Put_Cohort2_sct_ucell_031725.qs")
```

```{r}
sct_cleaned_CaH_Cohort2
sct_cleaned_Put_Cohort2
sct_cleaned_CaH_Put_Cohort2
```


```{r}
sct_cleaned_CaH_Cohort2@meta.data
sct_cleaned_Put_Cohort2@meta.data
sct_cleaned_CaH_Put_Cohort2@meta.data
```
















```{r}
xdp_co1@meta.data$cause_of_death = "Other"
xdp_co1$cause_of_death[xdp_co1$donor_id == "SCF-18-003" | 
                             xdp_co1$donor_id == "SCF-18-004" | 
                             xdp_co1$donor_id == "SCF-19-009"| 
                             xdp_co1$donor_id == "SCF-19-014" | 
                             xdp_co1$donor_id == "SCF-20-025" | 
                             xdp_co1$donor_id == "SCF_21-037CM2"] = "Infection-related Death"


cleaned_CaH_Put_Cohort2@meta.data$cause_of_death = "Other"
cleaned_CaH_Put_Cohort2$cause_of_death[cleaned_CaH_Put_Cohort2$donor_id == "SCF_23_072CF"] = "Infection-related Death"

xdp_co1$cause_of_death_case_control = paste(xdp_co1$Condition, xdp_co1$cause_of_death) 
cleaned_CaH_Put_Cohort2$cause_of_death_case_control = paste(cleaned_CaH_Put_Cohort2$Condition, cleaned_CaH_Put_Cohort2$cause_of_death) 


cleaned_CaH_Put_Cohort2_meta = cleaned_CaH_Put_Cohort2@meta.data
xdp_co1_meta = xdp_co1@meta.data
cleaned_CaH_Put_Cohort2_meta
xdp_co1_meta
```
```{r}
cleaned_CaH_Put_Cohort2_meta$neuron_classes = cleaned_CaH_Put_Cohort2_meta$neuron_cell_class
```


```{r}
histograms_by_all_celltype(cleaned_CaH_Put_Cohort2_meta, "ScoreNFKB", "ScoreNFKB", 20, 10)
```

```{r}
histograms_by_donor(xdp_co1_meta, "ScoreNFKB", "astrocyte" ,"ScoreNFKB", 25, 15)
```


```{r}
library(dplyr)
library(ggplot2)

data <- xdp_co1_meta
df = subset(data, subset = final_cell_class == "neuron")
df
df_avg <- df %>%
  group_by(donor_id, cause_of_death_case_control, Condition, cause_of_death) %>%
  summarise(avg_score = mean(Scoreinterferon_alpha), .groups = 'drop')
df_avg

# Plot the average score per donor for each condition
a = ggplot(df_avg, aes(x = Condition, y = avg_score, color = Condition, shape = cause_of_death)) +
  geom_point(aes(group = donor_id), size = 3, alpha = 0.8) + scale_color_manual(values = rev(scales::hue_pal()(2)))+ # One point per donor
  labs(x = "Condition", y = "Mean interferonalpha Score", color = "Condition", shape = "Cause of Death", title = "Astrocytes")+ scale_shape_manual(values = rev(c(16, 17))) +
  theme_minimal()+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  + theme(
            plot.title = element_text(size = 20), # title font size
            axis.line = element_line(color = "black"),  # Add axis lines
            axis.ticks = element_line(color = "black"),  # Add axis ticks
            axis.text = element_text(size = 14),  # Increase tick label font size
            axis.title = element_text(size = 15),  # Increase axis label font size
            axis.text.x = element_text(angle = 45, hjust = 1)  # tilt axis labels
        )+
  theme(
    legend.text = element_text(size = 16),  # Increase legend text size
    legend.title = element_text(size = 18),  # Increase legend title size
   # legend.key.size = unit(1.5, "cm")  # Increase legend key size
  ) 

print(a)

ggsave(a, filename = "dotplot.png", width = 7, height = 6)

```


#wilcox test
```{r}
xdp_meta = xdp_co1_meta
xdp_meta = subset(xdp_meta, final_cell_class == "astrocyte")
xdp_meta

xdp_meta_cases = subset(xdp_meta, Condition == "XDP")
xdp_meta_controls = subset(xdp_meta, Condition == "Control")
xdp_meta_cases
xdp_meta_controls

XDP_scores <- xdp_meta_cases$ScoreNFKB  
XDP_Control_scores <- xdp_meta_controls$ScoreNFKB 

result1 <- wilcox.test(XDP_scores, XDP_Control_scores, alternative = "two.sided")
result2 <- wilcox.test(XDP_scores, XDP_Control_scores, alternative = "less")
result3 <- wilcox.test(XDP_scores, XDP_Control_scores, alternative = "greater")
print(result1)
print(result2)
print(result3)
```



```{r}
cleaned_CaH_Put_Cohort2_meta
xdp_co1_meta
```

```{r}
histograms_by_all_celltype = function(final_df, score_col, xlab, width_size, height_size){

#celltypes = c("astrocyte", "microglia", "oligo", "opc", "endothelial","immune", "ependymal", "non_SPN", "eSPN", "SPN_matrix", "SPN_patch", "SPN_exotic")

 celltypes = unique(final_df[["neuron_classes"]])  
plots <- list()

for (celltype in celltypes) {
  df_sub = final_df[final_df$neuron_classes == celltype, ]
   
  if (nrow(df_sub) == 0) {
    message(paste("No data for cell type:", celltype, "- skipping."))
    next
  }
  
min = min(df_sub[[score_col]])
max = max(df_sub[[score_col]])
step = (max-min)/100

limits = seq(min, max, step)
  
  a = plot_overlapping_density_histogram(df = df_sub, 
                                          hist_col = df_sub[[score_col]],
                                          fill_col = "Condition",
                                          colors = c("XDP" = "red", "Control" = "blue","BICAN_V8" = "green", "pd" = "red", "ctr" = "blue", "XDP_18_006" = "orange"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: ", celltype),
                                          xlab = xlab,
                                          fig_filename = NULL)
  
    plots[[celltype]] = a
  
}

final_plot <- plot_grid(plotlist = plots, ncol = 4)

# Display or save the final plot
print(final_plot)  # Display
ggsave("combined_plot.png", final_plot, width = width_size, height = height_size)
}
```


```{r}
histograms_by_all_celltype(cleaned_CaH_Put_Cohort2_meta, "C_minus_scores", "C_minus_scores", 20, 10)
```

```{r}
histograms_by_all_celltype(xdp_co1_meta, "C_minus_scores", "C_minus_scores", 20, 10)
```


```{r}
histograms_by_donor(xdp_co1_meta, "ScoreNFKB", "astrocyte" ,"ScoreNFKB", 25, 15)
```

```{r}
table(cleaned_CaH_Put_Cohort2_meta$neuron_cell_class)
```



















#TAF1

```{r}
FeaturePlot(cleaned_CaH_Cohort2, features = c("TAF1"))
FeaturePlot(cleaned_Put_Cohort2, features = c("TAF1"))
```

```{r}
xdp_co1 = qread("/broad/macosko/kimkathl/temp_disco/more_temp_1205/xdp_cah_put_sct_new.qs")
xdp_co1
xdp_co1@meta.data
table(xdp_co1$neuron_classes, xdp_co1$Region)
```
```{r}
DefaultAssay(xdp_co1) = "RNA"
xdp_co1
```

```{r}
cleaned_CaH_Cohort2
cleaned_Put_Cohort2
xdp_co1
```
```{r}
# Define genes of interest
genes_of_interest <- c("TAF1", "GAPDH")  # Replace GAPDH with a suitable control gene

# Fetch data from each experiment
expr_nextgem <- FetchData(xdp_co1, vars = genes_of_interest)
expr_gemx <- GetAssayData(cleaned_CaH_Cohort2, slot = "counts")[gene_of_interest, ]
expr_nextgem
expr_gemx

# Add metadata
expr_nextgem$Experiment <- "Cohort1"
expr_gemx$Experiment <- "Cohort2"

# Add donor information
expr_nextgem$Donor <- xdp_co1@meta.data$donor_id
expr_gemx$Donor <- cleaned_CaH_Cohort2@meta.data$donor_id

# Add cell type information (optional)
expr_nextgem$Cell_Type <- xdp_co1@meta.data$cell_type_column
expr_gemx$Cell_Type <- cleaned_CaH_Cohort2@meta.data$cell_type_column

# Merge both experiments into one dataframe
expr_data <- bind_rows(expr_nextgem, expr_gemx)
expr_data
# Convert to long format for ggplot
expr_long <- expr_data %>%
  pivot_longer(cols = c("TAF1", "GAPDH"), names_to = "Gene", values_to = "Expression")
expr_long

```


```{r}
library(ggplot2)

ggplot(expr_long, aes(x = Experiment, y = Expression, fill = Experiment)) +
  geom_violin(alpha = 0.5) +
  geom_jitter(aes(color = Donor), width = 0.2, size = 2) +
  facet_wrap(~Gene, scales = "free_y") +  # Separate plots for TAF1 and the control gene
  stat_summary(fun = mean, geom = "errorbar", width = 0.2, aes(ymax = ..y.., ymin = ..y..), color = "black", size = 1.2) +
  theme_minimal() +
  labs(title = "TAF1 vs Control Gene Expression: NextGem vs GemX", 
       x = "Experiment", 
       y = "Expression")
# Wilcoxon test for TAF1
taf1_test <- wilcox.test(Expression ~ Experiment, data = filter(expr_long, Gene == "TAF1"))

# Wilcoxon test for control gene
gapdh_test <- wilcox.test(Expression ~ Experiment, data = filter(expr_long, Gene == "GAPDH"))

# Print p-values
print(paste("TAF1 p-value:", taf1_test$p.value))
print(paste("Control Gene p-value:", gapdh_test$p.value))
```


```{r}
# Normalize TAF1 expression to total counts per cell
expr_nextgem$TAF1_norm <- expr_nextgem$TAF1 / rowSums(FetchData(nextgem_obj, vars = rownames(nextgem_obj)))
expr_gemx$TAF1_norm <- expr_gemx$TAF1 / rowSums(FetchData(gemx_obj, vars = rownames(gemx_obj)))

# Merge back
expr_norm_data <- bind_rows(expr_nextgem, expr_gemx)

ggplot(expr_norm_data, aes(x = Experiment, y = TAF1_norm, fill = Experiment)) +
  geom_violin(alpha = 0.5) +
  geom_jitter(aes(color = Donor), width = 0.2, size = 2) +
  stat_summary(fun = mean, geom = "errorbar", width = 0.2, aes(ymax = ..y.., ymin = ..y..), color = "black", size = 1.2) +
  theme_minimal() +
  labs(title = "Normalized TAF1 Expression: NextGem vs GemX", 
       x = "Experiment", 
       y = "TAF1 Expression (Normalized)")

```







```{r}
# Define the gene of interest
gene_of_interest <- "TAF1"

# Extract expression data
expr_data <- FetchData(xdp_co1, vars = gene_of_interest)

# Get cell class metadata
expr_data$cell_class <- xdp_co1@meta.data$neuron_classes  # Change to your actual cell class column name
expr_data$donor =  xdp_co1@meta.data$donor_id
expr_data$Condition =  xdp_co1@meta.data$Condition
expr_data$Region = xdp_co1@meta.data$Region

# Compute average expression per cell class
avg_expr <- expr_data %>%
  group_by(cell_class, donor, Condition, Region) %>%
  summarise(avg_expression = mean(.data[[gene_of_interest]], na.rm = TRUE))
avg_expr

library(ggplot2)

avg_expr_cah = subset(avg_expr, subset = Region == "Caudate")

p_values <- avg_expr_cah %>%
  group_by(cell_class) %>%
  summarise(p_value = wilcox.test(avg_expression ~ Condition)$p.value) %>%
  mutate(p_label = paste0(cell_class, "\n(p = ", signif(p_value, 3), ")"))  # Format p-value

# Merge p-values with original data
avg_expr_cah <- avg_expr_cah %>%
  left_join(p_values, by = "cell_class")

# Plot with p-values in facet labels
ggplot(avg_expr_cah, aes(x = Condition, y = avg_expression)) +
  geom_point(aes(color = Condition), size = 2, position = position_jitter(width = 0.2, height = 0)) +  # One dot per donor
  stat_summary(fun = mean, geom = "errorbar", width = 0.2, aes(ymax = ..y.., ymin = ..y..), color = "black", size = 1.2) +  # Black bar for average
  facet_wrap(~p_label, nrow = 3, ncol = 4, scales = "free_y") +  # Use new labels with p-values
  theme_minimal() +
  labs(title = "Caudate: TAF1 Expression per Cell Class", 
       x = "Condition (Control vs Case)", 
       y = "Average Expression") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

avg_expr_put = subset(avg_expr, subset = Region == "Putamen")

p_values <- avg_expr_put %>%
  group_by(cell_class) %>%
  summarise(p_value = wilcox.test(avg_expression ~ Condition)$p.value) %>%
  mutate(p_label = paste0(cell_class, "\n(p = ", signif(p_value, 3), ")"))  # Format p-value

# Merge p-values with original data
avg_expr_put <- avg_expr_put %>%
  left_join(p_values, by = "cell_class")

# Plot with p-values in facet labels
ggplot(avg_expr_put, aes(x = Condition, y = avg_expression)) +
  geom_point(aes(color = Condition), size = 2, position = position_jitter(width = 0.2, height = 0)) +  # One dot per donor
  stat_summary(fun = mean, geom = "errorbar", width = 0.2, aes(ymax = ..y.., ymin = ..y..), color = "black", size = 1.2) +  # Black bar for average
  facet_wrap(~p_label, nrow = 3, ncol = 4, scales = "free_y") +  # Use new labels with p-values
  theme_minimal() +
  labs(title = "Putamen: TAF1 Expression per Cell Class", 
       x = "Condition (Control vs Case)", 
       y = "Average Expression") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))



```










#QC by library and donor
```{r}
#histogram for pct_intronic
summary_data <- filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(
    median_pct_intronic = median(pct_intronic, na.rm = TRUE), 
    log_nUMI = median(log10(nUmi), na.rm = TRUE),  # Compute log inside summarize
    saturation = median(nRead / nUmi, na.rm = TRUE),  # Ensure division inside summarize
    median_pct_mito = median(pct_mito, na.rm = TRUE)
  )

summary_data

```

```{r}
ggplot(filtered_Put_Cohort2@meta.data, aes(x = filtered_Put_Cohort2$pct_intronic)) + facet_wrap(~ donor_id) + geom_histogram(bins = 100) 
```

```{r}
cah_meta = filtered_CaH_Cohort2@meta.data
put_meta = filtered_Put_Cohort2@meta.data 
```


```{r}
medians <- cah_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
cah_meta$donor_id <- factor(cah_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(cah_meta, aes(x = pct_intronic, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_pct_intronic, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Caudate pct intronic by donor") +ylab("Donor")



# Compute median for each donor
medians <- put_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
put_meta$donor_id <- factor(put_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(put_meta, aes(x = pct_intronic, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_pct_intronic, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Putamen pct intronic by donor") +ylab("Donor")


medians <- cah_meta %>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_log_nUMI = median(log10(nUmi), na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
cah_meta$donor_id <- factor(cah_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(cah_meta, aes(x = log10(nUmi), fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_log_nUMI, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Caudate log(UMI) by donor") +ylab("Donor")



# Compute median for each donor
medians <- put_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_log_nUMI = median(log10(nUmi), na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
put_meta$donor_id <- factor(put_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(put_meta, aes(x = log10(nUmi), fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_log_nUMI, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Putamen log(UMI) by donor") +ylab("Donor")


cah_meta$saturation <- cah_meta$nRead/cah_meta$nUmi
put_meta$saturation <- put_meta$nRead/put_meta$nUmi

medians <- cah_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_saturation = median(saturation, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
cah_meta$donor_id <- factor(cah_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)


# Plot histogram with facet per donor and median as text
ggplot(cah_meta, aes(x = saturation, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_saturation, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Caudate saturation by donor") +ylab("Donor")



# Compute median for each donor
medians <- put_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_saturation = median(saturation, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
put_meta$donor_id <- factor(put_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text

ggplot(put_meta, aes(x = saturation, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_saturation, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Putamen saturation by donor") +ylab("Donor")


medians <- cah_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
cah_meta$donor_id <- factor(cah_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(cah_meta, aes(x = pct_mito, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_pct_mito, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Caudate pct mito by donor") +ylab("Donor")



# Compute median for each donor
medians <- put_meta%>%
  group_by(donor_id, Condition) %>%  # Include case/control in grouping
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  

# Convert donor_id to a factor with a specified order in BOTH datasets
put_meta$donor_id <- factor(put_meta$donor_id, levels = x_order)
medians$donor_id <- factor(medians$donor_id, levels = x_order)

# Plot histogram with facet per donor and median as text
ggplot(put_meta, aes(x = pct_mito, fill = Condition)) + 
  facet_wrap(~ donor_id) + 
  geom_histogram(bins = 100, alpha = 0.6) +
  scale_fill_manual(values = c("Control" = "blue", "XDP" = "red")) +  # Define colors
  geom_text(data = medians, aes(x = Inf, y = Inf, 
                                label = paste0("Median: ", round(median_pct_mito, 2))), 
            hjust = 1.1, vjust = 1.1, inherit.aes = FALSE, size = 4, color = "black") +
  theme_minimal() +
  labs(fill = "Condition") + ggtitle("Putamen pct mito by donor") +ylab("Donor")

```



```{r}
median_pct_intronic = median(filtered_Put_Cohort2$pct_intronic)
title = paste0("filtered_Put_Cohort2: pct intronic, " , "Median: ", sprintf("%0.3f", median_pct_intronic))
pct_intronic_hist = hist(filtered_Put_Cohort2$pct_intronic, main = title, xlab = "pct intronic", col="grey", breaks = 100)
```

```{r}
#histogram for nUMI --> take log10 of nUMI tonormalize
log_nUMI = log10(Put_Cohort2$nUmi)
title = paste0("Put_Cohort2: log10(nUMI)")
log_nUMI_hist = hist(log_nUMI, main= title, xlab = "log10(nUMI)", col="grey", breaks = 100)
```

```{r}
saturation <- filtered_Put_Cohort2$nRead/filtered_Put_Cohort2$nUmi
median_umi_per_read = median(saturation)
title = paste0("filtered_Put_Cohort2: nReads/nUMI, " , "Median: ", sprintf("%0.3f", median_umi_per_read))  
saturation_hist = hist(saturation, main= title, xlab = "nRead/nUMI", col="grey", breaks = 100)

```

```{r}
median_pct_mito = median(filtered_Put_Cohort2$pct_mito)
title = paste0("filtered_Put_Cohort2: pct mito, " , "Median: ", sprintf("%0.3f", median_pct_mito))
pct_intronic_hist = hist(sobj$pct_mito, main = title, xlab = "pct mito", col="grey", breaks = 100)
```













```{r}
CaH_Cohort2
Put_Cohort2
```


```{r}
table(filtered_CaH_Cohort2$donor_id)
table(filtered_Put_Cohort2$donor_id)
```
```{r}
filtered_CaH_Cohort2@meta.data
```


```{r}
total_donors_cells_caudate <- as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id))
colnames(total_donors_cells_caudate) <- c("donor_id", "cell_count")

# Add condition information
total_donors_cells_caudate$Condition <- filtered_CaH_Cohort2@meta.data$Condition[match(total_donors_cells_caudate$donor_id, filtered_CaH_Cohort2@meta.data$donor_id)]

total_donors_cells_caudate$donor_id <- factor(total_donors_cells_caudate$donor_id, levels = total_donors_cells_caudate$donor_id[order(total_donors_cells_caudate$cell_count, decreasing = TRUE)])


ggplot(total_donors_cells_caudate, aes(x = donor_id, y = cell_count, fill = Condition)) + 
  geom_bar(stat = "identity") + 
  geom_text(aes(label = cell_count), vjust = -0.3, color = "black", size = 3) + 
  xlab("Donors from Caudate Village") + 
  ylab("Number of Cells") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
scale_fill_manual(values = c("XDP" = "#F8766D", "Control" = "#00BFC4"))

```

```{r}
total_donors_cells_caudate <- as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id,filtered_CaH_Cohort2@meta.data$library ))
colnames(total_donors_cells_caudate) <- c("donor_id", "library", "cell_count")
total_donors_cells_caudate
```


```{r}
# Add condition information
total_donors_cells_caudate$Condition <- filtered_CaH_Cohort2@meta.data$Condition[match(total_donors_cells_caudate$donor_id, filtered_CaH_Cohort2@meta.data$donor_id)]
total_donors_cells_caudate
```



```{r}
caudate_cell = as.data.frame(table(filtered_CaH_Cohort2@meta.data$Condition))
caudate_cell$Condition = caudate_cell$Var1
caudate_cell$Var1 = NULL
caudate_cell$Village = "Caudate"
putamen_cell = as.data.frame(table(filtered_Put_Cohort2@meta.data$Condition))
putamen_cell$Condition = putamen_cell$Var1
putamen_cell$Var1 = NULL
putamen_cell$Village = "Putamen"
cells_villages = rbind(caudate_cell, putamen_cell)

total_freq_per_village <- aggregate(Freq ~ Village, data = cells_villages, FUN = sum)
total_freq_per_village

ggplot(cells_villages, aes(x = Village, y = Freq, fill = Condition)) +
  geom_bar(stat = "identity") +  
  geom_text(aes(label = Freq), position = position_stack(vjust = 0.5), size = 3, color = "white") +  
  geom_text(data = total_freq_per_village, aes(x = Village, y = Freq, label = Freq), vjust = -0.5, size = 4, inherit.aes = FALSE) +  
  labs(title = "Cell Counts by Region and Condition", x = "Region", y = "Cell Counts", fill = "Condition") +
  theme_minimal()


```

```{r}
caudate_cell = as.data.frame(table(filtered_CaH_Cohort2@meta.data$library))
caudate_cell$library = caudate_cell$Var1
caudate_cell$Var1 = NULL
caudate_cell$Village = "Caudate"
putamen_cell = as.data.frame(table(filtered_Put_Cohort2@meta.data$library))
putamen_cell$library = putamen_cell$Var1
putamen_cell$Var1 = NULL
putamen_cell$Village = "Putamen"
cells_villages = rbind(caudate_cell, putamen_cell)

total_freq_per_village <- aggregate(Freq ~ Village, data = cells_villages, FUN = sum)
total_freq_per_village

ggplot(cells_villages, aes(x = Village, y = Freq, fill = library)) +
  geom_bar(stat = "identity") +  
  geom_text(aes(label = Freq), position = position_stack(vjust = 0.5), size = 3, color = "white") +  
  geom_text(data = total_freq_per_village, aes(x = Village, y = Freq, label = Freq), vjust = -0.5, size = 4, inherit.aes = FALSE) +  
  labs(title = "Cell Counts by Region and Library", x = "Region", y = "Cell Counts", fill = "Condition") +
  theme_minimal()
```

```{r}
unique(filtered_CaH_Cohort2@meta.data$donor_id)
```


```{r}
per_donor = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id))
per_donor

cells_per_donor_per_rxn = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id, filtered_CaH_Cohort2@meta.data$library))
cells_per_donor_per_rxn = merge(cells_per_donor_per_rxn, per_donor, by = "Var1" )
cells_per_donor_per_rxn

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  


cells_per_donor_per_rxn$Var1 = factor(cells_per_donor_per_rxn$Var1, levels = x_order)

ggplot(cells_per_donor_per_rxn, aes(x = Var1, y = Freq.x, fill = Var2)) +
  geom_bar(stat = "identity") + xlab("Donors from Caudate Village") +  geom_text(data = cells_per_donor_per_rxn, aes(x = Var1, y = Freq.y, label = Freq.y), vjust = -0.2, size = 3, nudge_y = 0.5) + ylab("Number of Cells") +
 labs(fill = "Library")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r}
per_donor = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id))
per_donor

cells_per_donor_per_rxn = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id, filtered_Put_Cohort2@meta.data$library))
cells_per_donor_per_rxn = merge(cells_per_donor_per_rxn, per_donor, by = "Var1" )
cells_per_donor_per_rxn

x_order = c("SCF_23_072CF", "SCF_22_049CCF","SCF_23_079CF", "SCF_22_063CM","SCF_22_067CM",  "SCF_23_075CM",  "SCF_23_074CF", "SCF_22_065CM",   
"SCF_21_032","SCF_21_038",  "SCF_21_041","SCF_23_070_MMC", "SCF_21_034", "SCF_21_029","SCF_21_027")  


cells_per_donor_per_rxn$Var1 = factor(cells_per_donor_per_rxn$Var1, levels = x_order)

ggplot(cells_per_donor_per_rxn, aes(x = Var1, y = Freq.x, fill = Var2)) +
  geom_bar(stat = "identity") + xlab("Donors from Putamen Village") +  geom_text(data = cells_per_donor_per_rxn, aes(x = Var1, y = Freq.y, label = Freq.y), vjust = -0.2, size = 3, nudge_y = 0.5) + ylab("Number of Cells") +
 labs(fill = "Library")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r}
DimPlot(filtered_CaH_Cohort2, group.by = "Condition")
DimPlot(filtered_CaH_Cohort2, group.by = "library")
DimPlot(filtered_CaH_Cohort2, group.by = "donor_id")

DimPlot(filtered_Put_Cohort2, group.by = "Condition")
DimPlot(filtered_Put_Cohort2, group.by = "library")
DimPlot(filtered_Put_Cohort2, group.by = "donor_id")

FeaturePlot(filtered_CaH_Cohort2, features = c("nUmi", "pct_mito"))
```












```{r}
cells_per_donor_per_rxn = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id, filtered_CaH_Cohort2@meta.data$library))
cells_per_donor_per_rxn

ggplot(cells_per_donor_per_rxn, aes(x=Var1, y = Freq, fill=Var2)) + geom_point(shape = 21, size = 3)+ xlab("Donors") + ylab("Number of Cells") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylim(0, 2500)
```


```{r}
median_genes_per_donor <-filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_genes = median(nFeature_RNA, na.rm = TRUE))

median_genes_per_donor

median_genes_per_donor <-filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id, library) %>%
  summarize(median_genes = median(nFeature_RNA, na.rm = TRUE))

median_genes_per_donor

ggplot(median_genes_per_donor, aes(x=donor_id, y = median_genes, fill=library)) + geom_point(shape = 21, size = 3)+ xlab("Donors") + ylab("Median number of genes") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylim(0,4000) +ggtitle("Putamen: Median genes per library")
```

```{r}
median_umis_per_donor <-filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id, library) %>%
  summarize(mUMI = median(nUmi, na.rm = TRUE))

median_umis_per_donor

ggplot(median_umis_per_donor, aes(x=donor_id, y = mUMI, fill=library)) + geom_point(shape = 21, size = 3)+ xlab("Donors") + ylab("Median UMIs") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) +ggtitle("Putamen: Median UMIs")
```



```{r}
total_donors_cells_caudate_per_rxn = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id, filtered_CaH_Cohort2$library))

#install.packages("forcats")
library(forcats)

# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_caudate_per_rxn %>%
  group_by(Var1) %>%
  summarize(total_Freq = sum(Freq))

# Reorder Var1 based on total_Freq
total_donors_cells_caudate_per_rxn <- total_donors_cells_caudate_per_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_Freq[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>%
  mutate(Var1 = fct_reorder(Var1, -total_Freq))

# Create the stacked bar plot
ggplot(total_donors_cells_caudate_per_rxn, aes(x = Var1, y = Freq, fill = Var2)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_Freq, label = total_Freq), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Caudate Village") +
  ylab("Number of Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```


```{r}
total_donors_cells_putamen_per_rxn = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id, filtered_Put_Cohort2$library))
total_donors_cells_putamen_per_rxn

#install.packages("forcats")
#library(forcats)

# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_putamen_per_rxn %>%
  group_by(Var1) %>%
  summarize(total_Freq = sum(Freq))

# Reorder Var1 based on total_Freq
total_donors_cells_putamen_per_rxn <- total_donors_cells_putamen_per_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_Freq[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>%
  mutate(Var1 = fct_reorder(Var1, -total_Freq))

# Create the stacked bar plot
ggplot(total_donors_cells_putamen_per_rxn, aes(x = Var1, y = Freq, fill = Var2)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_Freq, label = total_Freq), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Putamen Village") +
  ylab("Number of Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```



#calculate how many cells are being filtered out from EACH donor
```{r}
total_donors_cells_caudate_by_rxn = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id, filtered_CaH_Cohort2@meta.data$library))
total_donors_cells_caudate_by_rxn

total_donors_cells_putamen_by_rxn = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id, filtered_Put_Cohort2@meta.data$library))
total_donors_cells_putamen_by_rxn

#prefilter
prefilter_caudate_by_rxn = as.data.frame(table(CaH_Cohort2@meta.data$donor_id, CaH_Cohort2@meta.data$library))
prefilter_caudate_by_rxn = subset(prefilter_caudate_by_rxn, subset = Var1 != "SCF_21_036B")
prefilter_caudate_by_rxn

prefilter_putamen_by_rxn = as.data.frame(table(Put_Cohort2@meta.data$donor_id, Put_Cohort2@meta.data$library))
prefilter_putamen_by_rxn
```

```{r}
total_donors_cells_caudate_by_rxn$filtered_cells = prefilter_caudate_by_rxn$Freq - total_donors_cells_caudate_by_rxn$Freq
total_donors_cells_caudate_by_rxn

total_donors_cells_putamen_by_rxn$filtered_cells =  prefilter_putamen_by_rxn$Freq - total_donors_cells_putamen_by_rxn$Freq
total_donors_cells_putamen_by_rxn

```
```{r}
total_donors_cells_caudate_by_rxn$proportion = total_donors_cells_caudate_by_rxn$filtered_cells/total_donors_cells_caudate_by_rxn$Freq
total_donors_cells_caudate_by_rxn$proportion <- round(total_donors_cells_caudate_by_rxn$proportion, 2)
total_donors_cells_caudate_by_rxn
```

```{r}
total_donors_cells_caudate_by_rxn
summary_data
```


```{r}
# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_caudate_by_rxn %>%
  group_by(Var1) %>%
  summarize(total_filtered_cells = sum(filtered_cells), total_cells = sum(Freq))

summary_data$total_proportion = summary_data$total_filtered_cells/summary_data$total_cells
summary_data$total_proportion <- round(summary_data$total_proportion, 2)

summary_data

# Reorder Var1 based on total_filtered_cells
total_donors_cells_caudate_by_rxn <- total_donors_cells_caudate_by_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_proportion[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>% mutate(Var1 = fct_reorder(Var1, -total_proportion))
summary_data

# Create the stacked bar plot
ggplot(summary_data, aes(x = Var1, y = total_proportion)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_proportion, label = total_proportion), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Caudate Village") +
  ylab("Proportion of Filtered Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```
```{r}
# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_caudate_by_rxn %>%
  group_by(Var1) %>%
  summarize(total_filtered_cells = sum(filtered_cells))

summary_data

# Reorder Var1 based on total_filtered_cells
total_donors_cells_caudate_by_rxn <- total_donors_cells_caudate_by_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_filtered_cells[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>% mutate(Var1 = fct_reorder(Var1, -total_filtered_cells))
summary_data

# Create the stacked bar plot
ggplot(total_donors_cells_caudate_by_rxn, aes(x = Var1, y = filtered_cells, fill = Var2)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_filtered_cells, label = total_filtered_cells), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Caudate Village") +
  ylab("Number of Filtered Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```


```{r}
# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_putamen_by_rxn %>%
  group_by(Var1) %>%
  summarize(total_filtered_cells = sum(filtered_cells))


# Reorder Var1 based on total_filtered_cells
total_donors_cells_putamen_by_rxn <- total_donors_cells_putamen_by_rxn %>%
mutate(Var1 = fct_reorder(Var1, -summary_data$total_filtered_cells[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>% mutate(Var1 = fct_reorder(Var1, -total_filtered_cells))
summary_data

# Create the stacked bar plot
ggplot(total_donors_cells_putamen_by_rxn, aes(x = Var1, y = filtered_cells, fill = Var2)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_filtered_cells, label = total_filtered_cells), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Putamen Village") +
  ylab("Number of Filtered Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```


```{r}
total_donors_cells_putamen_by_rxn$proportion = total_donors_cells_putamen_by_rxn$filtered_cells/total_donors_cells_putamen_by_rxn$Freq
total_donors_cells_putamen_by_rxn$proportion <- round(total_donors_cells_putamen_by_rxn$proportion, 2)
total_donors_cells_putamen_by_rxn

# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_putamen_by_rxn %>%
  group_by(Var1) %>%
  summarize(total_filtered_cells = sum(filtered_cells), total_cells = sum(Freq), total_proportion = sum(proportion))

summary_data


# Reorder Var1 based on total_filtered_cells
total_donors_cells_putamen_by_rxn <- total_donors_cells_putamen_by_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_proportion[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>% mutate(Var1 = fct_reorder(Var1, -total_proportion))
summary_data

# Create the stacked bar plot
ggplot(total_donors_cells_putamen_by_rxn, aes(x = Var1, y = proportion, fill = Var2)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_proportion, label = total_proportion), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Putamen Village") +
  ylab("Proportion of Filtered Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r}
# Summarize the data to get the sum of frequencies for each Var1
summary_data <- total_donors_cells_putamen_by_rxn %>%
  group_by(Var1) %>%
  summarize(total_filtered_cells = sum(filtered_cells), total_cells = sum(Freq))

summary_data$total_proportion = summary_data$total_filtered_cells/summary_data$total_cells
summary_data$total_proportion <- round(summary_data$total_proportion, 2)

summary_data

# Reorder Var1 based on total_filtered_cells
total_donors_cells_putamen_by_rxn <- total_donors_cells_putamen_by_rxn %>%
  mutate(Var1 = fct_reorder(Var1, -summary_data$total_proportion[match(Var1, summary_data$Var1)]))

# Update summary_data to use the same reordered Var1
summary_data <- summary_data %>% mutate(Var1 = fct_reorder(Var1, -total_proportion))
summary_data

# Create the stacked bar plot
ggplot(summary_data, aes(x = Var1, y = total_proportion)) +
  geom_bar(stat = "identity") +
  geom_text(data = summary_data, aes(x = Var1, y = total_proportion, label = total_proportion), 
            vjust = -0.3, color = "black", size = 3, inherit.aes = FALSE) +
  xlab("Donors from Putamen Village") +
  ylab("Proportion of Filtered Cells") + labs(fill = "Rxn")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```


```{r}
cells_per_donor = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(filtered_CaH_Cohort2@meta.data$cell_class, filtered_CaH_Cohort2@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df


cell_class_df

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_23_075CM", "SCF_22_067CM",  "SCF_23_074CF", "SCF_22_065CM", "SCF_23_072CF",   "SCF_23_079CF",
  "SCF_21_027", "SCF_23_070_MMC", "SCF_21_038", "SCF_21_041", "SCF_21_032", "SCF_21_029", "SCF_21_034")  
            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = x_order)

ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors from Caudate Village") + ylab("Number of cells by cell type") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors from Caudate Village") + ylab("Cell Type Proportion") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  
#+ facet_wrap(~ Var1)
```
```{r}
cells_per_donor = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(filtered_Put_Cohort2@meta.data$cell_class, filtered_Put_Cohort2@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df


cell_class_df

x_order = c("SCF_23_072CF", "SCF_22_049CCF","SCF_23_079CF", "SCF_22_063CM","SCF_22_065CM", "SCF_22_067CM",  "SCF_23_075CM",  "SCF_23_074CF",   
"SCF_21_032","SCF_21_038",  "SCF_21_041","SCF_23_070_MMC", "SCF_21_034", "SCF_21_029","SCF_21_027") 
            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = x_order)

ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors from Putamen Village") + ylab("Number of cells by cell type") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors from Putamen Village") + ylab("Cell Type Proportion") + labs(fill = "Cell Type")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  
#+ facet_wrap(~ Var1)
```
```{r}
qsave(CaH_Cohort2, "Cohort2/sobjs/CaH_Cohort2_unfiltered.qs")
qsave(Put_Cohort2, "Cohort2/sobjs/Put_Cohort2_unfiltered.qs")

qsave(filtered_CaH_Cohort2, "Cohort2/sobjs/filtered_CaH_Cohort2.qs")
qsave(filtered_Put_Cohort2, "Cohort2/sobjs/filtered_Put_Cohort2.qs")
```



```{r}
median(filtered_CaH_Cohort2$pct_mito)
```


```{r}
#totalcells
cells_per_library = as.data.frame(table(filtered_CaH_Cohort2@meta.data$donor_id))
names(cells_per_library)[names(cells_per_library) == "Var1"] ="library"
names(cells_per_library)[names(cells_per_library) == "Freq"] ="total_cells"
cells_per_library  

#median/mean nUMI
median_nUmi = filtered_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi

# mean_nUmi = filtered_CaH_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_UMI = mean(nUmi, na.rm = TRUE))
# mean_nUmi

#median/mean nGenes
median_nGenes = filtered_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_nGenes = median(nFeature_RNA, na.rm = TRUE))
median_nGenes

# mean_nGenes = filtered_CaH_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_nGenes = mean(nFeature_RNA, na.rm = TRUE))
# mean_nGenes

#median/mean reads/numi
median_reads_per_umi = filtered_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_reads_per_umi = median((nRead/nUmi), na.rm = TRUE))
median_reads_per_umi

# mean_reads_per_umi = filtered_CaH_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_reads_per_umi = mean((nRead/nUmi), na.rm = TRUE))
# mean_reads_per_umi

#median/mean pctintronic
median_pct_intronic = filtered_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))
median_pct_intronic

# mean_pct_intronic = filtered_CaH_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_pct_intronic = mean(pct_intronic, na.rm = TRUE))
# mean_pct_intronic

#median/mean pctmito
median_pct_mito = filtered_CaH_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))
median_pct_mito

# mean_pct_mito = filtered_CaH_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_pct_mito = mean(pct_mito, na.rm = TRUE))
# mean_pct_mito
```
```{r}
library(gridExtra)
caudate_library_qc_df_list = list(median_nUmi, median_nGenes, median_reads_per_umi, median_pct_intronic,  median_pct_mito)
caudate_library_qc_df = reduce(caudate_library_qc_df_list, left_join, by = "donor_id")
caudate_library_qc_df

caudate_library_qc_df[, -1] <- round(caudate_library_qc_df[, -1], digits = 2)
caudate_library_qc_df
table_grob = tableGrob(caudate_library_qc_df)
ggsave("caudate_donor_qc_df.png", table_grob, width = 15, height = 6, dpi = 300)
```


```{r}
library(purrr)
# List of data frames to merge
df_list <- list(median_nUmi, median_nGenes,median_reads_per_umi, median_pct_intronic, median_pct_mito )

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "donor_id"))

# Print the merged data frame
print(merged_df)

```


```{r}
qc_graphs = function(data, y, xlab, ylab, title){

x_order = c("SCF_22_049CCF", "SCF_22_063CM", "SCF_22_065CM", "SCF_22_067CM",  
  "SCF_23_072CF", "SCF_23_074CF", "SCF_23_075CM", "SCF_23_079CF",  
  "SCF_21_027", "SCF_21_029", "SCF_21_032", "SCF_21_034",  
  "SCF_21_038", "SCF_21_041", "SCF_23_070_MMC")  

data$donor_id = factor(data$donor_id, levels = x_order)
median = median(y)

ggplot(data, aes(x = donor_id, y = y, fill = donor_id)) +
  geom_bar(stat = "identity") + xlab(xlab) + ylab(ylab)+ labs(fill = "donor_id")+ ggtitle(label = title, subtitle = paste("Median: ", sprintf("%0.2f", median)))+geom_hline(yintercept = median, linetype = "dashed", color = "black") +
  geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
}
```

```{r}
qc_graphs(merged_df, merged_df$median_UMI, "Donors", "median UMI", "CaH Village: median UMI")
qc_graphs(merged_df, merged_df$median_nGenes, "Donors", "median nGenes", "CaH Village: median nGenes")
qc_graphs(merged_df, merged_df$median_reads_per_umi, "Donors", "median reads/UMI", "CaH Village: median reads/UMI")
qc_graphs(merged_df, merged_df$median_pct_intronic, "Donors", "median pct intronic", "CaH Village: median pct intronic")
qc_graphs(merged_df, merged_df$median_pct_mito, "Donors", "median pct mito", "CaH Village: median pct mito")
```


#putamen
```{r}
#totalcells
cells_per_library = as.data.frame(table(filtered_Put_Cohort2@meta.data$donor_id))
names(cells_per_library)[names(cells_per_library) == "Var1"] ="library"
names(cells_per_library)[names(cells_per_library) == "Freq"] ="total_cells"
cells_per_library  

#median/mean nUMI
median_nUmi = filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi

# mean_nUmi = filtered_Put_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_UMI = mean(nUmi, na.rm = TRUE))
# mean_nUmi

#median/mean nGenes
median_nGenes = filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_nGenes = median(nFeature_RNA, na.rm = TRUE))
median_nGenes

# mean_nGenes = filtered_Put_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_nGenes = mean(nFeature_RNA, na.rm = TRUE))
# mean_nGenes

#median/mean reads/numi
median_reads_per_umi = filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_reads_per_umi = median((nRead/nUmi), na.rm = TRUE))
median_reads_per_umi

# mean_reads_per_umi = filtered_Put_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_reads_per_umi = mean((nRead/nUmi), na.rm = TRUE))
# mean_reads_per_umi

#median/mean pctintronic
median_pct_intronic = filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))
median_pct_intronic

# mean_pct_intronic = filtered_Put_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_pct_intronic = mean(pct_intronic, na.rm = TRUE))
# mean_pct_intronic

#median/mean pctmito
median_pct_mito = filtered_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))
median_pct_mito

# mean_pct_mito = filtered_Put_Cohort2@meta.data %>%
#   group_by(donor_id) %>%
#   summarize(mean_pct_mito = mean(pct_mito, na.rm = TRUE))
# mean_pct_mito

# List of data frames to merge
df_list <- list(median_nUmi, median_nGenes,median_reads_per_umi, median_pct_intronic, median_pct_mito )

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "donor_id"))

# Print the merged data frame
print(merged_df)
```

```{r}
putamen_library_qc_df_list = list(median_nUmi, median_nGenes, median_reads_per_umi, median_pct_intronic,  median_pct_mito)
putamen_library_qc_df = reduce(putamen_library_qc_df_list, left_join, by = "donor_id")
putamen_library_qc_df

putamen_library_qc_df[, -1] <- round(putamen_library_qc_df[, -1], digits = 2)
putamen_library_qc_df
table_grob = tableGrob(putamen_library_qc_df)
ggsave("putamen_donor_qc_df.png", table_grob, width = 15, height = 6, dpi = 300)
```

```{r}
qc_graphs(merged_df, merged_df$median_UMI, "Donors", "median UMI", "Put Village: median UMI")
qc_graphs(merged_df, merged_df$median_nGenes, "Donors", "median nGenes", "Put Village: median nGenes")
qc_graphs(merged_df, merged_df$median_reads_per_umi, "Donors", "median reads/UMI", "Put Village: median reads/UMI")
qc_graphs(merged_df, merged_df$median_pct_intronic, "Donors", "median pct intronic", "Put Village: median pct intronic")
qc_graphs(merged_df, merged_df$median_pct_mito, "Donors", "median pct mito", "Put Village: median pct mito")
```



```{r}
#totalcells
cells_per_library = as.data.frame(table(filtered_CaH_Cohort2@meta.data$library))
names(cells_per_library)[names(cells_per_library) == "Var1"] ="library"
names(cells_per_library)[names(cells_per_library) == "Freq"] ="total_cells"
cells_per_library  

#median/mean nUMI
median_nUmi = filtered_CaH_Cohort2@meta.data %>%
  group_by(library) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi

#median/mean nGenes
median_nGenes = filtered_CaH_Cohort2@meta.data %>%
  group_by(library) %>%
  summarize(median_nGenes = median(nFeature_RNA, na.rm = TRUE))
median_nGenes

#median/mean reads/numi
median_reads_per_umi = filtered_CaH_Cohort2@meta.data %>%
  group_by(library) %>%
  summarize(median_reads_per_umi = median((nRead/nUmi), na.rm = TRUE))
median_reads_per_umi

#median/mean pctintronic
median_pct_intronic = filtered_CaH_Cohort2@meta.data %>%
  group_by(library) %>%
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))
median_pct_intronic


#median/mean pctmito
median_pct_mito = filtered_CaH_Cohort2@meta.data %>%
  group_by(library) %>%
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))
median_pct_mito

```
```{r}
library(gridExtra)
caudate_library_qc_df_list = list(median_nUmi, median_nGenes, median_reads_per_umi, median_pct_intronic,  median_pct_mito)
caudate_library_qc_df = reduce(caudate_library_qc_df_list, left_join, by = "library")
caudate_library_qc_df

caudate_library_qc_df[, -1] <- round(caudate_library_qc_df[, -1], digits = 2)
caudate_library_qc_df
table_grob = tableGrob(caudate_library_qc_df)
ggsave("caudate_library_qc_df.png", table_grob, width = 15, height = 6, dpi = 300)
```


```{r}
library(purrr)
# List of data frames to merge
df_list <- list(median_nUmi, median_nGenes,median_reads_per_umi, median_pct_intronic, median_pct_mito )

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "library"))

# Print the merged data frame
print(merged_df)

```

```{r}
merged_df_final_combined = rbind(merged_df,merged_df_final)
merged_df_final_combined
```



```{r}
qc_graphs_lib = function(data, y, xlab, ylab, title){

x_order = c("CaH_rxn1", "CaH_rxn2", "Put_rxn1", "Put_rxn2")  

data$library = factor(data$library, levels = x_order)
median = median(y)

ggplot(data, aes(x = library, y = y, fill = library)) +
  geom_bar(stat = "identity") + xlab(xlab) + ylab(ylab)+ labs(fill = "library")+ ggtitle(label = title, subtitle = paste("Median: ", sprintf("%0.2f", median)))+geom_hline(yintercept = median, linetype = "dashed", color = "black") +
  geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
}
```

```{r}
qc_graphs_lib(merged_df_final_combined, merged_df_final_combined$median_UMI, "Library", "median UMI", "CaH + Put: median UMI")
qc_graphs_lib(merged_df_final_combined, merged_df_final_combined$median_nGenes, "Library", "median nGenes", "CaH + Put: median nGenes")
qc_graphs_lib(merged_df_final_combined, merged_df_final_combined$median_reads_per_umi, "Library", "median reads/UMI", "CaH + Put: median reads/UMI")
qc_graphs_lib(merged_df_final_combined, merged_df_final_combined$median_pct_intronic, "Library", "median pct intronic", "CaH + Put: median pct intronic")
qc_graphs_lib(merged_df_final_combined, merged_df_final_combined$median_pct_mito, "Library", "median pct mito", "CaH + Put: median pct mito")
```



```{r}
filtered_CaH_Cohort2
filtered_Put_Cohort2
```






```{r}
1+1
```



```{r}
log_nUMI = log10(CaH_Cohort2$nUmi)
median_log_nUMI = median(log_nUMI)
title = paste0("CaH_Cohort2: log10(nUMI)", ", Median: ", sprintf("%0.3f", median_log_nUMI))
log_nUMI_hist = hist(log_nUMI, main= title, xlab = "log10(nUMI)", col="grey", breaks = 100)
```


















```{r}
sct_cleaned_CaH_Put_Cohort2 = qread("cleaned_CaH_Put_Cohort2_sct_ucell_031725.qs")
sct_cleaned_CaH_Put_Cohort2
sct_cleaned_CaH_Put_Cohort2@meta.data
```

```{r}
DefaultAssay(sct_cleaned_CaH_Put_Cohort2) = "RNA"
xdp_co2_neurons = subset(sct_cleaned_CaH_Put_Cohort2, subset = final_cell_class_merged == "neuron")
xdp_co2_neurons
```


```{r}
# this code should take 3-10 minutes
library(sctransform)
xdp_co2_neurons = SCTransform(xdp_co2_neurons, vars.to.regress = "pct_mito", verbose = FALSE)
DefaultAssay(xdp_co2_neurons) = "SCT"

hvgs = getSeuratVarFeatureIntersectByCol(xdp_co2_neurons, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
xdp_co2_neurons = (xdp_co2_neurons
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(xdp_co2_neurons, group.by = "donor_id") 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.2", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.3", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.4", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.5", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.6", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.7", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.8", label=T)
DimPlot(xdp_co2_neurons, group.by = "final_cell_class", label=T)
DimPlot(xdp_co2_neurons, group.by = "neuron_cell_class", label=T)
```

```{r}
Idents(xdp_co2_neurons) = xdp_co2_neurons$SCT_snn_res.0.4

classes = c("D1_matrix", "D2_matrix", "D1_patch", "eSPN", "D1_D2", "non_SPN", "D2_patch", "non_SPN", "non_SPN", "D1_D2", "D1_D2", "non_SPN", "non_SPN", "non_SPN", "unsure", "doublet", "D1_D2")


xdp_co2_neurons= assignCellClasses(xdp_co2_neurons, classes=classes, cluster_col="SCT_snn_res.0.4", class_col = "final_neuron_cell_class_merged")

DimPlot(xdp_co2_neurons, group.by = "final_neuron_cell_class_merged" ,label = T, raster = FALSE)

```

```{r}
library(harmony)
xdp_co2_neurons = (xdp_co2_neurons
    %>% RunHarmony(
        group.by = "donor_id")
    %>% FindNeighbors(reduction='harmony', dims=1:20)
    %>% FindClusters(res=0.05)
    %>% FindClusters(res=0.1)
    %>% FindClusters(res=0.2)
    %>% FindClusters(res=0.3)
    %>% FindClusters(res=0.4)
    %>% FindClusters(res=0.5)
    %>% FindClusters(res=0.6)
    %>% FindClusters(res=0.7)
    %>% FindClusters(res=0.8)    
    %>% RunUMAP(reduction="harmony", dims=1:20)
)
xdp_co2_neurons
xdp_co2_neurons@meta.data
DimPlot(xdp_co2_neurons, group.by = "donor_id") 
DimPlot(xdp_co2_neurons, group.by = "region")
DimPlot(xdp_co2_neurons, group.by = "Condition")
DimPlot(xdp_co2_neurons, group.by = "neuron_cell_class", label=T)
DimPlot(xdp_co2_neurons, group.by = "final_neuron_cell_class_merged", label=T)
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.05", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.1", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.2", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.3", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.4", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.5", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.6", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.7", label=T) 
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.8", label=T)
DimPlot(xdp_co2_neurons, group.by = "final_cell_class", label=T)
DimPlot(xdp_co2_neurons, group.by = "neuron_cell_class", label=T)
```
```{r}
FeaturePlot(xdp_co2_neurons, features = c("SYT1", "AQP4", "C1QA","FLT1"),raster=FALSE)
FeaturePlot(xdp_co2_neurons, features = c("OLIG1", "OLIG2", "MOG" ,"CD96"),raster=FALSE) 
FeaturePlot(xdp_co2_neurons, features = c("MBP", "MOBP", "TF", "ST18"),raster=FALSE)
FeaturePlot(xdp_co2_neurons, features = c("RBFOX3", "CX3CR1", "GFAP" ,"AQP4"),raster=FALSE) 
FeaturePlot(xdp_co2_neurons, features = c("GAD1", "GAD2", "SLC17A7" ,"SLC17A6"),raster=FALSE) 
FeaturePlot(xdp_co2_neurons, features = c("DRD1", "DRD2", "CASZ1" ,"PPP1R1B"),raster=FALSE) 
FeaturePlot(xdp_co2_neurons, features = c("EPHA4", "SEMA3E", "SLC17A7" ,"SLC17A6"),raster=FALSE) 
FeaturePlot(xdp_co2_neurons, features = c("P2RY12", "CX3CR1", "C1QB", "BCAS1"),raster=FALSE) 
FeaturePlot(xdp_co2_neurons, features = c("SEMA5B", "KREMEN1", "PDE1C"),raster=FALSE) 
FeaturePlot(xdp_co2_neurons, features = c("TAC1", "NNAT", "OPRM1", "ID4"),raster=FALSE) 
FeaturePlot(xdp_co2_neurons, features = c("SGK1", "EPHA4", "SV2B"),raster=FALSE) 
FeaturePlot(xdp_co2_neurons, features = c("SST", "CALB1", "PVALB", "LAMP5"),raster=FALSE) 
FeaturePlot(xdp_co2_neurons, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 
```

```{r}
Idents(xdp_co2_neurons) = xdp_co2_neurons$SCT_snn_res.0.4

classes = c("D1_matrix", "D2_matrix", "D1_patch", "D1_D2", "eSPN", "D2_patch", "D1_D2", "non_SPN", "non_SPN", "non_SPN", "D1_D2", "doublet", "non_SPN", "non_SPN", "non_SPN","non_SPN", "doublet")


xdp_co2_neurons= assignCellClasses(xdp_co2_neurons, classes=classes, cluster_col="SCT_snn_res.0.4", class_col = "final_neuron_cell_class_merged_harmony")

DimPlot(xdp_co2_neurons, group.by = "final_neuron_cell_class_merged_harmony" ,label = T, raster = FALSE)

```
```{r}
DimPlot(xdp_co2_neurons, group.by = "donor_id") 
DimPlot(xdp_co2_neurons, group.by = "region")
DimPlot(xdp_co2_neurons, group.by = "Condition")
DimPlot(xdp_co2_neurons, group.by = "neuron_cell_class", label=T)
DimPlot(xdp_co2_neurons, group.by = "SCT_snn_res.0.4", label=T)
DimPlot(xdp_co2_neurons, group.by = "final_neuron_cell_class_merged", label=T)
DimPlot(xdp_co2_neurons, group.by = "final_neuron_cell_class_merged_harmony", label=T)
```


```{r}
marker_genes <- c("CD96", "CX3CR1", "P2RY12", "C1QB", "C1QA", "CASZ1", "FLT1", "MOBP", "MOG", "MBP", "OLIG2", "OLIG1", "ST18", "GFAP", "AQP4", "SEMA3E", "EPHA4", "PPP1R1B", "DRD2", "DRD1", "GAD2", "GAD1", "SYT1", "RBFOX3", "SLC17A6", "SLC17A7", "SST", "CALB1")

Dotplot = DotPlot(object = xdp_co2_neurons, features = marker_genes, group.by = "SCT_snn_res.0.4")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```





#findmarkers
```{r}
Idents(xdp_co2_neurons) = "SCT_snn_res.0.4"
markers = FindAllMarkers(xdp_co2_neurons, only.pos = TRUE, min.pct = 0.2, logfc.threshold = 1.25)

markers1[markers1$cluster == "14",]


markers
markers1 = subset(markers, subset = pct.1 > 0.7)
markers1
markers2 = subset(markers1, subset = pct.2 < 0.3)
markers2


markers_11= subset(markers1, subset = cluster == "11")
markers_11
```

```{r}
xdp_co2_neurons = subset(xdp_co2_neurons, subset = final_neuron_cell_class_merged_harmony != "doublet")
```


```{r}
DimPlot(xdp_co2_neurons, group.by = "region")
DimPlot(xdp_co2_neurons, group.by = "donor_id")
DimPlot(xdp_co2_neurons, group.by = "Condition")
DimPlot(xdp_co2_neurons, group.by = "final_cell_class_merged", label=T)
DimPlot(xdp_co2_neurons, group.by = "neuron_cell_class", label=T)
DimPlot(xdp_co2_neurons, group.by = "final_neuron_cell_class_merged", label=T)
DimPlot(xdp_co2_neurons, group.by = "final_neuron_cell_class_merged_harmony", label=T)

```
```{r}
xdp_co2_neurons
```
```{r}
Idents(xdp_co2_neurons) = "final_neuron_cell_class_merged_harmony"
```

```{r}
unique(xdp_co2_neurons$final_neuron_cell_class_merged_harmony)
```


```{r}
xdp_co2_neurons@meta.data$final_neuron_subclass = xdp_co2_neurons$final_neuron_cell_class_merged_harmony
xdp_co2_neurons$final_neuron_subclass[xdp_co2_neurons$final_neuron_subclass == "D1_patch"] = "SPN_patch"
xdp_co2_neurons$final_neuron_subclass[xdp_co2_neurons$final_neuron_subclass == "D2_patch"] = "SPN_patch"
xdp_co2_neurons$final_neuron_subclass[xdp_co2_neurons$final_neuron_subclass == "D1_matrix"] = "SPN_matrix"
xdp_co2_neurons$final_neuron_subclass[xdp_co2_neurons$final_neuron_subclass == "D2_matrix"] = "SPN_matrix"
unique(xdp_co2_neurons$final_neuron_subclass)
```


```{r}
xdp_co2_neurons@meta.data
```

#MASC Combo
```{r}
library(lme4)
sobj = xdp_co2_neurons
sobj = subset(sobj, subset = donor_id != "SCF_22_049CCF")
sobj
model_cols = c("donor_id", "final_neuron_cell_class_merged_harmony", "Condition", "Gender", "Age.at.Death", "region") #include any other covariates of interest, replace cell_class with whatever column defines the cluster annotations
pd = sobj@meta.data[,model_cols]
pd = pd[complete.cases(pd),] # don't want any NAs
pd$case_control_factor = as.factor(pd$Condition)

masc_df = .sconline.MASCfn(
    dataset=pd,
    cluster=pd$final_neuron_cell_class_merged_harmony, # cluster annotations
    contrast="case_control_factor", # name of contrast annotations (what you want to run the test for)
    random_effects="donor_id", # name of random effects annotations (not interested in these coefficients, but account for variability in probable sources ob batch effects, in this case donor
    fixed_effects = c("Age.at.Death", "Gender", "region") # your covariates
)

masc_df

my_order <- unique(masc_df$cluster)
masc_df$cluster_name <- sub("cluster", "", masc_df$cluster) # for legibility
masc_df$log2_or = log2(masc_df$case_control_factorXDP.OR) # the names of the case_control_factor<whatever> columns will change from project to project depending on the unique values of your Condition column
masc_df$log2_or_ci_low = log2(masc_df$case_control_factorXDP.OR.95pct.ci.lower)
masc_df$log2_or_ci_high = log2(masc_df$case_control_factorXDP.OR.95pct.ci.upper)

# order the graph to put disease-enriched populations on top
masc_df = masc_df[order(-masc_df$log2_or), ] 
masc_df$cluster_name = factor(masc_df$cluster_name, levels = masc_df$cluster_name[order(masc_df$log2_or)])
masc_df


library(RColorBrewer)
library(ggplot2)

# Create the forest plot with ticks on error bars, axis lines with ticks, RdBu color map, and opaque white circles on top
a = ggplot(masc_df, aes(y = cluster_name, x = log2_or)) +
  ggtitle("Combined Cohort 2 CaH + Put neurons (no CCF)") +
  geom_vline(xintercept = 0, linetype = "dotted", color = "gray") +  # Add dotted vertical line at x=0
  geom_segment(aes(x = log2_or_ci_low, xend = log2_or_ci_high, y = cluster_name, yend = cluster_name, color = log2_or), size = 1) +  # Add horizontal error bars
  geom_point(size = 3, aes(color = log2_or), shape = 1) +  # Add points for effect sizes
  geom_point(size = 3, shape = 21, fill = "white") +  # Add opaque white circle on top of the error bar line
  scale_color_gradientn(colors = RColorBrewer::brewer.pal(10, "RdBu")) +  # Use RdBu color map
  theme_minimal() +  # Minimal theme
  labs(x = "log2(OR)", y = "Cell Classes") +  # Axis labels
  theme(
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), 
    legend.position = "none",
    plot.title = element_text(size=16),
    axis.line = element_line(color = "black"),  # Add axis lines
    axis.ticks = element_line(color = "black"),  # Add axis ticks
    axis.text = element_text(size = 14),  # Increase tick label font size
    axis.title = element_text(size = 15)  # Increase axis label font size
  )

ggsave(a, filename = "pic.png", width = 8, height = 8)
```

```{r}
cells_per_donor = as.data.frame(table(xdp_co2_neurons@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(xdp_co2_neurons@meta.data$final_neuron_cell_class_merged_harmony, xdp_co2_neurons@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df


cell_class_df

x_order = c("SCF_22_063CM", "SCF_22_049CCF", "SCF_22_067CM", "SCF_23_072CF", "SCF_23_074CF",  "SCF_22_065CM", "SCF_23_075CM", "SCF_23_079CF",
"SCF_21_041", "SCF_23_070_MMC", "SCF_21_038",  "SCF_21_027","SCF_21_032",  "SCF_21_034", "SCF_21_029")   
            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = x_order)

ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors") + ylab("Number of cells by Neuron Subclasses") + labs(fill = "Neuron Subclass")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors") + ylab("Neuron Subclass Proportion") + labs(fill = "Neuron Subclass")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  
```


#subclass

#MASC Combo
```{r}
library(lme4)
sobj = xdp_co2_neurons
sobj = subset(sobj, subset = donor_id != "SCF_22_049CCF")
sobj
model_cols = c("donor_id", "final_neuron_subclass", "Condition", "Gender", "Age.at.Death", "region") #include any other covariates of interest, replace cell_class with whatever column defines the cluster annotations
pd = sobj@meta.data[,model_cols]
pd = pd[complete.cases(pd),] # don't want any NAs
pd$case_control_factor = as.factor(pd$Condition)

masc_df = .sconline.MASCfn(
    dataset=pd,
    cluster=pd$final_neuron_subclass, # cluster annotations
    contrast="case_control_factor", # name of contrast annotations (what you want to run the test for)
    random_effects="donor_id", # name of random effects annotations (not interested in these coefficients, but account for variability in probable sources ob batch effects, in this case donor
    fixed_effects = c("Age.at.Death", "Gender", "region") # your covariates
)

masc_df

my_order <- unique(masc_df$cluster)
masc_df$cluster_name <- sub("cluster", "", masc_df$cluster) # for legibility
masc_df$log2_or = log2(masc_df$case_control_factorXDP.OR) # the names of the case_control_factor<whatever> columns will change from project to project depending on the unique values of your Condition column
masc_df$log2_or_ci_low = log2(masc_df$case_control_factorXDP.OR.95pct.ci.lower)
masc_df$log2_or_ci_high = log2(masc_df$case_control_factorXDP.OR.95pct.ci.upper)

# order the graph to put disease-enriched populations on top
masc_df = masc_df[order(-masc_df$log2_or), ] 
masc_df$cluster_name = factor(masc_df$cluster_name, levels = masc_df$cluster_name[order(masc_df$log2_or)])
masc_df


library(RColorBrewer)
library(ggplot2)

# Create the forest plot with ticks on error bars, axis lines with ticks, RdBu color map, and opaque white circles on top
a = ggplot(masc_df, aes(y = cluster_name, x = log2_or)) +
  ggtitle("Combined Cohort 2 CaH + Put neurons (no CCF)") +
  geom_vline(xintercept = 0, linetype = "dotted", color = "gray") +  # Add dotted vertical line at x=0
  geom_segment(aes(x = log2_or_ci_low, xend = log2_or_ci_high, y = cluster_name, yend = cluster_name, color = log2_or), size = 1) +  # Add horizontal error bars
  geom_point(size = 3, aes(color = log2_or), shape = 1) +  # Add points for effect sizes
  geom_point(size = 3, shape = 21, fill = "white") +  # Add opaque white circle on top of the error bar line
  scale_color_gradientn(colors = RColorBrewer::brewer.pal(10, "RdBu")) +  # Use RdBu color map
  theme_minimal() +  # Minimal theme
  labs(x = "log2(OR)", y = "Cell Classes") +  # Axis labels
  theme(
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), 
    legend.position = "none",
    plot.title = element_text(size=16),
    axis.line = element_line(color = "black"),  # Add axis lines
    axis.ticks = element_line(color = "black"),  # Add axis ticks
    axis.text = element_text(size = 14),  # Increase tick label font size
    axis.title = element_text(size = 15)  # Increase axis label font size
  )

ggsave(a, filename = "pic.png", width = 8, height = 8)
```

```{r}
DimPlot(xdp_co2_neurons, group.by = "final_neuron_subclass", label = T)
```


```{r}
cells_per_donor = as.data.frame(table(xdp_co2_neurons@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(xdp_co2_neurons@meta.data$final_neuron_subclass, xdp_co2_neurons@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df


cell_class_df

x_order = c("SCF_22_063CM", "SCF_22_049CCF", "SCF_22_067CM", "SCF_23_072CF", "SCF_23_074CF",  "SCF_22_065CM", "SCF_23_075CM", "SCF_23_079CF",
"SCF_21_041", "SCF_23_070_MMC", "SCF_21_038",  "SCF_21_027","SCF_21_032",  "SCF_21_034", "SCF_21_029")  
            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = x_order)

ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors") + ylab("Number of cells by Neuron Subclasses") + labs(fill = "Neuron Subclass")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors") + ylab("Neuron Subclass Proportion") + labs(fill = "Neuron Subclass")+ geom_vline(xintercept =  8.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  
```

```{r}
xdp_co2_neurons
xdp_co2_neurons@meta.data
```
```{r}
NFKB_genes =  c("MARCKS", "IL23A", "NINJ1", "TNFSF9", "SIK1", "ATF3", "SERPINE1", "MYC", "HES1", "CCNL1", "CCN1", "EGR1", "EGR2", "JAG1", "EGR3", "ABCA1", "GADD45B", "GADD45A", "PLK2", "KLF10", 
           "EIF1", "EHD1", "FOSL2", "FOSL1", "GPR183", "PLPP3", "IFIT2", "ICAM1", "ZC3H12A", "IER2", 
           "IL12B", "JUNB", "IER5", "IER3", "STAT5A", "DUSP5", "EDN1", "JUN", "DUSP4", "DUSP1", 
           "DUSP2", "TSC22D1", "CCL20", "SPHK1", "LIF", "IL18", "TUBB2A", "RHOB", "VEGFA", "PTPRE", 
           "IL1A", "TLR2", "IL1B", "BHLHE40", "ID2", "CLCF1", "REL", "FJX1", "SGK1", "BTG3", "BTG2", 
           "BTG1", "SDC4", "LITAF", "AREG", "SOCS3", "PANX1", "RIPK2", "NFIL3", "SERPINB2", "GCH1", 
           "IFNGR2", "G0S2", "FOS", "SERPINB8", "F3", "SPSB1", "FOSB", "PER1", "F2RL1", "HBEGF", 
           "CD44", "TRIP10", "CDKN1A", "PTGER4", "PTGS2", "IFIH1", "NAMPT", "OLR1", "ICOSLG", 
           "PHLDA1", "ZBTB10", "TAP1", "PNRC1", "CXCL10", "CXCL11", "IL6ST", "CD69", "SQSTM1", 
           "RELA", "CSF2", "CD83", "CSF1", "PPP1R15A", "CD80", "TNC", "TNF", "TANK", "RELB", "ZFP36", 
           "CCND1", "RNF19B", "CCRL2", "DENND5A", "PHLDA2", "MAP3K8", "LDLR", "SLC16A6", "SMAD3", 
           "TGIF1", "MAP2K3", "DDX58", "TRAF1", "INHBA", "NFKB1", "NFKB2", "GEM", "NR4A3", "MAFF", 
           "RCAN1", "NR4A2", "EFNA1", "MXD1", "BIRC2", "YRDC", "BIRC3", "IL7R", "PFKFB3", "IRS2", 
           "SLC2A3", "PLAU", "SLC2A6", "SAT1", "ETS2", "NR4A1", "SNN", "PMEPA1", "TNFRSF9", "MSC", 
           "TIPARP", "LAMB3", "GFPT2", "CFLAR", "TNIP1", "IRF1", "NFKBIA", "BMP2", "IL6", "TNIP2", 
           "BCL6", "BCL3", "NFKBIE", "NFE2L2", "B4GALT1", "NFAT5", "TNFAIP8", "BCL2A1", "TNFAIP6", 
           "TNFAIP3", "CXCL2", "CXCL1", "TNFAIP2", "CXCL3", "CXCL6", "FUT4", "DRAM1", "DNAJB4", 
           "PDE4B", "PDLIM5", "MCL1", "KDM6B", "IL15RA", "PLAUR", "ATP2B1", "KLF4", "KLF2", "SOD2", 
           "KLF9", "KLF6", "ACKR3", "PTX3", "B4GALT5", "TRIB1", "CEBPB", "CEBPD", "PLEK", "KYNU", 
           "CCL5", "CCL4", "CCL2")

interferon_alpha = c("IL4R", "CD74", "SP110", "MX1", "RSAD2", "TAP1", "PSMB9", "CXCL10", "CXCL11", "IFI44L", "LAP3", "LPAR6", "IFI27", "TRAFD1", "TRIM14", "LY6E", "IFITM2", "IFITM3", "SAMD9", "CSF1", 
           "IFITM1", "UBE2L6", "SAMD9L", "RNF31", "HERC6", "LAMP3", "CCRL2", "MVB12A", "B2M", "CMTR1", 
           "GBP2", "GBP4", "EPSTI1", "BATF2", "HLA-C", "TENT5A", "TDRD7", "ISG15", "IFI44", "PARP14", 
           "MOV10", "ISG20", "PARP12", "PLSCR1", "PROCR", "SELL", "PSME2", "PSME1", "CNP", "LGALS3BP", 
           "IFI35", "IFIT2", "IFI30", "DDX60", "IFIT3", "NUB1", "HELZ2", "PNPT1", "IL15", "NCOA7", 
           "EIF2AK2", "NMI", "OAS1", "BST2", "IRF1", "ELF1", "IL7", "IRF2", "IRF9", "IRF7", "UBA7", 
           "GMPR", "ADAR", "CASP8", "TRIM5", "RIPK2", "CASP1", "STAT2", "WARS1", "PARP9", "PSMA3", 
           "TXNIP", "CMPK2", "CD47", "RTP4", "C1S", "OGFR", "TMEM140", "USP18", "OASL", "SLC25A28", 
           "IFIH1", "PSMB8", "DHX58", "TRIM25", "TRIM26", "TRIM21")

antigen_processing_genes = c("CTSB", "CD74", "HSP90AA1", "KLRC3", "HSPA6", "HSPA5", "HSPA1L", "KLRC4", "HSPA8", "HSPA2", "TAP2", "HSPA4", "TAP1", "TAPBP", "CD4", "RFX5", "IFNG", "HSPA1A", "HLA-DRA", "HLA-DQB1", "KLRD1", "CALR", "HSPA1B", "TNF", "KIR2DL1", "KIR2DL2", "KIR2DL3", "KIR2DL4", "CTSS", "CTSL", "B2M", "HLA-DOB", "KIR2DL5A", "CIITA", "HLA-DQA2", "HLA-DOA", "HLA-DQA1", "PDIA3", "HLA-DRB5", 
"KIR2DS1", "NFYB", "KIR2DS2", "NFYC", "KIR2DS3", "KIR2DS4", "KIR2DS5", "HLA-C", "RFXANK", 
"KIR3DL1", "CD8B2", "KIR3DL2", "HLA-A", "HLA-B", "KIR3DL3", "HLA-G", "HLA-E", "HLA-F", "CREB1", "CD8B", "RFXAP", "CD8A", "CANX", "LGMN", "PSME2", "PSME3", "HLA-DPB1", "PSME1", "HLA-DRB4", 
"HLA-DRB3", "HLA-DRB1", "HSP90AB1", "IFI30", "HLA-DMB", "HLA-DPA1", "KLRC1", "KLRC2", "NFYA","HLA-DMA")
C_minus = qread("temp_disco/more_temp_1205/final_c_minus_genes.qs")
C_plus = qread("temp_disco/more_temp_1205/final_c_plus_genes.qs")
```

```{r}
xdp_co2_neurons <- AddModuleScore_UCell(xdp_co2_neurons,
  features = list(Score = NFKB_genes ),
  name = 'NFKB'
)

xdp_co2_neurons <- AddModuleScore_UCell(xdp_co2_neurons,
  features = list(Score = interferon_alpha),
  name = 'interferon_alpha'
)

xdp_co2_neurons <- AddModuleScore_UCell(xdp_co2_neurons,features = list(C_minus = C_minus), name = '_scores')
xdp_co2_neurons <- AddModuleScore_UCell(xdp_co2_neurons,features = list(C_plus = C_plus), name = '_scores')

xdp_co2_neurons <- AddModuleScore_UCell(xdp_co2_neurons,features = list(antigen_processing = antigen_processing_genes), name = '_scores')


qsave(xdp_co2_neurons, "XDP_CaH_Put_Cohort2_subclustered_neurons_032025.qs")
```

---
title: "R Notebook"
output: html_notebook
---
#merge everything
```{r}
library(Seurat)
library(qs)
library(tidyverse)
library(ggplot2)
```

```{r}
xdp_co1
sct_cleaned_CaH_Put_Cohort2

xdp_co1@meta.data
sct_cleaned_CaH_Put_Cohort2@meta.data

xdp_co1@meta.data$final_cell_class_merged = xdp_co1@meta.data$final_cell_class
```




```{r}
Idents(xdp_co1) = "final_cell_class_merged"
Idents(sct_cleaned_CaH_Put_Cohort2) = "final_cell_class_merged"
```

```{r}
xdp_co1@meta.data$ROI_astrocyte = NULL
xdp_co1@meta.data$ROI_endo = NULL
xdp_co1@meta.data$ROI_oligo = NULL
xdp_co1@meta.data$ROI_opc = NULL
xdp_co1@meta.data$ROI_neuron = NULL
xdp_co1@meta.data$ROI_micro_roi_genes = NULL
xdp_co1@meta.data$ROI_upreg_astrocyte = NULL
xdp_co1@meta.data$ROI_upreg_endo = NULL
xdp_co1@meta.data$ROI_upreg_oligo = NULL
xdp_co1@meta.data$ROI_upreg_opc = NULL
xdp_co1@meta.data$ROI_upreg_neuron = NULL
xdp_co1@meta.data$ROI_upreg_mg = NULL
xdp_co1@meta.data$ROI_downreg_astrocyte = NULL
xdp_co1@meta.data$ROI_downreg_oligo = NULL
xdp_co1@meta.data$ROI_downreg_neuron = NULL
xdp_co1@meta.data$ROI_downreg_mg = NULL
xdp_co1@meta.data$ROI_upreg_SPN_matrix = NULL
xdp_co1@meta.data$ROI_downreg_SPN_matrix = NULL
xdp_co1@meta.data$ROI_mg = NULL
xdp_co1@meta.data$ROI_upreg_new_astrocyte = NULL
xdp_co1@meta.data$ROI_upreg_new_endo = NULL
xdp_co1@meta.data$ROI_upreg_new_oligo = NULL
xdp_co1@meta.data$ROI_upreg_new_opc = NULL
xdp_co1@meta.data$ROI_upreg_new_neuron = NULL
xdp_co1@meta.data$ROI_upreg_new_mg = NULL
xdp_co1@meta.data$ROI_downreg_new_astrocyte = NULL
xdp_co1@meta.data$ROI_downreg_new_oligo = NULL
xdp_co1@meta.data$ROI_downreg_new_neuron = NULL
xdp_co1@meta.data$ROI_downreg_new_mg = NULL
xdp_co1@meta.data$Scoreinflammatory = NULL
xdp_co1@meta.data$Scoreinterferon_gamma = NULL
xdp_co1@meta.data$ScoreMHC1_antigen = NULL
xdp_co1@meta.data$ScoreMHC2_antigen = NULL
xdp_co1@meta.data$Scorebase_excision = NULL
xdp_co1@meta.data$Scoremismatch_repair = NULL
xdp_co1@meta.data$HLA_A = NULL
xdp_co1@meta.data$HLA_combined = NULL
```


```{r}
colnames(xdp_co1@meta.data)
colnames(sct_cleaned_CaH_Put_Cohort2@meta.data)
```

```{r}
xdp_co1@meta.data$Cohort = "Cohort_1"
sct_cleaned_CaH_Put_Cohort2@meta.data$Cohort = "Cohort_2"
sct_cleaned_CaH_Put_Cohort2@meta.data$Region = sct_cleaned_CaH_Put_Cohort2$region
sct_cleaned_CaH_Put_Cohort2@meta.data$Sex= sct_cleaned_CaH_Put_Cohort2$sex
```


```{r}
sobj_list = list(xdp_co1,
sct_cleaned_CaH_Put_Cohort2)

counts_list <- lapply(sobj_list, function(sobj) GetAssayData(sobj, slot = "counts"))
merged_counts <- do.call(cbind, counts_list)
metadata_list <- lapply(sobj_list, function(sobj) sobj@meta.data)
```


```{r}
# Get all unique column names from both metadata dataframes
all_cols <- unique(unlist(lapply(metadata_list, colnames)))
all_cols
# Ensure each metadata dataframe has all columns, filling missing ones with NA
metadata_list <- lapply(metadata_list, function(df) {
  df[, setdiff(all_cols, colnames(df))] <- NA
  df <- df[, all_cols]  # Reorder columns to match
  return(df)
})

# Now merge metadata
merged_metadata <- do.call(rbind, metadata_list)
merged_metadata
```

```{r}
merged_metadata <- do.call(rbind, metadata_list)
merged_metadata

stopifnot(all(rownames(merged_metadata) == colnames(merged_counts)))

# Create Seurat object
XDP_Cohorts_1_2_full <- CreateSeuratObject(counts = merged_counts, meta.data = merged_metadata)
XDP_Cohorts_1_2_full
```

```{r}
table(XDP_Cohorts_1_2_full@meta.data$donor_id)
```
```{r}
table(XDP_Cohorts_1_2_full@meta.data$library_cohort)
```

```{r}
XDP_Cohorts_1_2_full$Region[XDP_Cohorts_1_2_full$Region == "caudate"] = "Caudate" 
XDP_Cohorts_1_2_full$Region[XDP_Cohorts_1_2_full$Region == "putamen"] = "Putamen" 

XDP_Cohorts_1_2_full@meta.data$library_cohort = paste0(XDP_Cohorts_1_2_full$library, "_", XDP_Cohorts_1_2_full$Cohort)
```

```{r}
library(sctransform)
options(future.globals.maxSize = 400 * 1024^3)  
XDP_Cohorts_1_2_full = SCTransform(XDP_Cohorts_1_2_full, vars.to.regress = "pct_mito", verbose = FALSE)
DefaultAssay(XDP_Cohorts_1_2_full) = "SCT"
XDP_Cohorts_1_2_full
```


```{r}
# this code should take 3-10 minutes
hvgs = getSeuratVarFeatureIntersectByCol(XDP_Cohorts_1_2_full, subset_col="donor_id", original_nfeatures=2500)
n_dims_use=20
XDP_Cohorts_1_2_full = (XDP_Cohorts_1_2_full
   %>% NormalizeData() # log normalizes raw counts
   %>% ScaleData(features=hvgs, split.by="donor_id") # within each cell, for each gene scale the data (i.e. subtract the donor's mean and divide by the donor's standard deviation of that gene)
   %>% RunPCA(features=hvgs, npcs=n_dims_use) # Reduce the dimensions to the n_dims_use dimensions that best explain the data https://en.wikipedia.org/wiki/Principal_component_analysis
   %>% FindNeighbors(dims = 1:n_dims_use) # Finds every cells closest neighbors in the PCA space
   %>% FindClusters(resolution = 0.2) # finds clusters at a variety of resolutions
   %>% FindClusters(resolution = 0.3) # after clustering, the cluster labels are accessible via filtered_merged_caudate$RNA_snn_res.{resolution}
   %>% FindClusters(resolution = 0.4) # e.g.: filtered_merged_caudate$RNA_snn_res.0.4 here
   %>% FindClusters(resolution = 0.5)
   %>% FindClusters(resolution = 0.6)
   %>% FindClusters(resolution = 0.7)
   %>% FindClusters(resolution = 0.8)
   %>% FindClusters(resolution = 0.9)
   %>% FindClusters(resolution = 1)
   %>% RunUMAP(dims = 1:n_dims_use) # projects the PCA onto 2 dimensions, such that both "global structure" and "local structure" are optimally preserved
)

setwh = function(w, h){
   # set the width of the ensuing figures to w, height to h
   options(repr.plot.width = w, repr.plot.height = h)
}

DimPlot(XDP_Cohorts_1_2_full, group.by = "donor_id") 
DimPlot(XDP_Cohorts_1_2_full, group.by = "Region")
DimPlot(XDP_Cohorts_1_2_full, group.by = "Condition")
DimPlot(XDP_Cohorts_1_2_full, group.by = "Cohort")
DimPlot(XDP_Cohorts_1_2_full, group.by = "Region", label=T)
DimPlot(XDP_Cohorts_1_2_full, group.by = "final_cell_class_merged", label=T)
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.2", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.3", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.4", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.5", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.6", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.7", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.8", label=T)
```


```{r}
library(harmony)
XDP_Cohorts_1_2_full = (XDP_Cohorts_1_2_full
    %>% RunHarmony(
        group.by = "donor_id")
    %>% FindNeighbors(reduction='harmony', dims=1:20)
    %>% FindClusters(res=0.05)
    %>% FindClusters(res=0.1)
    %>% FindClusters(res=0.2)
    %>% FindClusters(res=0.3)
    %>% FindClusters(res=0.4)
    %>% FindClusters(res=0.5)
    %>% FindClusters(res=0.6)
    %>% FindClusters(res=0.7)
    %>% FindClusters(res=0.8)    
    %>% RunUMAP(reduction="harmony", dims=1:20)
)
XDP_Cohorts_1_2_full
XDP_Cohorts_1_2_full@meta.data
DimPlot(XDP_Cohorts_1_2_full, group.by = "donor_id", raster = F) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.05", label=T, raster = F) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.1", label=T, raster = F) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.2", label=T, raster = F) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.3", label=T, raster = F) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.4", label=T, raster = F) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.5", label=T, raster = F) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.6", label=T, raster = F) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.7", label=T, raster = F) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.8", label=T, raster = F)
DimPlot(XDP_Cohorts_1_2_full, group.by = "library", raster = F)
DimPlot(XDP_Cohorts_1_2_full, group.by = "Cohort", raster = F)
DimPlot(XDP_Cohorts_1_2_full, group.by = "library_cohort", raster = F)
DimPlot(XDP_Cohorts_1_2_full, group.by = "final_cell_class_merged", label=T, raster = F)
DimPlot(XDP_Cohorts_1_2_full, group.by = "Condition", label=T, raster = F)
DimPlot(XDP_Cohorts_1_2_full, group.by = "Region", label=T, raster = F)
```

```{r}
FeaturePlot(XDP_Cohorts_1_2_full, features = c("SYT1", "AQP4", "C1QA","FLT1"),raster=FALSE)
FeaturePlot(XDP_Cohorts_1_2_full, features = c("OLIG1", "OLIG2", "MOG" ,"CD96"),raster=FALSE) 
FeaturePlot(XDP_Cohorts_1_2_full, features = c("MBP", "MOBP", "WIF1", "TNC"),raster=FALSE)
FeaturePlot(XDP_Cohorts_1_2_full, features = c("RBFOX3", "CX3CR1", "GFAP" ,"AQP4"),raster=FALSE) 
FeaturePlot(XDP_Cohorts_1_2_full, features = c("GAD1", "GAD2", "SLC17A7" ,"SLC17A6"),raster=FALSE) 
FeaturePlot(XDP_Cohorts_1_2_full, features = c("pct_mito", "pct_intronic", "nUmi"),raster=FALSE) 
```

```{r}
XDP_Cohorts_1_2_full= qread("XDP_Cohorts_1_2_full_032825.qs")
XDP_Cohorts_1_2_full@meta.data
```


```{r}
Idents(XDP_Cohorts_1_2_full) = xdp_co1_neurons$SCT_snn_res.0.2

classes = c("oligo", "neuron", "astrocyte", "oligo", "oligo", "microglia", "opc", "neuron", "neuron", "oligo", "endothelial", "neuron", "neuron", "ependymal", "immune")


XDP_Cohorts_1_2_full= assignCellClasses(XDP_Cohorts_1_2_full, classes=classes, cluster_col="SCT_snn_res.0.2", class_col = "final_cell_class_merged_harmony")

DimPlot(XDP_Cohorts_1_2_full, group.by = "final_cell_class_merged_harmony" ,label = T, raster = FALSE)
Idents(XDP_Cohorts_1_2_full) = "final_cell_class_merged_harmony"
```

```{r}
XDP_Cohorts_1_2_full@meta.data
```

```{r}
#co2_more_metadata = read.csv("co_2_additional_metadata.csv")
co2_more_metadata$Sex = co2_more_metadata$Gender
co2_more_metadata

co2_more_metadata$Gender = NULL
co2_more_metadata$RNA.Volume = NULL
co2_more_metadata$RNA.Concentration = NULL
co2_more_metadata$DNA.Volume.extracted..ul. = NULL
co2_more_metadata$DNA.Concentration.extracted..ng.ul. = NULL
co2_more_metadata
```

```{r}
meta = XDP_Cohorts_1_2_full@meta.data
```


#added new metadata
```{r}
# Join without breaking rownames
meta<- left_join(XDP_Cohorts_1_2_full@meta.data, co2_more_metadata, 
                      by = c("donor_id" = "Donor"))
meta
rownames(meta) <- rownames(XDP_Cohorts_1_2_full@meta.data)
meta
```

```{r}
meta$RQS[meta$donor_id == "SCF_22_067CM"] = 6.08
meta$DV200[meta$donor_id == "SCF_22_067CM"] = 0.73
meta$MQS_numerical[meta$donor_id == "SCF_22_067CM"] = 5
meta$MQS[meta$donor_id == "SCF_22_067CM"] = "A"
meta$Age.of.Death[meta$donor_id == "SCF_22_067CM"] = 19
```

```{r}
meta$Age.of.Onset <- coalesce(meta$Age.of.Onset.x, meta$Age.of.Onset.y)
meta$Age.of.Death <- coalesce(meta$Age.of.Death.x, meta$Age.of.Death.y)
meta$RQS <- coalesce(meta$RQS.x, meta$RQS.y)
meta$DV200 <- coalesce(meta$DV200.x, meta$DV200.y)
meta$MQS_numerical <- coalesce(meta$MQS_numerical.x, meta$MQS_numerical.y)
meta$MQS <- coalesce(meta$MQS.x, meta$MQS.y)
```

```{r}
meta$Sex.y = NULL
meta$Sex = meta$Sex.x
meta$Sex.x = NULL
meta$Age.of.Onset.x = NULL
meta$Age.of.Onset.y = NULL
meta$Age.of.Death.x = NULL
meta$Age.of.Death.y = NULL
meta$RQS.x = NULL
meta$RQS.y = NULL
meta$DV200.x = NULL
meta$DV200.y = NULL
meta$MQS_numerical.x = NULL
meta$MQS_numerical.y = NULL
meta$MQS.x = NULL
meta$MQS.y = NULL
meta$sex = NULL
meta$Gender.x = NULL
meta$Gender.y = NULL
```

```{r}
meta
```

```{r}
XDP_Cohorts_1_2_full <- AddMetaData(XDP_Cohorts_1_2_full, metadata = meta)
```

```{r}
XDP_Cohorts_1_2_full@meta.data
```

```{r}
XDP_Cohorts_1_2_full@meta.data$region = NULL
XDP_Cohorts_1_2_full@meta.data$Disease_duration.x = NULL
XDP_Cohorts_1_2_full@meta.data$Disease_duration.y = NULL
XDP_Cohorts_1_2_full@meta.data$Gender = NULL
```

```{r}
XDP_Cohorts_1_2_full@meta.data$Disease_duration <- coalesce(XDP_Cohorts_1_2_full@meta.data$Disease_duration.x, XDP_Cohorts_1_2_full@meta.data$Disease_duration.y)
```



```{r}
DimPlot(XDP_Cohorts_1_2_full, group.by = "donor_id", raster = T)
DimPlot(XDP_Cohorts_1_2_full, group.by = "final_cell_class_merged_harmony", raster = T)
```

```{r}
qsave(XDP_Cohorts_1_2_full, "XDP_Cohorts_1_2_full_032825.qs")
```



#create masks for bican and xdp recons
#run findallmarkers for each region
#intersect genes in recons vs merged cohort1 and 2, rerun scores 
#plot by region 


```{r}
bican_recon = qread("Current_sobj/bican_recon_250327_sct_mask.qs")
bican_recon@meta.data
```

```{r}
ggplot(bican_recon@meta.data, aes(x = x_um, y = y_um, color =cell_class_annot)) +geom_point(alpha = 0.5, size = 0.3) + facet_wrap(~ cell_class_annot)
```


```{r}
#You can pull out your spatial coordinates and field to color by (ex. seurat clusters) from metadata into a separate df (xy = obj@meta.data[, c('x_um', 'y_um', 'seurat_clusters')]). Make sure your x and y columns are called 'x' and 'y' accordingly, so you may need to change the name and also omit NA values (unplaced cells)

xy = bican_recon@meta.data[, c('x_um', 'y_um', 'cell_class_annot')]
xy
xy = na.omit(xy)
xy$x = xy$x_um
xy$x_um = NULL
xy$y = xy$y_um
xy$y_um = NULL
xy$seurat_clusters = xy$cell_class_annot 
xy$cell_class_annot  = NULL

xy

```

```{r}
xy_neurons = subset(xy, subset = seurat_clusters == "neuron")
xy_neurons
```
```{r}
bican_recon <- bican_recon[, !is.na(bican_recon@meta.data$x_um)]
```


```{r}
#Draw a polygon around your ROI with poly <- anno_points(xy, lines=T, splines=F)
poly <- anno_points(xy_neurons, lines=T, splines=F)
```

```{r}
caudate_mask_bican <- in.bounds(xy, poly)
bican_recon@meta.data$bican_mask <- caudate_mask_bican
bican_recon
bican_recon@meta.data$CAP = bican_recon@meta.data$bican_mask
bican_recon@meta.data$CAP[bican_recon@meta.data$bican_mask == TRUE] = "Caudate"
bican_recon@meta.data
table(bican_recon$CAP)
```

```{r}
#Draw a polygon around your ROI with poly <- anno_points(xy, lines=T, splines=F)
poly <- anno_points(xy_neurons, lines=T, splines=F)
```

```{r}
putamen_mask_bican <- in.bounds(xy, poly)
bican_recon@meta.data$bican_mask2 <- putamen_mask_bican
bican_recon
bican_recon@meta.data$CAP[bican_recon@meta.data$bican_mask2 == TRUE] = "Putamen"
bican_recon@meta.data
table(bican_recon$CAP)
```
```{r}
#Draw a polygon around your ROI with poly <- anno_points(xy, lines=T, splines=F)
poly <- anno_points(xy_neurons, lines=T, splines=F)
```

```{r}
NAC_mask_bican <- in.bounds(xy, poly)
bican_recon@meta.data$bican_mask3 <- NAC_mask_bican
bican_recon
bican_recon@meta.data$CAP[bican_recon@meta.data$bican_mask3 == TRUE] = "Nucleus_Accumbens"
bican_recon@meta.data
table(bican_recon$CAP)
```
```{r}
#Draw a polygon around your ROI with poly <- anno_points(xy, lines=T, splines=F)
poly <- anno_points(xy_neurons, lines=T, splines=F)
```

```{r}
IC_mask_bican <- in.bounds(xy, poly)
bican_recon@meta.data$bican_mask4 <- IC_mask_bican
bican_recon
bican_recon@meta.data$CAP[bican_recon@meta.data$bican_mask4 == TRUE] = "Internal_Capsule"
bican_recon@meta.data
table(bican_recon$CAP)
```
```{r}
bican_recon@meta.data$CAP[bican_recon@meta.data$CAP == FALSE] = "Other"
table(bican_recon@meta.data$CAP)
```

```{r}
ggplot(bican_recon@meta.data, aes(x = x_um, y = y_um, color =CAP)) +geom_point(alpha = 0.5, size = 0.3) + facet_wrap(~ cell_class_annot)
```
```{r}
bican_recon@meta.data$CAP_cell_classes = paste0(bican_recon$CAP, "_", bican_recon$cell_class_annot)
table(bican_recon$CAP_cell_classes)
```

```{r}
Idents(bican_recon) = "CAP_cell_classes"
markers = FindAllMarkers(bican_recon, only.pos = TRUE, min.pct = 0.2, logfc.threshold = 1.25)
markers
markers_filtered = subset(markers, subset = pct.1 > 0.8)
markers_filtered
markers_filtered_2 = subset(markers_filtered, subset = pct.2 < 0.2)
markers_filtered_2
```

```{r}
table(markers_filtered_2$cluster)
```

```{r}
bican_recon@meta.data
qsave(bican_recon, "Current_sobj/bican_recon_250331_sct_mask.qs")
```

#xdp mask
```{r}
xdp_recon@meta.data
```

```{r}
ggplot(xdp_recon@meta.data, aes(x = x_um, y = y_um, color =new_cell_class)) +geom_point(alpha = 0.5, size = 0.3) + facet_wrap(~ new_cell_class)
```


```{r}
#You can pull out your spatial coordinates and field to color by (ex. seurat clusters) from metadata into a separate df (xy = obj@meta.data[, c('x_um', 'y_um', 'seurat_clusters')]). Make sure your x and y columns are called 'x' and 'y' accordingly, so you may need to change the name and also omit NA values (unplaced cells)

xy = xdp_recon@meta.data[, c('x_um', 'y_um', 'new_cell_class')]
xy
xy = na.omit(xy)
xy$x = xy$x_um
xy$x_um = NULL
xy$y = xy$y_um
xy$y_um = NULL
xy$seurat_clusters = xy$new_cell_class 
xy$new_cell_class  = NULL

xy

```
```{r}
xdp_recon <- xdp_recon[, !is.na(xdp_recon@meta.data$x_um)]
xdp_recon
```

```{r}
xy_neurons = subset(xy, subset = seurat_clusters == "neuron")
xy_neurons
```


```{r}
#Draw a polygon around your ROI with poly <- anno_points(xy, lines=T, splines=F)
poly <- anno_points(xy_neurons, lines=T, splines=F)
```

```{r}
caudate_mask_xdp <- in.bounds(xy, poly)
xdp_recon@meta.data$xdp_mask <- caudate_mask_xdp
xdp_recon
xdp_recon@meta.data$CAP = xdp_recon@meta.data$xdp_mask
xdp_recon@meta.data$CAP[xdp_recon@meta.data$xdp_mask == TRUE] = "Caudate"
xdp_recon@meta.data
table(xdp_recon$CAP)
```

```{r}
#Draw a polygon around your ROI with poly <- anno_points(xy, lines=T, splines=F)
poly <- anno_points(xy_neurons, lines=T, splines=F)
```

```{r}
putamen_mask_xdp <- in.bounds(xy, poly)
xdp_recon@meta.data$xdp_mask2 <- putamen_mask_xdp
xdp_recon
xdp_recon@meta.data$CAP[xdp_recon@meta.data$xdp_mask2 == TRUE] = "Putamen"
xdp_recon@meta.data
table(xdp_recon$CAP)
```

```{r}
#Draw a polygon around your ROI with poly <- anno_points(xy, lines=T, splines=F)
poly <- anno_points(xy_neurons, lines=T, splines=F)
```

```{r}
NAC_mask_xdp <- in.bounds(xy, poly)
xdp_recon@meta.data$xdp_mask3 <- NAC_mask_xdp
xdp_recon
xdp_recon@meta.data$CAP[xdp_recon@meta.data$xdp_mask3 == TRUE] = "Nucleus_Accumbens"
xdp_recon@meta.data
table(xdp_recon$CAP)
```

```{r}
#Draw a polygon around your ROI with poly <- anno_points(xy, lines=T, splines=F)
poly <- anno_points(xy_neurons, lines=T, splines=F)
```

```{r}
IC_mask_xdp <- in.bounds(xy, poly)
xdp_recon@meta.data$xdp_mask4 <- IC_mask_xdp
xdp_recon
xdp_recon@meta.data$CAP[xdp_recon@meta.data$xdp_mask4 == TRUE] = "Internal_Capsule"
xdp_recon@meta.data
table(xdp_recon$CAP)
```

```{r}
xdp_recon@meta.data$CAP[xdp_recon@meta.data$CAP == FALSE] = "Other"
table(xdp_recon@meta.data$CAP)
```

```{r}
ggplot(xdp_recon@meta.data, aes(x = x_um, y = y_um, color =CAP)) +geom_point(alpha = 0.5, size = 0.3) + facet_wrap(~ new_cell_class)
```{r}
xdp_recon@meta.data$CAP_cell_classes = paste0(xdp_recon$CAP, "_", xdp_recon$new_cell_class)
table(xdp_recon$CAP_cell_classes)
```

#intersect ucells
```{r}
bican_recon_sct_rownames = rownames(bican_recon)
length(bican_recon_sct_rownames)
```

```{r}
XDP_Cohorts_1_2_full = qread("XDP_Cohorts_1_2_full_032825.qs")
XDP_Cohorts_1_2_full

xdp_all_sct_rownames = rownames(XDP_Cohorts_1_2_full)
length(xdp_all_sct_rownames)
```

```{r}
xdp_recon = qread("xdp_recon_full_feb2025.qs")
xdp_recon
```
```{r}
xdp_recon_sct_rownames = rownames(xdp_recon)
length(xdp_recon_sct_rownames)
```



```{r}
library(UCell)

NFKB_genes =  c("MARCKS", "IL23A", "NINJ1", "TNFSF9", "SIK1", "ATF3", "SERPINE1", "MYC", "HES1", "CCNL1", "CCN1", "EGR1", "EGR2", "JAG1", "EGR3", "ABCA1", "GADD45B", "GADD45A", "PLK2", "KLF10", 
           "EIF1", "EHD1", "FOSL2", "FOSL1", "GPR183", "PLPP3", "IFIT2", "ICAM1", "ZC3H12A", "IER2", 
           "IL12B", "JUNB", "IER5", "IER3", "STAT5A", "DUSP5", "EDN1", "JUN", "DUSP4", "DUSP1", 
           "DUSP2", "TSC22D1", "CCL20", "SPHK1", "LIF", "IL18", "TUBB2A", "RHOB", "VEGFA", "PTPRE", 
           "IL1A", "TLR2", "IL1B", "BHLHE40", "ID2", "CLCF1", "REL", "FJX1", "SGK1", "BTG3", "BTG2", 
           "BTG1", "SDC4", "LITAF", "AREG", "SOCS3", "PANX1", "RIPK2", "NFIL3", "SERPINB2", "GCH1", 
           "IFNGR2", "G0S2", "FOS", "SERPINB8", "F3", "SPSB1", "FOSB", "PER1", "F2RL1", "HBEGF", 
           "CD44", "TRIP10", "CDKN1A", "PTGER4", "PTGS2", "IFIH1", "NAMPT", "OLR1", "ICOSLG", 
           "PHLDA1", "ZBTB10", "TAP1", "PNRC1", "CXCL10", "CXCL11", "IL6ST", "CD69", "SQSTM1", 
           "RELA", "CSF2", "CD83", "CSF1", "PPP1R15A", "CD80", "TNC", "TNF", "TANK", "RELB", "ZFP36", 
           "CCND1", "RNF19B", "CCRL2", "DENND5A", "PHLDA2", "MAP3K8", "LDLR", "SLC16A6", "SMAD3", 
           "TGIF1", "MAP2K3", "DDX58", "TRAF1", "INHBA", "NFKB1", "NFKB2", "GEM", "NR4A3", "MAFF", 
           "RCAN1", "NR4A2", "EFNA1", "MXD1", "BIRC2", "YRDC", "BIRC3", "IL7R", "PFKFB3", "IRS2", 
           "SLC2A3", "PLAU", "SLC2A6", "SAT1", "ETS2", "NR4A1", "SNN", "PMEPA1", "TNFRSF9", "MSC", 
           "TIPARP", "LAMB3", "GFPT2", "CFLAR", "TNIP1", "IRF1", "NFKBIA", "BMP2", "IL6", "TNIP2", 
           "BCL6", "BCL3", "NFKBIE", "NFE2L2", "B4GALT1", "NFAT5", "TNFAIP8", "BCL2A1", "TNFAIP6", 
           "TNFAIP3", "CXCL2", "CXCL1", "TNFAIP2", "CXCL3", "CXCL6", "FUT4", "DRAM1", "DNAJB4", 
           "PDE4B", "PDLIM5", "MCL1", "KDM6B", "IL15RA", "PLAUR", "ATP2B1", "KLF4", "KLF2", "SOD2", 
           "KLF9", "KLF6", "ACKR3", "PTX3", "B4GALT5", "TRIB1", "CEBPB", "CEBPD", "PLEK", "KYNU", 
           "CCL5", "CCL4", "CCL2")

interferon_alpha = c("IL4R", "CD74", "SP110", "MX1", "RSAD2", "TAP1", "PSMB9", "CXCL10", "CXCL11", "IFI44L", "LAP3", "LPAR6", "IFI27", "TRAFD1", "TRIM14", "LY6E", "IFITM2", "IFITM3", "SAMD9", "CSF1", 
           "IFITM1", "UBE2L6", "SAMD9L", "RNF31", "HERC6", "LAMP3", "CCRL2", "MVB12A", "B2M", "CMTR1", 
           "GBP2", "GBP4", "EPSTI1", "BATF2", "HLA-C", "TENT5A", "TDRD7", "ISG15", "IFI44", "PARP14", 
           "MOV10", "ISG20", "PARP12", "PLSCR1", "PROCR", "SELL", "PSME2", "PSME1", "CNP", "LGALS3BP", 
           "IFI35", "IFIT2", "IFI30", "DDX60", "IFIT3", "NUB1", "HELZ2", "PNPT1", "IL15", "NCOA7", 
           "EIF2AK2", "NMI", "OAS1", "BST2", "IRF1", "ELF1", "IL7", "IRF2", "IRF9", "IRF7", "UBA7", 
           "GMPR", "ADAR", "CASP8", "TRIM5", "RIPK2", "CASP1", "STAT2", "WARS1", "PARP9", "PSMA3", 
           "TXNIP", "CMPK2", "CD47", "RTP4", "C1S", "OGFR", "TMEM140", "USP18", "OASL", "SLC25A28", 
           "IFIH1", "PSMB8", "DHX58", "TRIM25", "TRIM26", "TRIM21")

antigen_processing_genes = c("CTSB", "CD74", "HSP90AA1", "KLRC3", "HSPA6", "HSPA5", "HSPA1L", "KLRC4", "HSPA8", "HSPA2", "TAP2", "HSPA4", "TAP1", "TAPBP", "CD4", "RFX5", "IFNG", "HSPA1A", "HLA-DRA", "HLA-DQB1", "KLRD1", "CALR", "HSPA1B", "TNF", "KIR2DL1", "KIR2DL2", "KIR2DL3", "KIR2DL4", "CTSS", "CTSL", "B2M", "HLA-DOB", "KIR2DL5A", "CIITA", "HLA-DQA2", "HLA-DOA", "HLA-DQA1", "PDIA3", "HLA-DRB5", 
"KIR2DS1", "NFYB", "KIR2DS2", "NFYC", "KIR2DS3", "KIR2DS4", "KIR2DS5", "HLA-C", "RFXANK", 
"KIR3DL1", "CD8B2", "KIR3DL2", "HLA-A", "HLA-B", "KIR3DL3", "HLA-G", "HLA-E", "HLA-F", "CREB1", "CD8B", "RFXAP", "CD8A", "CANX", "LGMN", "PSME2", "PSME3", "HLA-DPB1", "PSME1", "HLA-DRB4", 
"HLA-DRB3", "HLA-DRB1", "HSP90AB1", "IFI30", "HLA-DMB", "HLA-DPA1", "KLRC1", "KLRC2", "NFYA","HLA-DMA")
C_minus = qread("temp_disco/more_temp_1205/final_c_minus_genes.qs")
C_plus = qread("temp_disco/more_temp_1205/final_c_plus_genes.qs")

length(NFKB_genes)
NFKB_genes_new = intersect(NFKB_genes, bican_recon_sct_rownames)
length(NFKB_genes_new)
NFKB_genes_new = intersect(NFKB_genes_new, xdp_all_sct_rownames)
length(NFKB_genes_new)
NFKB_genes_new = intersect(NFKB_genes_new, xdp_recon_sct_rownames)
length(NFKB_genes_new)

length(interferon_alpha)
interferon_alpha_new = intersect(interferon_alpha, bican_recon_sct_rownames)
length(interferon_alpha_new)
interferon_alpha_new = intersect(interferon_alpha_new, xdp_all_sct_rownames)
length(interferon_alpha_new)
interferon_alpha_new = intersect(interferon_alpha_new, xdp_recon_sct_rownames)
length(interferon_alpha_new)

length(antigen_processing_genes)
antigen_processing_genes_new = intersect(antigen_processing_genes, bican_recon_sct_rownames)
length(antigen_processing_genes_new)
antigen_processing_genes_new = intersect(antigen_processing_genes_new, xdp_all_sct_rownames)
length(antigen_processing_genes_new)
antigen_processing_genes_new = intersect(antigen_processing_genes_new, xdp_recon_sct_rownames)
length(antigen_processing_genes_new)

length(C_minus)
C_minus_new = intersect(C_minus, bican_recon_sct_rownames)
length(C_minus_new)
C_minus_new = intersect(C_minus_new, xdp_all_sct_rownames)
length(C_minus_new)
C_minus_new = intersect(C_minus_new, xdp_recon_sct_rownames)
length(C_minus_new)

length(C_plus)
C_plus_new = intersect(C_plus, bican_recon_sct_rownames)
length(C_plus_new)
C_plus_new = intersect(C_plus_new, xdp_all_sct_rownames)
length(C_plus_new)
C_plus_new = intersect(C_plus_new, xdp_recon_sct_rownames)
length(C_plus_new)
```


```{r}
xdp_recon <- AddModuleScore_UCell(xdp_recon, features = list(Score = NFKB_genes_new), name = 'NFKB')

xdp_recon <- AddModuleScore_UCell(xdp_recon, features = list(Score = interferon_alpha_new), name = 'interferon_alpha'
)

xdp_recon <- AddModuleScore_UCell(xdp_recon,features = list(antigen_processing = antigen_processing_genes_new), name = '_scores')

xdp_recon <- AddModuleScore_UCell(xdp_recon,features = list(C_minus = C_minus_new), name = '_scores')
xdp_recon <- AddModuleScore_UCell(xdp_recon,features = list(C_plus = C_plus_new), name = '_scores')

qsave(xdp_recon, "xdp_recon_full_mar2025.qs")
```


```{r}
rm(xdp_recon)
```

```{r}
XDP_Cohorts_1_2_full = qread("XDP_Cohorts_1_2_full_032825.qs")
XDP_Cohorts_1_2_full@meta.data
```

```{r}
XDP_Cohorts_1_2_full <- AddModuleScore_UCell(XDP_Cohorts_1_2_full, features = list(Score = NFKB_genes_new), name = 'NFKB')

XDP_Cohorts_1_2_full <- AddModuleScore_UCell(XDP_Cohorts_1_2_full, features = list(Score = interferon_alpha_new), name = 'interferon_alpha'
)

XDP_Cohorts_1_2_full <- AddModuleScore_UCell(XDP_Cohorts_1_2_full,features = list(antigen_processing = antigen_processing_genes_new), name = '_scores')

XDP_Cohorts_1_2_full <- AddModuleScore_UCell(XDP_Cohorts_1_2_full,features = list(C_minus = C_minus_new), name = '_scores')
XDP_Cohorts_1_2_full <- AddModuleScore_UCell(XDP_Cohorts_1_2_full,features = list(C_plus = C_plus_new), name = '_scores')

qsave(XDP_Cohorts_1_2_full, "XDP_Cohorts_1_2_full_032825.qs")
```



```{r}
xdp_cohorts_meta = XDP_Cohorts_1_2_full@meta.data
rm(XDP_Cohorts_1_2_full)
bican_recon = qread("bican_recon_feb2025_sct.qs")
bican_recon@meta.data


bican_recon <- AddModuleScore_UCell(bican_recon, features = list(Score = NFKB_genes_new), name = 'NFKB')

bican_recon <- AddModuleScore_UCell(bican_recon, features = list(Score = interferon_alpha_new), name = 'interferon_alpha'
)

bican_recon <- AddModuleScore_UCell(bican_recon,features = list(antigen_processing = antigen_processing_genes_new), name = '_scores')

bican_recon <- AddModuleScore_UCell(bican_recon,features = list(C_minus = C_minus_new), name = '_scores')
bican_recon <- AddModuleScore_UCell(bican_recon,features = list(C_plus = C_plus_new), name = '_scores')

qsave(bican_recon, "bican_recon_mar2025_sct.qs")
```


```{r}
xdp_recon = qread("xdp_recon_full_mar2025.qs")
xdp_recon_meta = xdp_recon@meta.data
xdp_recon_meta
```


```{r}
neuron_df = select(xdp_recon_meta, "CAP")
neuron_df
```

```{r}
xdp_recon_meta$cell_id = rownames(xdp_recon_meta)
neuron_df$cell_id = rownames(neuron_df)
neuron_df
```

```{r}
xdp_recon_meta_new = left_join(xdp_recon_meta, neuron_df, by = "cell_id")
xdp_recon_meta_new
```

```{r}
rownames(bican_recon_meta_new) = bican_recon_meta_new$cell_id
```



```{r}
xdp_cohorts_meta = XDP_Cohorts_1_2_full@meta.data
```

```{r}
bican_recon_meta = bican_recon@meta.data
```

```{r}
xdp_recon@meta.data
```

```{r}
xdp_recon_meta = xdp_recon@meta.data
xdp_recon_meta
```


```{r}
xdp_cohorts_meta
bican_recon_meta
xdp_recon_meta
```

```{r}
xdp_recon_meta_new$Condition = "18_006"
```


```{r}
bican_recon = qread("Current_sobj/bican_recon_250331_sct_mask.qs")
bican_recon@meta.data
```
```{r}
neuron_df = select(bican_recon@meta.data, "GM_WM_IC", "CAP")
neuron_df
```

```{r}
bican_recon_meta$cell_id = rownames(bican_recon_meta)
neuron_df$cell_id = rownames(neuron_df)
neuron_df
```

```{r}
bican_recon_meta_new = left_join(bican_recon_meta, neuron_df, by = "cell_id")
bican_recon_meta_new
```

```{r}
rownames(bican_recon_meta_new) = bican_recon_meta_new$cell_id

bican_recon_meta_new$donor_id = "BICAN_donor"
bican_recon_meta_new$Condition = "Control_BICAN"
bican_recon_meta_new
```

```{r}
neuron_df = select(XDP_neurons_Cohorts_1_2@meta.data, "final_neuron_subclass_merged")
neuron_df
```

```{r}
xdp_cohorts_meta$cell_id = rownames(xdp_cohorts_meta)
neuron_df$cell_id = rownames(neuron_df)
```

```{r}
xdp_cohorts_meta_new = left_join(xdp_cohorts_meta, neuron_df, by = "cell_id")
xdp_cohorts_meta_new
```


```{r}
xdp_cohorts_meta_new$final_neuron_subclass_merged[xdp_cohorts_meta_new$final_cell_class_merged_harmony == "oligo"] = "oligo" 
xdp_cohorts_meta_new$final_neuron_subclass_merged[xdp_cohorts_meta_new$final_cell_class_merged_harmony == "opc"] = "opc" 
xdp_cohorts_meta_new$final_neuron_subclass_merged[xdp_cohorts_meta_new$final_cell_class_merged_harmony == "astrocyte"] = "astrocyte" 
xdp_cohorts_meta_new$final_neuron_subclass_merged[xdp_cohorts_meta_new$final_cell_class_merged_harmony == "ependymal"] = "ependymal" 
xdp_cohorts_meta_new$final_neuron_subclass_merged[xdp_cohorts_meta_new$final_cell_class_merged_harmony == "endothelial"] = "endothelial" 
xdp_cohorts_meta_new$final_neuron_subclass_merged[xdp_cohorts_meta_new$final_cell_class_merged_harmony == "immune"] = "immune" 
xdp_cohorts_meta_new$final_neuron_subclass_merged[xdp_cohorts_meta_new$final_cell_class_merged_harmony == "microglia"] = "microglia" 
```

```{r}
xdp_cohorts_meta_new
rownames(xdp_cohorts_meta_new) = xdp_cohorts_meta_new$cell_id
xdp_cohorts_meta_new
```

```{r}
table(xdp_cohorts_meta_new$donor_id, xdp_cohorts_meta_new$final_neuron_subclass_merged)
```


```{r}
library(dplyr)
library(purrr)

common_cols = c("donor_id", "nUmi", "nUMI", "pct_mito", "percent.mt" ,"nCount_RNA", "nFeature_RNA", "library", "nCount_SCT", "nFeature_SCT", "Sex", "Condition", "Age.of.Onset", "Age.of.Death", "Disease_duration", "Region", "final_cell_class_merged_harmony", "neuron_classes", "Cohort", "cell_class_annot", "reclustered_patch_matrix_exotic", "x_um", "y_um", "x_um_rotated", "y_um_rotated", "new_reclustered_patch_matrix_exotic", "new_cell_class","ScoreNFKB", "Scoreinterferon_alpha", "C_minus_scores", "C_plus_scores", "antigen_processing_scores","final_neuron_subclass_merged", "CAP", "GM_WM_IC")

# Ensure each dataframe has all columns in column_vector, filling missing ones with NA
dfs_filled <- lapply(list(xdp_cohorts_meta_new, bican_recon_meta_new, xdp_recon_meta_new), function(df) {
  df[common_cols[!(common_cols %in% colnames(df))]] <- NA  # Add missing columns with NA
  return(df[common_cols])  # Reorder columns to match the vector
})

# Merge all dataframes by row (since row names are different)
everything_metadata <- bind_rows(dfs_filled)

everything_metadata

everything_metadata$nUmi = coalesce(everything_metadata$nUmi, everything_metadata$nUMI)
everything_metadata$nUMI = NULL
everything_metadata$pct_mito = coalesce(everything_metadata$pct_mito, everything_metadata$percent.mt)
everything_metadata$percent.mt = NULL

everything_metadata$ALL_final_cell_class = coalesce(everything_metadata$final_cell_class_merged_harmony, everything_metadata$cell_class_annot,everything_metadata$new_cell_class)

everything_metadata$ALL_neuron_classes = coalesce(everything_metadata$final_neuron_subclass_merged, everything_metadata$reclustered_patch_matrix_exotic,everything_metadata$new_reclustered_patch_matrix_exotic)


everything_metadata$CAP_Region = coalesce(everything_metadata$Region, everything_metadata$CAP)
everything_metadata

everything_metadata$ALL_final_cell_class[everything_metadata$ALL_final_cell_class == "astro"] = "astrocyte"
everything_metadata$ALL_final_cell_class[everything_metadata$ALL_final_cell_class == "endo"] = "endothelial"
everything_metadata$ALL_final_cell_class[everything_metadata$ALL_final_cell_class == "mg"] = "microglia"

everything_metadata$ALL_neuron_classes[everything_metadata$ALL_neuron_classes == "matrix_SPN"] = "SPN_matrix" 
everything_metadata$ALL_neuron_classes[everything_metadata$ALL_neuron_classes == "patch_SPN"] = "SPN_patch" 
everything_metadata$ALL_neuron_classes[everything_metadata$ALL_neuron_classes == "interneuron"] = "non_SPN" 
everything_metadata$ALL_neuron_classes[everything_metadata$ALL_neuron_classes == "eSPN"] = "non_SPN" 
```

```{r}
table(everything_metadata$donor_id, everything_metadata$ALL_final_cell_class)

table(everything_metadata$donor_id, everything_metadata$ALL_neuron_classes)

```

#add donorid for recons!

```{r}
everything_metadata$CAP[is.na(everything_metadata$CAP)] <- "Unplaced"
```


```{r}
everything_metadata %>%
  group_by(donor_id, CAP) %>%
  summarise(n = n())
```
```{r}
table(everything_metadata$Cohort,everything_metadata$CAP)
```

```{r}
everything_metadata$CAP[everything_metadata$Region == "Caudate"] = "Caudate"
everything_metadata$CAP[everything_metadata$Region == "Putamen"] = "Putamen"
```


#plot scores
```{r}
everything_metadata$Condition[everything_metadata$Condition == "18_006"] = "XDP_18_006"
```

```{r}
everything_metadata$Cohort[everything_metadata$Cohort == "Control_BICAN"] = "RECON"
everything_metadata$Cohort[everything_metadata$Cohort == "XDP_18_006"] = "RECON"
```


```{r}
unique(everything_metadata$Cohort)
```

```{r}
everything_metadata$Cohort_Condition = paste0(everything_metadata$Cohort, "_", everything_metadata$Condition)
unique(everything_metadata$Cohort_Condition)
```




```{r}
histograms_by_all_celltype(XDP_Cohorts_1_2_full@meta.data, "ScoreNFKB", "ScoreNFKB", 25, 5)
```

```{r}
histograms_by_all_celltype_cohort(XDP_Cohorts_1_2_full@meta.data, "ScoreNFKB", "ScoreNFKB", 30, 5)
```

```{r}
histograms_by_donor(XDP_Cohorts_1_2_full@meta.data, "ScoreNFKB", "astrocyte" ,"ScoreNFKB", 30, 15)
```

```{r}
qsave(everything_metadata, "everything_metadata.qs")
everything_metadata
```


```{r}
final_df = everything_metadata
final_df =subset(final_df, subset = ALL_final_cell_class == "astrocyte") 
final_df
regions = unique(everything_metadata$CAP_Region)

for (region in regions) {
  
new_df =subset(final_df, subset = CAP_Region == region) 
new_df
xlab = "ScoreNFKB"
min = min(new_df$ScoreNFKB)
max = max(new_df$ScoreNFKB)
step = (max-min)/100

limits = seq(min, max, step)
  
a = plot_overlapping_density_histogram(df = new_df, 
                                          hist_col = new_df$ScoreNFKB,
                                          fill_col = "Cohort_Condition",
                                    colors = c("Cohort_1_XDP" = "red", "Cohort_2_XDP" = "orange", "RECON_XDP_18_006" = "pink", "Cohort_1_Control" = "blue","Cohort_2_Control" = "cyan", "RECON_Control_BICAN" = "green"),
                                          breaks = limits,
                                          title = paste0(region),
                                          xlab = xlab,
                                          fig_filename = NULL) +facet_wrap(~ Cohort_Condition)
  
  

# Display or save the final plot
print(a)
#ggsave("combined_plot.png", a, width = 7, height =5 )
}
```

```{r}
final_df = everything_metadata
final_df =subset(final_df, subset = ALL_final_cell_class == "astrocyte") 
final_df
regions = unique(everything_metadata$CAP_Region)

for (region in regions) {
  
new_df =subset(final_df, subset = CAP_Region == region) 
new_df
xlab = "ScoreNFKB"
min = min(new_df$ScoreNFKB)
max = max(new_df$ScoreNFKB)
step = (max-min)/100

limits = seq(min, max, step)
  
a = plot_overlapping_density_histogram(df = new_df, 
                                          hist_col = new_df$ScoreNFKB,
                                          fill_col = "Cohort_Condition",
                                         colors = c("Cohort_1_XDP" = "red", "Cohort_2_XDP" = "orange", "RECON_XDP_18_006" = "pink", "Cohort_1_Control" = "blue","Cohort_2_Control" = "cyan", "RECON_Control_BICAN" = "green"),
                                          breaks = limits,
                                          title = paste0(region),
                                          xlab = xlab,
                                          fig_filename = NULL) +facet_wrap(~ Cohort)
  
  

# Display or save the final plot
print(a)
#ggsave("combined_plot.png", a, width = 7, height =5 )
}
```


```{r}
final_df = everything_metadata
final_df =subset(final_df, subset = ALL_final_cell_class == "astrocyte") 
final_df
regions = unique(everything_metadata$CAP_Region)

for (region in regions) {
  
new_df =subset(final_df, subset = CAP_Region == region) 
new_df
xlab = "ScoreNFKB"
min = min(new_df$ScoreNFKB)
max = max(new_df$ScoreNFKB)
step = (max-min)/100

limits = seq(min, max, step)
  
a = plot_overlapping_density_histogram(df = new_df, 
                                          hist_col = new_df$ScoreNFKB,
                                          fill_col = "Condition",
                                         colors = c("Cohort_1_XDP" = "red", "Cohort_2_XDP" = "orange", "RECON_XDP_18_006" = "pink", "Cohort_1_Control" = "blue","Cohort_2_Control" = "cyan", "RECON_Control_BICAN" = "green"),
                                          breaks = limits,
                                          title = paste0(region),
                                          xlab = xlab,
                                          fig_filename = NULL) +facet_wrap(~ Condition)
  
  

# Display or save the final plot
print(a)
#ggsave("combined_plot.png", a, width = 7, height =5 )
}
```












```{r}
plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$ScoreNFKB,
                                          fill_col = "Condition_Cohort",
                                           colors = c("XDP_Cohort_1" = "red", "Control_Cohort_1" = "blue","XDP_Cohort_2" = "orange", "Control_Cohort_2" = "green"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL) 

plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$ScoreNFKB,
                                          fill_col = "Condition_Cohort",
                                           colors = c("XDP_Cohort_1" = "red", "Control_Cohort_1" = "blue","XDP_Cohort_2" = "orange", "Control_Cohort_2" = "green"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL) + facet_wrap(~ Cohort)

plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$ScoreNFKB,
                                          fill_col = "Condition_Cohort",
                                           colors = c("XDP_Cohort_1" = "red", "Control_Cohort_1" = "blue","XDP_Cohort_2" = "orange", "Control_Cohort_2" = "green"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL) + facet_wrap(~ Condition)

  

```


```{r}
histograms_by_donor <- function(final_df, score_col, celltype, xlab, width_size, height_size) {
  
  unique_donors <- unique(final_df$donor_id)

  # Identify controls and cases
  controls <- unique_donors[grepl("CM$|CF$|CM2$", unique_donors)]
  cases <- unique_donors[!grepl("CM$|CF$|CM2$", unique_donors)]

  # Alphabetically sort both groups
  controls <- sort(controls)
  cases <- sort(cases)

  # Combine ordered donor IDs (cases first, then controls)
  ordered_donors <- c(cases, controls)
  
  print(ordered_donors)

  plots <- list()

  # Subset dataframe for the specific cell type
  final_df_cellclass <- subset(final_df, final_neuron_cell_class_harmony_merged == celltype)

  # Determine global x-axis limits (bins)
  min_val <- min(final_df_cellclass[[score_col]], na.rm = TRUE)
  max_val <- max(final_df_cellclass[[score_col]], na.rm = TRUE)
  step <- (max_val - min_val) / 100
  limits <- seq(min_val, max_val, step)

  # **Determine global y-axis limit** by finding the maximum count across all donors
  max_peak_y <- 0  # Initialize
  
   for (donor in ordered_donors) {
    df_sub <- final_df_cellclass[final_df_cellclass$donor_id == donor, ]
    if (nrow(df_sub) > 0) {
      hist_data <- hist(df_sub[[score_col]], breaks = limits, plot = FALSE)
      max_peak_y <- max(max_peak_y, max(hist_data$density, na.rm = TRUE))  # Use density (peak) instead of total count
    }
  }
  
  # **Now generate histograms with consistent y-axis limits**
  for (donor in ordered_donors) {
    df_sub <- final_df_cellclass[final_df_cellclass$donor_id == donor, ]
    
    if (nrow(df_sub) == 0) {
      message(paste("No data for donor:", donor, "- skipping."))
      next
    }

    cell_count <- nrow(df_sub)
    
    a <- plot_overlapping_density_histogram(
      df = df_sub, 
      hist_col = df_sub[[score_col]],
      fill_col = "Condition_Cohort",
      colors = c("XDP_Cohort_1" = "red", "Control_Cohort_1" = "blue","XDP_Cohort_2" = "orange", "Control_Cohort_2" = "green"),
      breaks = limits,
      title = paste0(donor, ": ", celltype, " (Cells: ", cell_count, ")"),
      xlab = xlab,
      fig_filename = NULL
    )
    
    # Modify the y-axis limit directly using ggplot2
    a <- a + ggplot2::ylim(0, max_peak_y)  # Set consistent y-axis limit
    
    plots[[donor]] <- a
  }
  
  final_plot <- plot_grid(plotlist = plots, ncol = 5)
  
  print(final_plot)  
  ggsave("combined_plot.png", final_plot, width = width_size, height = height_size)
}

```

```{r}
XDP_Cohorts_1_2_full@meta.data
XDP_Cohorts_1_2_full <- AddModuleScore_UCell(XDP_Cohorts_1_2_full, features = list(Score = interferon_alpha), name = 'interferon_alpha')
```


```{r}
histograms_by_donor(XDP_Cohorts_1_2_full@meta.data, "Scoreinterferon_alpha", "astrocyte" ,"Scoreinterferon_alpha", 30, 15)
```

```{r}
final_df = XDP_Cohorts_1_2_full@meta.data
xlab = "Scoreinterferon_alpha"
min = min(final_df$Scoreinterferon_alpha)
max = max(final_df$Scoreinterferon_alpha)
step = (max-min)/100

limits = seq(min, max, step)
  
a = plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$Scoreinterferon_alpha,
                                          fill_col = "Condition",
                                          colors = c("XDP" = "red", "Control" = "blue","BICAN_V8" = "green", "pd" = "red", "ctr" = "blue", "XDP_18_006" = "orange"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL)
  
  

# Display or save the final plot
ggsave("combined_plot.png", a, width = 7, height =5 )

```


```{r}
plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$Scoreinterferon_alpha,
                                          fill_col = "Condition_Cohort",
                                           colors = c("XDP_Cohort_1" = "red", "Control_Cohort_1" = "blue","XDP_Cohort_2" = "orange", "Control_Cohort_2" = "green"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL) 

plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$Scoreinterferon_alpha,
                                          fill_col = "Condition_Cohort",
                                           colors = c("XDP_Cohort_1" = "red", "Control_Cohort_1" = "blue","XDP_Cohort_2" = "orange", "Control_Cohort_2" = "green"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL) + facet_wrap(~ Cohort)

plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$Scoreinterferon_alpha,
                                          fill_col = "Condition_Cohort",
                                           colors = c("XDP_Cohort_1" = "red", "Control_Cohort_1" = "blue","XDP_Cohort_2" = "orange", "Control_Cohort_2" = "green"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL) + facet_wrap(~ Condition)

  

```


































```{r}
bican_recon_temp = subset(bican_recon, subset =cell_class_annot == "neuron")
bican_recon_temp
```

```{r}
Idents(bican_recon_temp) = "CAP_cell_classes"
markers = FindAllMarkers(bican_recon_temp, only.pos = TRUE, min.pct = 0.2, logfc.threshold = 2)
markers
markers_filtered = subset(markers, subset = pct.1 > 0.6)
markers_filtered
markers_filtered_2 = subset(markers_filtered, subset = pct.2 < 0.3)
markers_filtered_2
```

```{r}
NAC_neuron_markers = subset(markers, subset = cluster == "Nucleus_Accumbens_neuron")
NAC_neuron_markers
```

```{r}
NAC_neuron_markers_genes = rownames(NAC_neuron_markers)
```

```{r}
bican_recon <- AddModuleScore_UCell(bican_recon, features = list(Score = NAC_neuron_markers_genes), name = 'NAC_neurons')

bican_recon@meta.data
```

```{r}
ggplot(bican_recon@meta.data, aes(x= x_um, y = y_um, color = ScoreNAC_neurons)) + geom_point(size = 0.2) + scale_color_gradientn(colors = c("gray","#FFD700", "#87CEEB", "#4682B4", "#0000FF")) +   labs(color = "NAC Scores") +  theme_void() + ylab(NULL) + facet_wrap(~ cell_class_annot)

```















```{r}
bican_recon_astro = subset(bican_recon, subset = cell_class_annot == "astro")
bican_recon_astro

ggplot(bican_recon_astro@meta.data, aes(x = x_um, y = y_um, color =cell_class_annot)) +geom_point(alpha = 0.5, size = 0.3) + facet_wrap(~ GM_WM_IC)
```

```{r}
Idents(bican_recon_astro) = "GM_WM_IC"
markers = FindAllMarkers(bican_recon_astro, only.pos = TRUE, min.pct = 0.2, logfc.threshold = 1.25)
markers
markers_filtered = subset(markers, subset = pct.1 > 0.6)
markers_filtered
markers_filtered_2 = subset(markers_filtered, subset = pct.2 < 0.3)
markers_filtered_2
```

```{r}
DefaultAssay(bican_recon) = "SCT"
```


```{r}
#FOS, GADD45B, HSPB1

gene_of_interest = "SRRM3" 

#replace seurat_obj here
metadata_df = bican_recon@meta.data
metadata_df$cell_ids = rownames(metadata_df)

#replace seurat_obj here
gene_df = as.data.frame(FetchData(bican_recon, vars = gene_of_interest))
gene_df$cell_ids = rownames(gene_df)

df_final = merge(metadata_df, gene_df, by = "cell_ids" )
df_final

#Replace GENE with gene name
ggplot(df_final, aes(x= x_um, y = y_um, color = SRRM3)) + geom_point() +  scale_color_gradient(low = "lightgrey", high = "blue") +  labs(color = "Expression") +
  theme_void() +  facet_wrap(~ cell_class_annot) +  ylab(NULL)
```















```{r}
#Generate a mask for the polygon with 
mask <- in.bounds(xy, poly)
mask

#qsave(mask, "bican_greymatter_mask.qs")
```

```{r}
#bican_recon <- bican_recon[, !is.na(bican_recon@meta.data$x_um)]
bican_recon@meta.data$gray_matter_roi <- mask
bican_recon
```

```{r}
bican_recon@meta.data$GM_WM = bican_recon@meta.data$gray_matter_roi
bican_recon@meta.data$GM_WM[bican_recon@meta.data$GM_WM == TRUE] = "Gray_Matter"
bican_recon@meta.data$GM_WM[bican_recon@meta.data$GM_WM == FALSE] = "White_Matter"
bican_recon@meta.data
```

```{r}
ggplot(bican_recon@meta.data, aes(x = x_um, y = y_um, color =GM_WM)) +geom_point(alpha = 0.5, size = 0.3) + facet_wrap(~ cell_class_annot)
```

```{r}
bican_recon_astro = subset(bican_recon, subset = cell_class_annot == "astro")
bican_recon_astro
```
```{r}
ggplot(bican_recon_astro@meta.data, aes(x = x_um, y = y_um, color =cell_class_annot)) +geom_point(alpha = 0.5, size = 0.3) + facet_wrap(~ GM_WM)
```
```{r}
Idents(bican_recon_astro) = "GM_WM"
markers = FindAllMarkers(bican_recon_astro, only.pos = TRUE, min.pct = 0.2, logfc.threshold = 1.25)
```
```{r}
markers_filtered = subset(markers, subset = pct.1 > 0.7)
markers_filtered
markers_filtered_2 = subset(markers_filtered, subset = pct.2 < 0.3)
markers_filtered_2
```
```{r}
FeaturePlot(bican_recon, features = c("WIF1", "SLIT1", "MIAT", "ENSG00000287544"), raster = F)
FeaturePlot(bican_recon, features = c("TMEM132C", "LINC00499", "PTCH1"), raster = F)

FeaturePlot(bican_recon, features = c("ENSG00000287704", "TNC", "DCLK1", "ADAMTSL3"), raster = F)
FeaturePlot(bican_recon, features = c("SLC24A4", "GRIK4", "DLGAP1"), raster = F)
```


```{r}
DimPlot(bican_recon, group.by = "cell_class_annot", raster = F)
DimPlot(bican_recon, group.by = "GM_WM", raster = F)
FeaturePlot(bican_recon, features = c("WIF1", "TNC"), raster = F)
```

```{r}
gene_of_interest = "WIF1" 

#replace seurat_obj here
metadata_df = bican_recon@meta.data
metadata_df$cell_ids = rownames(metadata_df)

#replace seurat_obj here
gene_df = as.data.frame(FetchData(bican_recon, vars = gene_of_interest))
gene_df$cell_ids = rownames(gene_df)

df_final = merge(metadata_df, gene_df, by = "cell_ids" )
df_final

#Replace GENE with gene name
ggplot(df_final, aes(x= x_um, y = y_um, color = WIF1)) + geom_point() +  scale_color_gradient(low = "lightgrey", high = "blue") +  labs(color = "Expression") +
  theme_void() +  facet_wrap(~ cell_class_annot) +  ylab(NULL)



gene_of_interest = "TNC" 

#replace seurat_obj here
metadata_df = bican_recon@meta.data
metadata_df$cell_ids = rownames(metadata_df)

#replace seurat_obj here
gene_df = as.data.frame(FetchData(bican_recon, vars = gene_of_interest))
gene_df$cell_ids = rownames(gene_df)

df_final = merge(metadata_df, gene_df, by = "cell_ids" )
df_final

#Replace GENE with gene name
ggplot(df_final, aes(x= x_um, y = y_um, color = TNC)) + geom_point() +  scale_color_gradient(low = "lightgrey", high = "blue") +  labs(color = "Expression") +
  theme_void() +  facet_wrap(~ cell_class_annot)+  ylab(NULL)
```



















































```{r}
qsave(XDP_Cohorts_1_2_full, "XDP_Cohorts_1_2_full_032425.qs")
```


#Testing out UCell

```{r}
DefaultAssay(sct_cleaned_CaH_Put_Cohort2) = "RNA"
temp_sobj_RNA = sct_cleaned_CaH_Put_Cohort2
temp_sobj_RNA <- AddModuleScore_UCell(temp_sobj_RNA, features = list(Score = NFKB_genes), name = 'NFKB_RNA')

temp_astro = subset(temp_sobj_RNA, subset = final_cell_class_merged == "astrocyte")
temp_astro =  AddModuleScore_UCell(temp_astro, features = list(Score = NFKB_genes), name = 'NFKB_RNA_astro_only')

# this code should take 3-10 minutes
library(sctransform)

temp_sobj_RNA = SCTransform(temp_sobj_RNA, vars.to.regress = "pct_mito", verbose = FALSE)
DefaultAssay(temp_sobj_RNA) = "SCT"

temp_sobj_RNA <- AddModuleScore_UCell(temp_sobj_RNA, features = list(Score = NFKB_genes), name = 'NFKB_SCT')

temp_astro_SCT = subset(temp_sobj_RNA, subset = final_cell_class_merged == "astrocyte")
temp_astro_SCT =  AddModuleScore_UCell(temp_astro_SCT, features = list(Score = NFKB_genes), name = 'NFKB_SCT_astro_only')
```
```{r}
temp_sobj_RNA@meta.data
temp_astro@meta.data
temp_astro_SCT@meta.data
```

```{r}
ggplot(temp_sobj_RNA@meta.data, aes(x = ScoreNFKB_RNA, y = ScoreNFKB_SCT, color = final_cell_class_merged)) + geom_point(size = 0.5, alpha = 0.5) + xlab("RNA assay NFKB Scores") + ylab("SCT assay NFKB Scores")

ggplot(temp_astro@meta.data, aes(x = ScoreNFKB_RNA, y = ScoreNFKB_RNA_astro_only, color = final_cell_class_merged)) + geom_point(size = 0.5, alpha = 0.5)+ xlab("NFKB Scores - all cell classes") + ylab("NFKB Scores - astrocytes subset")

ggplot(temp_astro_SCT@meta.data, aes(x = ScoreNFKB_SCT, y = ScoreNFKB_SCT_astro_only, color = final_cell_class_merged)) + geom_point(size = 0.5, alpha = 0.5)

ggplot(temp_sobj_RNA@meta.data, aes(x = ScoreNFKB, y = ScoreNFKB_SCT, color = final_cell_class_merged)) + geom_point(size = 0.5, alpha = 0.5) + xlab("Original NFKB Scores") + ylab("SCT NFKB Scores")

```
```{r}
temp_df = merge(final_df, temp_sobj_RNA@meta.data, by = "cell")
temp_df

ggplot(temp_sobj_RNA@meta.data, aes(x = ScoreNFKB, y = ScoreNFKB_SCT, color = final_cell_class_merged)) + geom_point(size = 0.5, alpha = 0.5)
```


#Subclustering astrocytes??
```{r}
DimPlot(XDP_Cohorts_1_2_full, label = T)
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.05", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.1", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.2", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.3", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.4", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.5", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.6", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.7", label=T) 
DimPlot(XDP_Cohorts_1_2_full, group.by = "SCT_snn_res.0.8", label=T)
```
```{r}
FeaturePlot(XDP_Cohorts_1_2_full, features = c("VIM", "GFAP", "IGFBP5","DBI"),raster=FALSE)
FeaturePlot(XDP_Cohorts_1_2_full, features = c("LIMA1", "MARCKS", "MARCKSL1", "RACK1"),raster=FALSE)
FeaturePlot(XDP_Cohorts_1_2_full, features = c("GRIA2", "SLC7A10", "FGFR3" ,"VEGFA"),raster=FALSE) 
```

```{r}
marker_genes <- c("AQP4", "VIM", "GFAP", "IGFBP5","DBI","LIMA1", "MARCKS", "MARCKSL1", "RACK1", "GRIA2", "SLC7A10", "FGFR3" ,"VEGFA")

Dotplot = DotPlot(object = XDP_Cohorts_1_2_full, features = marker_genes, group.by = "SCT_snn_res.0.3")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)


marker_genes <- c("WIF1", "SLIT1", "TMEM132C", "PTCH1",  "TNC", "DCLK1", "ADAMTSL3","SLC24A4", "GRIK4", "DLGAP1")

Dotplot = DotPlot(object = XDP_Cohorts_1_2_full, features = marker_genes, group.by = "SCT_snn_res.0.3")
Dotplot  <- Dotplot  + theme(axis.text.x = element_text(angle = 45, hjust = 1))+ coord_flip() 
print(Dotplot)
```
#Let's get markers from bican 
```{r}
FeaturePlot(XDP_Cohorts_1_2_full, features = c("WIF1", "TNC", "LMO2"),raster=FALSE)
#Grey
#No LINC00499,MIAT,ENSG00000287544
FeaturePlot(XDP_Cohorts_1_2_full, features = c("WIF1", "SLIT1", "TMEM132C", "PTCH1"), raster = F)
#White
#No ENSG00000287704
FeaturePlot(XDP_Cohorts_1_2_full, features = c("ENSG00000287704", "TNC", "DCLK1", "ADAMTSL3"), raster = F)
FeaturePlot(XDP_Cohorts_1_2_full, features = c("SLC24A4", "GRIK4", "DLGAP1"), raster = F)
```

```{r}
Idents(XDP_Cohorts_1_2_full) = XDP_Cohorts_1_2_full$RNA_snn_res.0.3

classes = c("GM_astrocytes", "WM_astrocytes", "WM_astrocytes", "WM_astrocytes", "WM_astrocytes", "WM_astrocytes", "WM_astrocytes", "WM_astrocytes", "WM_astrocytes")

XDP_Cohorts_1_2_full= assignCellClasses(XDP_Cohorts_1_2_full, classes=classes, cluster_col="RNA_snn_res.0.3", class_col = "GM_WM_astrocytes")

DimPlot(XDP_Cohorts_1_2_full, group.by = "GM_WM_astrocytes" ,label = T, raster = FALSE)
```

```{r}
XDP_Cohorts_1_2_full@meta.data
```


```{r}
cells_per_donor = as.data.frame(table(XDP_Cohorts_1_2_full@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(XDP_Cohorts_1_2_full@meta.data$GM_WM_astrocytes, XDP_Cohorts_1_2_full@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df

cell_class_df$Cohort <- XDP_Cohorts_1_2_full@meta.data$Cohort[match(cell_class_df$Var2, XDP_Cohorts_1_2_full@meta.data$donor_id)]
cell_class_df

# Count the number of cells per donor
donor_counts <- table(XDP_Cohorts_1_2_full@meta.data$donor_id)

# Convert to dataframe for sorting
donor_df <- as.data.frame(donor_counts)
colnames(donor_df) <- c("donor_id", "cell_count")

# Identify controls (CM, CF) - ensuring it only checks ENDING
controls <- donor_df[grepl("CM$|CF$|CM2$", donor_df$donor_id), ]
cases <- donor_df[!grepl("CM$|CF$|CM2$", donor_df$donor_id), ]

# Sort both groups by cell count in descending order
controls <- controls[order(-controls$cell_count), ]
cases <- cases[order(-cases$cell_count), ]

# Combine ordered donor IDs
ordered_donors <- c(controls$donor_id, cases$donor_id)

# Print or use the ordered donor IDs
print(ordered_donors)


            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = ordered_donors)
# Extract Cohort information
cell_class_df <- cell_class_df %>%
  arrange(Cohort, match(Var2, ordered_donors)) %>%
  mutate(Var2 = factor(Var2, levels = unique(Var2))) 


cell_class_df
ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors") + ylab("Number of cells by Astrocyte Subclasses") + labs(fill = "Astrocyte Subclass") +theme(axis.text.x = element_text(angle = 45, hjust = 1))  +facet_grid(~ Cohort, scales = "free_x", space = "free_x")

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors") + ylab("Astrocyte Subclass Proportion") + labs(fill = "Astrocyte Subclass")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))   + facet_grid(~ Cohort, scales = "free_x", space = "free_x")
```

```{r}
ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors") + ylab("Number of cells by Astrocyte Subclass") + labs(fill = "Astrocyte Subclass")+ geom_vline(xintercept =  12.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors") + ylab("Astrocyte Subclass Proportion") + labs(fill = "Astrocyte Subclass")+ geom_vline(xintercept =  12.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  

```

```{r}
library(lme4)
sobj = XDP_Cohorts_1_2_full
sobj = subset(sobj, donor_id != "SCF_22_049CCF")
model_cols = c("donor_id", "GM_WM_astrocytes", "Condition", "Gender", "Age.at.Death") #include any other covariates of interest, replace cell_class with whatever column defines the cluster annotations
pd = sobj@meta.data[,model_cols]
pd = pd[complete.cases(pd),] # don't want any NAs
pd$case_control_factor = as.factor(pd$Condition)

masc_df = .sconline.MASCfn(
    dataset=pd,
    cluster=pd$GM_WM_astrocytes, # cluster annotations
    contrast="case_control_factor", # name of contrast annotations (what you want to run the test for)
    random_effects=c("donor_id"), # name of random effects annotations (not interested in these coefficients, but account for variability in probable sources ob batch effects, in this case donor
    fixed_effects = c("Age.at.Death", "Gender") # your covariates
)

masc_df

my_order <- unique(masc_df$cluster)
masc_df$cluster_name <- sub("cluster", "", masc_df$cluster) # for legibility
masc_df$log2_or = log2(masc_df$case_control_factorXDP.OR) # the names of the case_control_factor<whatever> columns will change from project to project depending on the unique values of your Condition column
masc_df$log2_or_ci_low = log2(masc_df$case_control_factorXDP.OR.95pct.ci.lower)
masc_df$log2_or_ci_high = log2(masc_df$case_control_factorXDP.OR.95pct.ci.upper)

# order the graph to put disease-enriched populations on top
masc_df = masc_df[order(-masc_df$log2_or), ] 
masc_df$cluster_name = factor(masc_df$cluster_name, levels = masc_df$cluster_name[order(masc_df$log2_or)])
masc_df


library(RColorBrewer)
library(ggplot2)

# Create the forest plot with ticks on error bars, axis lines with ticks, RdBu color map, and opaque white circles on top
a = ggplot(masc_df, aes(y = cluster_name, x = log2_or)) +
  ggtitle("Cohort 1 + 2 Astrocytes (no CCF)") +
  geom_vline(xintercept = 0, linetype = "dotted", color = "gray") +  # Add dotted vertical line at x=0
  geom_segment(aes(x = log2_or_ci_low, xend = log2_or_ci_high, y = cluster_name, yend = cluster_name, color = log2_or), size = 1) +  # Add horizontal error bars
  geom_point(size = 3, aes(color = log2_or), shape = 1) +  # Add points for effect sizes
  geom_point(size = 3, shape = 21, fill = "white") +  # Add opaque white circle on top of the error bar line
  scale_color_gradientn(colors = RColorBrewer::brewer.pal(10, "RdBu")) +  # Use RdBu color map
  theme_minimal() +  # Minimal theme
  labs(x = "log2(OR)", y = "Cell Classes") +  # Axis labels
  theme(
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), 
    legend.position = "none",
    plot.title = element_text(size=16),
    axis.line = element_line(color = "black"),  # Add axis lines
    axis.ticks = element_line(color = "black"),  # Add axis ticks
    axis.text = element_text(size = 14),  # Increase tick label font size
    axis.title = element_text(size = 15)  # Increase axis label font size
  )

ggsave(a, filename = "pic.png", width = 7, height = 8)
```


```{r}
qsave(XDP_Cohorts_1_2_full, "WIP_XDP_Cohorts_1_2_full_032025.qs")
```

```{r}
xlab = "ScoreNFKB"
plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$ScoreNFKB,
                                          fill_col = "Condition_Cohort",
                                           colors = c("XDP_Cohort_1" = "red", "Control_Cohort_1" = "blue","XDP_Cohort_2" = "orange", "Control_Cohort_2" = "green"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL) 

plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$ScoreNFKB,
                                          fill_col = "Condition_Cohort",
                                           colors = c("XDP_Cohort_1" = "red", "Control_Cohort_1" = "blue","XDP_Cohort_2" = "orange", "Control_Cohort_2" = "green"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL) + facet_wrap(~GM_WM_astrocytes)

plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$ScoreNFKB,
                                          fill_col = "Condition",
                                           colors = c("XDP" = "red", "Control" = "blue"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL) + facet_wrap(~ GM_WM_astrocytes)

plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$ScoreNFKB,
                                          fill_col = "Cohort",
                                           colors = c("Cohort_1" = "red", "Cohort_2" = "blue"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL) + facet_wrap(~GM_WM_astrocytes)
```


```{r}
plot_overlapping_density_histogram(df = final_df, 
                                          hist_col = final_df$ScoreNFKB,
                                          fill_col = "Condition_Cohort",
                                           colors = c("XDP_Cohort_1" = "red", "Control_Cohort_1" = "blue","XDP_Cohort_2" = "orange", "Control_Cohort_2" = "green"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: Astrocytes"),
                                          xlab = xlab,
                                          fig_filename = NULL) + facet_wrap(~ Condition)

  

```

```{r}
Idents(sct_cleaned_CaH_Put_Cohort2) = "final_cell_class_merged"
DimPlot(sct_cleaned_CaH_Put_Cohort2, label = T)
FeaturePlot(sct_cleaned_CaH_Put_Cohort2, features = c("WIF1", "TNC"))
```

```{r}
cell_class_df
```


```{r}
# Create a contingency table
astro_counts <- matrix(c(WM_count_dataset1, GM_count_dataset1,
                         WM_count_dataset2, GM_count_dataset2),
                       nrow = 2, byrow = TRUE)

# Perform Chi-square test
chisq.test(astro_counts)

```


```{r}
prop.test(c(WM_count_dataset1, WM_count_dataset2),
          c(total_astrocytes_dataset1, total_astrocytes_dataset2))

```

```{r}
astro_counts <- as.matrix(table(XDP_Cohorts_1_2_full@meta.data$GM_WM_astrocytes, XDP_Cohorts_1_2_full@meta.data$Cohort),
                       nrow = 2, byrow = TRUE)
chisq.test(astro_counts)
chisq_result <- chisq.test(astro_counts)
chisq_result$stdres  # Extract standardized residuals

```

```{r}
donor_counts <- table(XDP_Cohorts_1_2_full@meta.data$GM_WM_astrocytes, 
                      XDP_Cohorts_1_2_full@meta.data$donor_id)
print(donor_counts)

apply(donor_counts, 2, function(x) chisq.test(matrix(x, nrow = 2))$p.value)

control_vs_case_counts <- table(XDP_Cohorts_1_2_full@meta.data$GM_WM_astrocytes, 
                                XDP_Cohorts_1_2_full@meta.data$Condition) # Assuming Diagnosis is "Control"/"Case"

print(control_vs_case_counts)
chisq.test(control_vs_case_counts)
chisq.test(control_vs_case_counts)$stdres

```


```{r}
astro_data <- data.frame(
  donor_id = XDP_Cohorts_1_2_full@meta.data$donor_id,
  cohort = XDP_Cohorts_1_2_full@meta.data$Cohort,
  condition = XDP_Cohorts_1_2_full@meta.data$Condition,
  astrocyte_type = XDP_Cohorts_1_2_full@meta.data$GM_WM_astrocytes
)
astro_data
astro_data$astrocyte_type <- ifelse(astro_data$astrocyte_type == "WM_astrocytes", 1, 0)

# Logistic regression predicting WM astrocytes based on cohort and diagnosis
glm_model <- glm(astrocyte_type ~ cohort + condition, data = astro_data, family = binomial)
summary(glm_model)


```
```{r}
cells_per_donor = as.data.frame(table(XDP_Cohorts_1_2_full@meta.data$Cohort))
cells_per_donor

cell_class_df = as.data.frame(table(XDP_Cohorts_1_2_full@meta.data$GM_WM_astrocytes, XDP_Cohorts_1_2_full@meta.data$Cohort))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df

cell_class_df$Cohort <- XDP_Cohorts_1_2_full@meta.data$Cohort[match(cell_class_df$Var2, XDP_Cohorts_1_2_full@meta.data$Cohort)]
cell_class_df

# Count the number of cells per donor
donor_counts <- table(XDP_Cohorts_1_2_full@meta.data$Cohort)

# Convert to dataframe for sorting
donor_df <- as.data.frame(donor_counts)
colnames(donor_df) <- c("Cohort", "cell_count")


cell_class_df
ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Cohort") + ylab("Number of cells by Astrocyte Subclasses") + labs(fill = "Astrocyte Subclass") +theme(axis.text.x = element_text(angle = 45, hjust = 1))  +facet_grid(~ Cohort, scales = "free_x", space = "free_x")

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Cohort") + ylab("Astrocyte Subclass Proportion") + labs(fill = "Astrocyte Subclass")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))   + facet_grid(~ Cohort, scales = "free_x", space = "free_x")
```

```{r}
FeaturePlot(XDP_Cohorts_1_2_full, features = "ScoreNFKB")
```


```{r}
XDP_neurons_Cohorts_1_2@meta.data
sct_cleaned_CaH_Put_Cohort2@meta.data
```
```{r}
table(XDP_neurons_Cohorts_1_2$final_neuron_subclass_merged)
```

```{r}
XDP_neurons_Cohorts_1_2@meta.data$final_SPN_merged = XDP_neurons_Cohorts_1_2$final_neuron_subclass_merged
XDP_neurons_Cohorts_1_2$final_SPN_merged[XDP_neurons_Cohorts_1_2$final_SPN_merged == "eSPN"] = "non_SPN"
XDP_neurons_Cohorts_1_2$final_SPN_merged[XDP_neurons_Cohorts_1_2$final_SPN_merged == "SPN_matrix"] = "SPN"
XDP_neurons_Cohorts_1_2$final_SPN_merged[XDP_neurons_Cohorts_1_2$final_SPN_merged == "SPN_patch"] = "SPN"
XDP_neurons_Cohorts_1_2$final_SPN_merged[XDP_neurons_Cohorts_1_2$final_SPN_merged == "SPN_patch_2"] = "SPN"

median_nUmi = XDP_neurons_Cohorts_1_2@meta.data %>%
  filter(final_SPN_merged == "SPN") %>%
  group_by(donor_id, Cohort, Condition) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi

a = ggplot(median_nUmi, aes(x = donor_id, y =median_UMI, color = Condition)) + geom_point() + theme(axis.text.x = element_text(angle = 45, hjust = 1))   + facet_grid(~ Cohort, scales = "free_x", space = "free_x") + ylab("Median SPN UMIs") + xlab("Donor")

ggsave(a, filename = "pic.png", width = 8, height = 4)
```


```{r}
unique(XDP_neurons_Cohorts_1_2$final_SPN_merged)
```
```{r}
xdp_co1
sct_cleaned_CaH_Put_Cohort2
```


```{r}
caudate_SPN = as.data.frame(table(XDP_neurons_Cohorts_1_2$donor_id, XDP_neurons_Cohorts_1_2$final_SPN_merged))
caudate_SPN = subset(caudate_SPN, subset = Var2 == "SPN")
caudate_SPN$CaH_SPN = caudate_SPN$Freq 
caudate_SPN$Freq = NULL

caudate_SPN$Condtion = XDP_neurons_Cohorts_1_2@meta.data$Condition[match(caudate_SPN$Var1, XDP_neurons_Cohorts_1_2@meta.data$donor_id)]
caudate_SPN

total_cells1 =as.data.frame(table(xdp_co1$donor_id))
total_cells1
total_cells2=as.data.frame(table(sct_cleaned_CaH_Put_Cohort2$donor_id))
total_cells2

total_cells = rbind(total_cells1,total_cells2)
total_cells
```


```{r}
merged_SPN_final = merge(caudate_SPN, total_cells, by = "Var1")
merged_SPN_final
merged_SPN_final$SPN_proportion = merged_SPN_final$CaH_SPN/merged_SPN_final$Freq
merged_SPN_final
```

```{r}
b= ggplot(data = merged_SPN_final, aes(x =SPN_proportion, y = Freq, color = Condtion)) + 
  geom_point(size = 4) +
  labs(x="SPN Fraction in All Cells (Cohort 1 and 2)" , y= "Total Cells")  + ggtitle("SPN Fraction in All Cells vs Total SPNs") + scale_color_discrete(name = "Condition")+ geom_text(aes(label = Var1), vjust = -1, hjust = 0.5) 

ggsave("SPN_fraction.png", plot= b, width = 20, height = 10)
```

```{r}
total_nuclei	mUMI_all_nuclei	total_SPN_nuclei	SPN_mUMI	SPN fraction of total nuclei	median_pct_intronic	median_pct_mito	median_number_of_genes
```



```{r}
#totalcells
cells_per_donor <- sct_cleaned_CaH_Put_Cohort2@meta.data %>% 
  group_by(donor_id) %>%
  summarize(n = n())
cells_per_donor

#median nUMI
median_nUmi = sct_cleaned_CaH_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
median_nUmi

SPN_cells_per_library <- XDP_neurons_Cohorts_1_2@meta.data %>% 
  filter(final_SPN_merged == "SPN") %>%
  group_by(donor_id) %>%
  summarize(n = n())
SPN_cells_per_library

SPN_median_nUmi = XDP_neurons_Cohorts_1_2@meta.data %>%
  filter(final_SPN_merged == "SPN") %>%
  group_by(donor_id, Cohort, Condition) %>%
  summarize(median_UMI = median(nUmi, na.rm = TRUE))
SPN_median_nUmi


#median/mean nGenes
median_nGenes = sct_cleaned_CaH_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_nGenes = median(nFeature_RNA, na.rm = TRUE))
median_nGenes


#median/mean pctintronic
median_pct_intronic = sct_cleaned_CaH_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_intronic = median(pct_intronic, na.rm = TRUE))
median_pct_intronic

#median/mean pctmito
median_pct_mito = sct_cleaned_CaH_Put_Cohort2@meta.data %>%
  group_by(donor_id) %>%
  summarize(median_pct_mito = median(pct_mito, na.rm = TRUE))
median_pct_mito

# List of data frames to merge
df_list <- list(cells_per_donor, median_nUmi, median_nGenes, median_pct_intronic, median_pct_mito )

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "donor_id"))

# Print the merged data frame
print(merged_df)

write.csv(merged_df, "merged_df.csv")
```

```{r}
df_list <- list(SPN_cells_per_library, SPN_median_nUmi)

# Merge all data frames on the common column Donor.ID
merged_df <- reduce(df_list, function(x, y) merge(x, y, by = "donor_id"))

# Print the merged data frame
print(merged_df)

write.csv(merged_df, "merged_df.csv")
```














#Neuron Ucells
```{r}
library(UCell)

NFKB_genes =  c("MARCKS", "IL23A", "NINJ1", "TNFSF9", "SIK1", "ATF3", "SERPINE1", "MYC", "HES1", "CCNL1", "CCN1", "EGR1", "EGR2", "JAG1", "EGR3", "ABCA1", "GADD45B", "GADD45A", "PLK2", "KLF10", 
           "EIF1", "EHD1", "FOSL2", "FOSL1", "GPR183", "PLPP3", "IFIT2", "ICAM1", "ZC3H12A", "IER2", 
           "IL12B", "JUNB", "IER5", "IER3", "STAT5A", "DUSP5", "EDN1", "JUN", "DUSP4", "DUSP1", 
           "DUSP2", "TSC22D1", "CCL20", "SPHK1", "LIF", "IL18", "TUBB2A", "RHOB", "VEGFA", "PTPRE", 
           "IL1A", "TLR2", "IL1B", "BHLHE40", "ID2", "CLCF1", "REL", "FJX1", "SGK1", "BTG3", "BTG2", 
           "BTG1", "SDC4", "LITAF", "AREG", "SOCS3", "PANX1", "RIPK2", "NFIL3", "SERPINB2", "GCH1", 
           "IFNGR2", "G0S2", "FOS", "SERPINB8", "F3", "SPSB1", "FOSB", "PER1", "F2RL1", "HBEGF", 
           "CD44", "TRIP10", "CDKN1A", "PTGER4", "PTGS2", "IFIH1", "NAMPT", "OLR1", "ICOSLG", 
           "PHLDA1", "ZBTB10", "TAP1", "PNRC1", "CXCL10", "CXCL11", "IL6ST", "CD69", "SQSTM1", 
           "RELA", "CSF2", "CD83", "CSF1", "PPP1R15A", "CD80", "TNC", "TNF", "TANK", "RELB", "ZFP36", 
           "CCND1", "RNF19B", "CCRL2", "DENND5A", "PHLDA2", "MAP3K8", "LDLR", "SLC16A6", "SMAD3", 
           "TGIF1", "MAP2K3", "DDX58", "TRAF1", "INHBA", "NFKB1", "NFKB2", "GEM", "NR4A3", "MAFF", 
           "RCAN1", "NR4A2", "EFNA1", "MXD1", "BIRC2", "YRDC", "BIRC3", "IL7R", "PFKFB3", "IRS2", 
           "SLC2A3", "PLAU", "SLC2A6", "SAT1", "ETS2", "NR4A1", "SNN", "PMEPA1", "TNFRSF9", "MSC", 
           "TIPARP", "LAMB3", "GFPT2", "CFLAR", "TNIP1", "IRF1", "NFKBIA", "BMP2", "IL6", "TNIP2", 
           "BCL6", "BCL3", "NFKBIE", "NFE2L2", "B4GALT1", "NFAT5", "TNFAIP8", "BCL2A1", "TNFAIP6", 
           "TNFAIP3", "CXCL2", "CXCL1", "TNFAIP2", "CXCL3", "CXCL6", "FUT4", "DRAM1", "DNAJB4", 
           "PDE4B", "PDLIM5", "MCL1", "KDM6B", "IL15RA", "PLAUR", "ATP2B1", "KLF4", "KLF2", "SOD2", 
           "KLF9", "KLF6", "ACKR3", "PTX3", "B4GALT5", "TRIB1", "CEBPB", "CEBPD", "PLEK", "KYNU", 
           "CCL5", "CCL4", "CCL2")

interferon_alpha = c("IL4R", "CD74", "SP110", "MX1", "RSAD2", "TAP1", "PSMB9", "CXCL10", "CXCL11", "IFI44L", "LAP3", "LPAR6", "IFI27", "TRAFD1", "TRIM14", "LY6E", "IFITM2", "IFITM3", "SAMD9", "CSF1", 
           "IFITM1", "UBE2L6", "SAMD9L", "RNF31", "HERC6", "LAMP3", "CCRL2", "MVB12A", "B2M", "CMTR1", 
           "GBP2", "GBP4", "EPSTI1", "BATF2", "HLA-C", "TENT5A", "TDRD7", "ISG15", "IFI44", "PARP14", 
           "MOV10", "ISG20", "PARP12", "PLSCR1", "PROCR", "SELL", "PSME2", "PSME1", "CNP", "LGALS3BP", 
           "IFI35", "IFIT2", "IFI30", "DDX60", "IFIT3", "NUB1", "HELZ2", "PNPT1", "IL15", "NCOA7", 
           "EIF2AK2", "NMI", "OAS1", "BST2", "IRF1", "ELF1", "IL7", "IRF2", "IRF9", "IRF7", "UBA7", 
           "GMPR", "ADAR", "CASP8", "TRIM5", "RIPK2", "CASP1", "STAT2", "WARS1", "PARP9", "PSMA3", 
           "TXNIP", "CMPK2", "CD47", "RTP4", "C1S", "OGFR", "TMEM140", "USP18", "OASL", "SLC25A28", 
           "IFIH1", "PSMB8", "DHX58", "TRIM25", "TRIM26", "TRIM21")

antigen_processing_genes = c("CTSB", "CD74", "HSP90AA1", "KLRC3", "HSPA6", "HSPA5", "HSPA1L", "KLRC4", "HSPA8", "HSPA2", "TAP2", "HSPA4", "TAP1", "TAPBP", "CD4", "RFX5", "IFNG", "HSPA1A", "HLA-DRA", "HLA-DQB1", "KLRD1", "CALR", "HSPA1B", "TNF", "KIR2DL1", "KIR2DL2", "KIR2DL3", "KIR2DL4", "CTSS", "CTSL", "B2M", "HLA-DOB", "KIR2DL5A", "CIITA", "HLA-DQA2", "HLA-DOA", "HLA-DQA1", "PDIA3", "HLA-DRB5", 
"KIR2DS1", "NFYB", "KIR2DS2", "NFYC", "KIR2DS3", "KIR2DS4", "KIR2DS5", "HLA-C", "RFXANK", 
"KIR3DL1", "CD8B2", "KIR3DL2", "HLA-A", "HLA-B", "KIR3DL3", "HLA-G", "HLA-E", "HLA-F", "CREB1", "CD8B", "RFXAP", "CD8A", "CANX", "LGMN", "PSME2", "PSME3", "HLA-DPB1", "PSME1", "HLA-DRB4", 
"HLA-DRB3", "HLA-DRB1", "HSP90AB1", "IFI30", "HLA-DMB", "HLA-DPA1", "KLRC1", "KLRC2", "NFYA","HLA-DMA")
C_minus = qread("temp_disco/more_temp_1205/final_c_minus_genes.qs")
C_plus = qread("temp_disco/more_temp_1205/final_c_plus_genes.qs")

XDP_Cohorts_1_2_full <- AddModuleScore_UCell(XDP_Cohorts_1_2_full, features = list(Score = NFKB_genes), name = 'NFKB')

XDP_Cohorts_1_2_full <- AddModuleScore_UCell(XDP_Cohorts_1_2_full, features = list(Score = interferon_alpha), name = 'interferon_alpha'
)

XDP_Cohorts_1_2_full <- AddModuleScore_UCell(XDP_Cohorts_1_2_full,features = list(antigen_processing = antigen_processing_genes), name = '_scores')

XDP_Cohorts_1_2_full <- AddModuleScore_UCell(XDP_Cohorts_1_2_full,features = list(C_minus = C_minus), name = '_scores')
XDP_Cohorts_1_2_full <- AddModuleScore_UCell(XDP_Cohorts_1_2_full,features = list(C_plus = C_plus), name = '_scores')

qsave(XDP_Cohorts_1_2_full, "XDP_Cohorts_1_2_full_sct_ucell_032125.qs")
```
```{r}
# Count the number of cells per donor
donor_counts <- table(XDP_Cohorts_1_2_full@meta.data$donor_id)

# Convert to dataframe for sorting
donor_df <- as.data.frame(donor_counts)
colnames(donor_df) <- c("donor_id", "cell_count")

# Identify controls (CM, CF) - ensuring it only checks ENDING
controls <- donor_df[grepl("CM$|CF$", donor_df$donor_id), ]
cases <- donor_df[!grepl("CM$|CF$", donor_df$donor_id), ]

# Sort both groups by cell count in descending order
controls <- controls[order(-controls$cell_count), ]
cases <- cases[order(-cases$cell_count), ]

# Combine ordered donor IDs
ordered_donors <- c(controls$donor_id, cases$donor_id)

# Print or use the ordered donor IDs
print(ordered_donors)


```


```{r}
cells_per_donor = as.data.frame(table(XDP_Cohorts_1_2_full@meta.data$donor_id))
cells_per_donor

cell_class_df = as.data.frame(table(XDP_Cohorts_1_2_full@meta.data$final_neuron_subclass_merged, XDP_Cohorts_1_2_full@meta.data$donor_id))


cell_class_df = merge(cell_class_df, cells_per_donor, by.x="Var2", by.y = "Var1")
cell_class_df

cell_class_df$cell_proportions = cell_class_df$Freq.x/cell_class_df$Freq.y
cell_class_df


cell_class_df

# Count the number of cells per donor
donor_counts <- table(XDP_Cohorts_1_2_full@meta.data$donor_id)

# Convert to dataframe for sorting
donor_df <- as.data.frame(donor_counts)
colnames(donor_df) <- c("donor_id", "cell_count")

# Identify controls (CM, CF) - ensuring it only checks ENDING
controls <- donor_df[grepl("CM$|CF$|CM2$", donor_df$donor_id), ]
cases <- donor_df[!grepl("CM$|CF$|CM2$", donor_df$donor_id), ]

# Sort both groups by cell count in descending order
controls <- controls[order(-controls$cell_count), ]
cases <- cases[order(-cases$cell_count), ]

# Combine ordered donor IDs
ordered_donors <- c(controls$donor_id, cases$donor_id)

# Print or use the ordered donor IDs
print(ordered_donors)


            
cell_class_df$Var2 = factor(cell_class_df$Var2, levels = ordered_donors)

ggplot(cell_class_df, aes(x = Var2, y = Freq.x, fill = Var1)) +
  geom_bar(stat = "identity") + xlab("Donors") + ylab("Number of cells by Neuron Subclasses") + labs(fill = "Neuron Subclass")+ geom_vline(xintercept =  12.5, linetype = "dashed", color = "black") +theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(cell_class_df, aes(x = Var2, y = cell_proportions, fill = Var1)) +
  geom_bar(stat = "identity", postion= "stack") + xlab("Donors") + ylab("Neuron Subclass Proportion") + labs(fill = "Neuron Subclass")+ geom_vline(xintercept =  12.5, linetype = "dashed", color = "black")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  
```


```{r}
histograms_by_all_celltype = function(final_df, score_col, xlab, width_size, height_size){

#celltypes = c("astrocyte", "microglia", "oligo", "opc", "endothelial","immune", "ependymal", "non_SPN", "eSPN", "SPN_matrix", "SPN_patch", "SPN_exotic")

 celltypes = unique(final_df[["final_neuron_cell_class_harmony_merged"]])  
plots <- list()

for (celltype in celltypes) {
  df_sub = final_df[final_df$final_neuron_cell_class_harmony_merged == celltype, ]
   
  if (nrow(df_sub) == 0) {
    message(paste("No data for cell type:", celltype, "- skipping."))
    next
  }
  
min = min(df_sub[[score_col]])
max = max(df_sub[[score_col]])
step = (max-min)/100

limits = seq(min, max, step)
  
  a = plot_overlapping_density_histogram(df = df_sub, 
                                          hist_col = df_sub[[score_col]],
                                          fill_col = "Condition",
                                          colors = c("XDP" = "red", "Control" = "blue","BICAN_V8" = "green", "pd" = "red", "ctr" = "blue", "XDP_18_006" = "orange"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: ", celltype),
                                          xlab = xlab,
                                          fig_filename = NULL)
  
    plots[[celltype]] = a
  
}

final_plot <- plot_grid(plotlist = plots, ncol = 5)

# Display or save the final plot
print(final_plot)  # Display
ggsave("combined_plot.png", final_plot, width = width_size, height = height_size)
}


histograms_by_all_celltype_cohort = function(final_df, score_col, xlab, width_size, height_size){

 celltypes = sort(unique(final_df[["final_neuron_cell_class_harmony_merged"]]))
plots <- list()

for (celltype in celltypes) {
  df_sub = final_df[final_df$final_neuron_cell_class_harmony_merged == celltype, ]
   
  if (nrow(df_sub) == 0) {
    message(paste("No data for cell type:", celltype, "- skipping."))
    next
  }
  
min = min(df_sub[[score_col]])
max = max(df_sub[[score_col]])
step = (max-min)/100

limits = seq(min, max, step)
  
  a = plot_overlapping_density_histogram(df = df_sub, 
                                          hist_col = df_sub[[score_col]],
                                          fill_col = "Condition_Cohort",
                                          colors = c("XDP_Cohort_1" = "red", "Control_Cohort_1" = "blue","XDP_Cohort_2" = "orange", "Control_Cohort_2" = "green"),
                                          breaks = limits,
                                          title = paste0("XDP vs Control: ", celltype),
                                          xlab = xlab,
                                          fig_filename = NULL)
  
    plots[[celltype]] = a
  
}

final_plot <- plot_grid(plotlist = plots, ncol = 5)

# Display or save the final plot
print(final_plot)  # Display
ggsave("combined_plot.png", final_plot, width = width_size, height = height_size)
}

histograms_by_donor <- function(final_df, score_col, celltype, xlab, width_size, height_size) {
  
  unique_donors <- unique(final_df$donor_id)

  # Identify controls and cases
  controls <- unique_donors[grepl("CM$|CF$|CM2$", unique_donors)]
  cases <- unique_donors[!grepl("CM$|CF$|CM2$", unique_donors)]

  # Alphabetically sort both groups
  controls <- sort(controls)
  cases <- sort(cases)

  # Combine ordered donor IDs (cases first, then controls)
  ordered_donors <- c(cases, controls)
  
  print(ordered_donors)

  plots <- list()

  # Subset dataframe for the specific cell type
  final_df_cellclass <- subset(final_df, final_neuron_cell_class_harmony_merged == celltype)

  # Determine global x-axis limits (bins)
  min_val <- min(final_df_cellclass[[score_col]], na.rm = TRUE)
  max_val <- max(final_df_cellclass[[score_col]], na.rm = TRUE)
  step <- (max_val - min_val) / 100
  limits <- seq(min_val, max_val, step)

  # **Determine global y-axis limit** by finding the maximum count across all donors
  max_peak_y <- 0  # Initialize
  
   for (donor in ordered_donors) {
    df_sub <- final_df_cellclass[final_df_cellclass$donor_id == donor, ]
    if (nrow(df_sub) > 0) {
      hist_data <- hist(df_sub[[score_col]], breaks = limits, plot = FALSE)
      max_peak_y <- max(max_peak_y, max(hist_data$density, na.rm = TRUE))  # Use density (peak) instead of total count
    }
  }
  
  # **Now generate histograms with consistent y-axis limits**
  for (donor in ordered_donors) {
    df_sub <- final_df_cellclass[final_df_cellclass$donor_id == donor, ]
    
    if (nrow(df_sub) == 0) {
      message(paste("No data for donor:", donor, "- skipping."))
      next
    }

    cell_count <- nrow(df_sub)
    
    a <- plot_overlapping_density_histogram(
      df = df_sub, 
      hist_col = df_sub[[score_col]],
      fill_col = "Condition",
      colors = c("XDP" = "red", "Control" = "blue", "BICAN_V8" = "green", 
                 "pd" = "red", "ctr" = "blue", "XDP_18_006" = "orange"),
      breaks = limits,
      title = paste0(donor, ": ", celltype, " (Cells: ", cell_count, ")"),
      xlab = xlab,
      fig_filename = NULL
    )
    
    # Modify the y-axis limit directly using ggplot2
    a <- a + ggplot2::ylim(0, max_peak_y)  # Set consistent y-axis limit
    
    plots[[donor]] <- a
  }
  
  final_plot <- plot_grid(plotlist = plots, ncol = 5)
  
  print(final_plot)  
  ggsave("combined_plot.png", final_plot, width = width_size, height = height_size)
}

```



```{r}
histograms_by_donor(XDP_Cohorts_1_2_full@meta.data, "antigen_processing_scores", "SPN_matrix" ,"antigen_processing_scores", 25, 15)
```


```{r}
library(dplyr)
library(ggplot2)

data <- xdp_co1_meta
df = subset(data, subset = final_cell_class == "neuron")
df
df_avg <- df %>%
  group_by(donor_id, cause_of_death_case_control, Condition, cause_of_death) %>%
  summarise(avg_score = mean(Scoreinterferon_alpha), .groups = 'drop')
df_avg

# Plot the average score per donor for each condition
a = ggplot(df_avg, aes(x = Condition, y = avg_score, color = Condition, shape = cause_of_death)) +
  geom_point(aes(group = donor_id), size = 3, alpha = 0.8) + scale_color_manual(values = rev(scales::hue_pal()(2)))+ # One point per donor
  labs(x = "Condition", y = "Mean interferonalpha Score", color = "Condition", shape = "Cause of Death", title = "Astrocytes")+ scale_shape_manual(values = rev(c(16, 17))) +
  theme_minimal()+ theme(axis.text.x = element_text(angle = 45, hjust = 1))  + theme(
            plot.title = element_text(size = 20), # title font size
            axis.line = element_line(color = "black"),  # Add axis lines
            axis.ticks = element_line(color = "black"),  # Add axis ticks
            axis.text = element_text(size = 14),  # Increase tick label font size
            axis.title = element_text(size = 15),  # Increase axis label font size
            axis.text.x = element_text(angle = 45, hjust = 1)  # tilt axis labels
        )+
  theme(
    legend.text = element_text(size = 16),  # Increase legend text size
    legend.title = element_text(size = 18),  # Increase legend title size
   # legend.key.size = unit(1.5, "cm")  # Increase legend key size
  ) 

print(a)

ggsave(a, filename = "dotplot.png", width = 7, height = 6)

```


#wilcox test
```{r}
xdp_meta = xdp_co1_meta
xdp_meta = subset(xdp_meta, final_cell_class == "astrocyte")
xdp_meta

xdp_meta_cases = subset(xdp_meta, Condition == "XDP")
xdp_meta_controls = subset(xdp_meta, Condition == "Control")
xdp_meta_cases
xdp_meta_controls

XDP_scores <- xdp_meta_cases$ScoreNFKB  
XDP_Control_scores <- xdp_meta_controls$ScoreNFKB 

result1 <- wilcox.test(XDP_scores, XDP_Control_scores, alternative = "two.sided")
result2 <- wilcox.test(XDP_scores, XDP_Control_scores, alternative = "less")
result3 <- wilcox.test(XDP_scores, XDP_Control_scores, alternative = "greater")
print(result1)
print(result2)
print(result3)
```



```{r}
cleaned_CaH_Put_Cohort2_meta
xdp_co1_meta
```


```{r}
histograms_by_all_celltype(cleaned_CaH_Put_Cohort2_meta, "C_minus_scores", "C_minus_scores", 20, 10)
```

```{r}
histograms_by_all_celltype(xdp_co1_meta, "C_minus_scores", "C_minus_scores", 20, 10)
```


```{r}
histograms_by_donor(xdp_co1_meta, "ScoreNFKB", "astrocyte" ,"ScoreNFKB", 25, 15)
```

```{r}
table(cleaned_CaH_Put_Cohort2_meta$neuron_cell_class)
```


```{r}
xdp_cohorts_meta
```



```{r}
library(lme4)
sobj = xdp_cohorts_meta#XDP_Cohorts_1_2_full
sobj = subset(sobj, donor_id != "SCF_22_049CCF")
model_cols = c("donor_id", "final_cell_class_merged_harmony", "Condition", "Sex", "Cohort") #include any other covariates of interest, replace cell_class with whatever column defines the cluster annotations
pd = sobj[,model_cols]
pd = pd[complete.cases(pd),] # don't want any NAs
pd$case_control_factor = as.factor(pd$Condition)

masc_df = .sconline.MASCfn(
    dataset=pd,
    cluster=pd$final_cell_class_merged_harmony, # cluster annotations
    contrast="case_control_factor", # name of contrast annotations (what you want to run the test for)
    random_effects=c("donor_id"), # name of random effects annotations (not interested in these coefficients, but account for variability in probable sources ob batch effects, in this case donor
    fixed_effects = c("Cohort", "Sex") # your covariates
)

masc_df

my_order <- unique(masc_df$cluster)
masc_df$cluster_name <- sub("cluster", "", masc_df$cluster) # for legibility
masc_df$log2_or = log2(masc_df$case_control_factorXDP.OR) # the names of the case_control_factor<whatever> columns will change from project to project depending on the unique values of your Condition column
masc_df$log2_or_ci_low = log2(masc_df$case_control_factorXDP.OR.95pct.ci.lower)
masc_df$log2_or_ci_high = log2(masc_df$case_control_factorXDP.OR.95pct.ci.upper)

# order the graph to put disease-enriched populations on top
masc_df = masc_df[order(-masc_df$log2_or), ] 
masc_df$cluster_name = factor(masc_df$cluster_name, levels = masc_df$cluster_name[order(masc_df$log2_or)])
masc_df


library(RColorBrewer)
library(ggplot2)

# Create the forest plot with ticks on error bars, axis lines with ticks, RdBu color map, and opaque white circles on top
a = ggplot(masc_df, aes(y = cluster_name, x = log2_or)) +
  ggtitle("Cohort 1 + 2 Astrocytes (no CCF)") +
  geom_vline(xintercept = 0, linetype = "dotted", color = "gray") +  # Add dotted vertical line at x=0
  geom_segment(aes(x = log2_or_ci_low, xend = log2_or_ci_high, y = cluster_name, yend = cluster_name, color = log2_or), size = 1) +  # Add horizontal error bars
  geom_point(size = 3, aes(color = log2_or), shape = 1) +  # Add points for effect sizes
  geom_point(size = 3, shape = 21, fill = "white") +  # Add opaque white circle on top of the error bar line
  scale_color_gradientn(colors = RColorBrewer::brewer.pal(10, "RdBu")) +  # Use RdBu color map
  theme_minimal() +  # Minimal theme
  labs(x = "log2(OR)", y = "Cell Classes") +  # Axis labels
  theme(
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), 
    legend.position = "none",
    plot.title = element_text(size=16),
    axis.line = element_line(color = "black"),  # Add axis lines
    axis.ticks = element_line(color = "black"),  # Add axis ticks
    axis.text = element_text(size = 14),  # Increase tick label font size
    axis.title = element_text(size = 15)  # Increase axis label font size
  )

ggsave(a, filename = "pic.png", width = 7, height = 8)
```








